<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • PublicDataResources</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="PublicDataResources">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PublicDataResources</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/PublicDataResources.html">Workshop</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    External Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html">PharmacoGx</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/bioc/vignettes/GEOquery/inst/doc/GEOquery.html">GEOquery</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/devel/bioc/html/TCGAbiolinks.html">TCGAbiolinks</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicDataCommons/inst/doc/overview.html">GenomicDataCommons</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/curatedTCGAData/inst/doc/curatedTCGAData.html">curatedTCGAData</a>
    </li>
    <li>
      <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/TCGAutils/inst/doc/TCGAutils.html">TCGAutils</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/curatedMetagenomicData/inst/doc/curatedMetagenomicData.html">curatedMetagenomicData</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/html/HMP16SData.html">HMP16SData</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/PublicDataResources">
    <span class="fas fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://hub.docker.com/repository/docker/waldronlab/publicdataresources">
    <span class="fas fa-docker"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PublicDataResources_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>PublicDataResources.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>If you are using the Docker container provided with this workshop, everything (including the workshop) is already installed. Otherwise, you will need to install the following packages.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">required_pkgs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"GEOquery"</span>, 
  <span class="st">"GenomicDataCommons"</span>,
  <span class="st">"curatedTCGAData"</span>,
  <span class="st">"TCGAutils"</span>,
  <span class="st">"cBioPortalData"</span>,
  <span class="st">"recount"</span>,
  <span class="st">"curatedMetagenomicData"</span>,
  <span class="st">"phyloseq"</span>,
  <span class="st">"HMP16SData"</span>,
  <span class="st">"PharmacoGx"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="va">required_pkgs</span><span class="op">)</span></code></pre></div>
</div>
<div id="sean-davis" class="section level1">
<h1 class="hasAnchor">
<a href="#sean-davis" class="anchor"></a>Sean Davis</h1>
<div id="geoquery" class="section level2">
<h2 class="hasAnchor">
<a href="#geoquery" class="anchor"></a>GEOquery</h2>
<p><span class="citation">(Davis and Meltzer 2007)</span></p>
<p>The NCBI Gene Expression Omnibus (GEO) serves as a public repository for a wide range of high-throughput experimental data. These data include single and dual channel microarray-based experiments measuring mRNA, genomic DNA, and protein abundance, as well as non-array techniques such as serial analysis of gene expression (SAGE), mass spectrometry proteomic data, and high-throughput sequencing data. The <em><a href="https://bioconductor.org/packages/3.12/GEOquery">GEOquery</a></em> package <span class="citation">(Davis and Meltzer 2007)</span> forms a bridge between this public repository and the analysis capabilities in Bioconductor.</p>
<div id="overview-of-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-of-geo" class="anchor"></a>Overview of GEO</h3>
<p>At the most basic level of organization of GEO, there are four basic entity types. The first three (Sample, Platform, and Series) are supplied by users; the fourth, the dataset, is compiled and curated by GEO staff from the user-submitted data. See <a href="https://www.ncbi.nlm.nih.gov/geo/">the GEO home page</a> for more information.</p>
<div id="platforms" class="section level4">
<h4 class="hasAnchor">
<a href="#platforms" class="anchor"></a>Platforms</h4>
<p>A Platform record describes the list of elements on the array (e.g., cDNAs, oligonucleotide probesets, ORFs, antibodies) or the list of elements that may be detected and quantified in that experiment (e.g., SAGE tags, peptides). Each Platform record is assigned a unique and stable GEO accession number (GPLxxx). A Platform may reference many Samples that have been submitted by multiple submitters.</p>
</div>
<div id="samples" class="section level4">
<h4 class="hasAnchor">
<a href="#samples" class="anchor"></a>Samples</h4>
<p>A Sample record describes the conditions under which an individual Sample was handled, the manipulations it underwent, and the abundance measurement of each element derived from it. Each Sample record is assigned a unique and stable GEO accession number (GSMxxx). A Sample entity must reference only one Platform and may be included in multiple Series.</p>
</div>
<div id="series" class="section level4">
<h4 class="hasAnchor">
<a href="#series" class="anchor"></a>Series</h4>
<p>A Series record defines a set of related Samples considered to be part of a group, how the Samples are related, and if and how they are ordered. A Series provides a focal point and description of the experiment as a whole. Series records may also contain tables describing extracted data, summary conclusions, or analyses. Each Series record is assigned a unique and stable GEO accession number (GSExxx). Series records are available in a couple of formats which are handled by GEOquery independently. The smaller and new GSEMatrix files are quite fast to parse; a simple flag is used by GEOquery to choose to use GSEMatrix files (see below).</p>
</div>
<div id="datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h4>
<p>GEO DataSets (GDSxxx) are curated sets of GEO Sample data. There are hundreds of GEO datasets available, but GEO discontinued creating GDS records several years ago. We mention them here for completeness only.</p>
</div>
</div>
<div id="getting-started-using-geoquery" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started-using-geoquery" class="anchor"></a>Getting Started using GEOquery</h3>
<p>Getting data from GEO is really quite easy. There is only one command that is needed, <code>getGEO</code>. This one function interprets its input to determine how to get the data from GEO and then parse the data into useful R data structures.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/seandavi/GEOquery">GEOquery</a></span><span class="op">)</span></code></pre></div>
<p>With the library loaded, we are free to access any GEO accession.</p>
</div>
<div id="use-case-mds-plot-of-cancer-data" class="section level3">
<h3 class="hasAnchor">
<a href="#use-case-mds-plot-of-cancer-data" class="anchor"></a>Use case: MDS plot of cancer data</h3>
<p>The data we are going to access are from <a href="https://doi.org/10.1158/1055-9965.EPI-17-0461">this paper</a>.</p>
<blockquote>
<p>Background: The tumor microenvironment is an important factor in cancer immunotherapy response. To further understand how a tumor affects the local immune system, we analyzed immune gene expression differences between matching normal and tumor tissue.Methods: We analyzed public and new gene expression data from solid cancers and isolated immune cell populations. We also determined the correlation between CD8, FoxP3 IHC, and our gene signatures.Results: We observed that regulatory T cells (Tregs) were one of the main drivers of immune gene expression differences between normal and tumor tissue. A tumor-specific CD8 signature was slightly lower in tumor tissue compared with normal of most (12 of 16) cancers, whereas a Treg signature was higher in tumor tissue of all cancers except liver. Clustering by Treg signature found two groups in colorectal cancer datasets. The high Treg cluster had more samples that were consensus molecular subtype 1/4, right-sided, and microsatellite-instable, compared with the low Treg cluster. Finally, we found that the correlation between signature and IHC was low in our small dataset, but samples in the high Treg cluster had significantly more CD8+ and FoxP3+ cells compared with the low Treg cluster.Conclusions: Treg gene expression is highly indicative of the overall tumor immune environment.Impact: In comparison with the consensus molecular subtype and microsatellite status, the Treg signature identifies more colorectal tumors with high immune activation that may benefit from cancer immunotherapy.</p>
</blockquote>
<p>In this little exercise, we will:</p>
<ol style="list-style-type: decimal">
<li>Access public omics data using the GEOquery package</li>
<li>Convert the public omics data to a <code>SummarizedExperiment</code> object.</li>
<li>Perform a simple unsupervised analysis to visualize these public data.</li>
</ol>
<p>Use the <a href="https://bioconductor.org/packages/GEOquery">GEOquery</a> package to fetch data about <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE103512">GSE103512</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html">getGEO</a></span><span class="op">(</span><span class="st">"GSE103512"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code>## Warning: 102 parsing failures.
##   row     col           expected    actual         file
## 54614 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54615 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54616 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54617 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54618 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## ..... ....... .................. ......... ............
## See problems(...) for more details.</code></pre>
<p>Note that <code>getGEO</code>, when used to retrieve <em>GSE</em> records, returns a list. The members of the list each represent one <em>GEO Platform</em>, since each <em>GSE</em> record can contain multiple related datasets (eg., gene expression and DNA methylation). In this case, the list is of length one, but it is still necessary to grab the first elment.</p>
<p>The first step–a detail–is to convert from the older Bioconductor data structure (GEOquery was written in 2007), the <code>ExpressionSet</code>, to the newer <code>SummarizedExperiment</code>. One line suffices.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="va">se</span> <span class="op">=</span> <span class="fu">as</span><span class="op">(</span><span class="va">gse</span>, <span class="st">"SummarizedExperiment"</span><span class="op">)</span></code></pre></div>
<p>Examine two variables of interest, cancer type and tumor/normal status.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">`cancer.type.ch1`</span>,<span class="va">`normal.ch1`</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                normal.ch1
## cancer.type.ch1 no yes
##           BC    65  10
##           CRC   57  12
##           NSCLC 60   9
##           PCA   60   7</code></pre>
<p>Filter gene expression by variance to find most informative genes.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sds</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'exprs'</span><span class="op">)</span>,<span class="fl">1</span>,<span class="va">sd</span><span class="op">)</span>
<span class="va">dat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'exprs'</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">sds</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span><span class="op">]</span>,<span class="op">]</span></code></pre></div>
<p>Perform <a href="https://en.wikipedia.org/wiki/Multidimensional_scaling">multidimensional scaling</a> and prepare for plotting. We will be using ggplot2, so we need to make a data.frame before plotting.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mdsvals</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cmdscale.html">cmdscale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">mdsvals</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">mdsvals</span><span class="op">)</span>
<span class="va">mdsvals</span><span class="op">$</span><span class="va">Type</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span>,<span class="st">'cancer.type.ch1'</span><span class="op">]</span><span class="op">)</span>
<span class="va">mdsvals</span><span class="op">$</span><span class="va">Normal</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span>,<span class="st">'normal.ch1'</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mdsvals</span><span class="op">)</span></code></pre></div>
<pre><code>##                   V1        V2 Type Normal
## GSM2772660  8.531331 -18.57115   BC     no
## GSM2772661  8.991591 -13.63764   BC     no
## GSM2772662 10.788973 -13.48403   BC     no
## GSM2772663  3.127105 -19.13529   BC     no
## GSM2772664 13.056599 -13.88711   BC     no
## GSM2772665  7.903717 -13.24731   BC     no</code></pre>
<p>And do the plot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">mdsvals</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">V1</span>,y<span class="op">=</span><span class="va">V2</span>,shape<span class="op">=</span><span class="va">Normal</span>,color<span class="op">=</span><span class="va">Type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span> alpha<span class="op">=</span><span class="fl">0.6</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="PublicDataResources_files/figure-html/mdsplot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="accessing-raw-data-from-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-raw-data-from-geo" class="anchor"></a>Accessing Raw Data from GEO</h3>
<p>NCBI GEO accepts (but has not always required) raw data such as .CEL files, .CDF files, images, etc. It is also not uncommon for some RNA-seq or other sequencing datasets to supply <em>only</em> raw data (with accompanying sample information, of course), necessitating Sometimes, it is useful to get quick access to such data. A single function, <code>getGEOSuppFiles</code>, can take as an argument a GEO accession and will download all the raw data associate with that accession. By default, the function will create a directory in the current working directory to store the raw data for the chosen GEO accession.</p>
</div>
</div>
<div id="genomicdatacommons" class="section level2">
<h2 class="hasAnchor">
<a href="#genomicdatacommons" class="anchor"></a>GenomicDataCommons</h2>
<p>From the <a href="https://gdc.nci.nih.gov/about-gdc">Genomic Data Commons (GDC) website</a>:</p>
<blockquote>
<p>The National Cancer Institute’s (NCI’s) Genomic Data Commons (GDC) is a data sharing platform that promotes precision medicine in oncology. It is not just a database or a tool; it is an expandable knowledge network supporting the import and standardization of genomic and clinical data from cancer research programs. The GDC contains NCI-generated data from some of the largest and most comprehensive cancer genomic datasets, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Therapies (TARGET). For the first time, these datasets have been harmonized using a common set of bioinformatics pipelines, so that the data can be directly compared. As a growing knowledge system for cancer, the GDC also enables researchers to submit data, and harmonizes these data for import into the GDC. As more researchers add clinical and genomic data to the GDC, it will become an even more powerful tool for making discoveries about the molecular basis of cancer that may lead to better care for patients.</p>
</blockquote>
<p>The <a href="https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components">data model for the GDC is complex</a>, but it worth a quick overview and a graphical representation is included here.</p>
<div class="figure">
<img src="figures/all_nodes_040318.png" alt=""><p class="caption">The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details.</p>
</div>
<p>The GDC API exposes these nodes and edges in a somewhat simplified set of <a href="https://en.wikipedia.org/wiki/Representational_state_transfer">RESTful</a> endpoints.</p>
<div id="quickstart" class="section level3">
<h3 class="hasAnchor">
<a href="#quickstart" class="anchor"></a>Quickstart</h3>
<p>This quickstart section is just meant to show basic functionality. More details of functionality are included further on in this vignette and in function-specific help.</p>
<p>To report bugs or problems, either <a href="https://github.com/Bioconductor/GenomicDataCommons/issues">submit a new issue</a> or submit a <code><a href="https://rdrr.io/r/utils/bug.report.html">bug.report(package='GenomicDataCommons')</a></code> from within R (which will redirect you to the new issue on GitHub).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></span><span class="op">)</span></code></pre></div>
<div id="check-connectivity-and-status" class="section level4">
<h4 class="hasAnchor">
<a href="#check-connectivity-and-status" class="anchor"></a>Check connectivity and status</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> package relies on having network connectivity. In addition, the NCI GDC API must also be operational and not under maintenance. Checking <code>status</code> can be used to check this connectivity and functionality.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/status.html">status</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## $commit
## [1] "dfa394478bd39c11b89d3819a398898d99575a24"
## 
## $data_release
## [1] "Data Release 29.0 - March 31, 2021"
## 
## $status
## [1] "OK"
## 
## $tag
## [1] "3.0.0"
## 
## $version
## [1] 1</code></pre>
</div>
<div id="find-data" class="section level4">
<h4 class="hasAnchor">
<a href="#find-data" class="anchor"></a>Find data</h4>
<p>The following code builds a <code>manifest</code> that can be used to guide the download of raw data. Here, filtering finds gene expression files quantified as raw counts using <code>HTSeq</code> from ovarian cancer patients.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ge_manifest</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span> <span class="op">&amp;</span>
                <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span> <span class="op">&amp;</span>
                <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'HTSeq - Counts'</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="download-data" class="section level4">
<h4 class="hasAnchor">
<a href="#download-data" class="anchor"></a>Download data</h4>
<p>After the 379 gene expression files specified in the query above. Using multiple processes to do the download very significantly speeds up the transfer in many cases. On a standard 1Gb connection, the following completes in about 30 seconds. The first time the data are downloaded, R will ask to create a cache directory (see <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_cache.html">?gdc_cache</a></code> for details of setting and interacting with the cache). Resulting downloaded files will be stored in the cache directory. Future access to the same files will be directly from the cache, alleviating multiple downloads.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fnames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">ge_manifest</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>,<span class="va">gdcdata</span><span class="op">)</span></code></pre></div>
<p>If the download had included controlled-access data, the download above would have needed to include a <code>token</code>. Details are available in <a href="#authentication">the authentication section below</a>.</p>
</div>
<div id="metadata-queries" class="section level4">
<h4 class="hasAnchor">
<a href="#metadata-queries" class="anchor"></a>Metadata queries</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> can access the significant clinical, demographic, biospecimen, and annotation information contained in the NCI GDC.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">expands</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"diagnoses"</span>,<span class="st">"annotations"</span>,
             <span class="st">"demographic"</span>,<span class="st">"exposures"</span><span class="op">)</span>
<span class="va">projResults</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">projResults</span>,list.len<span class="op">=</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 9
##  $ id                    : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-LGG" "TCGA-SARC" ...
##  $ primary_site          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Testis" "Gallbladder" "Unknown" "Other and unspecified parts of biliary tract" ...
##   ..$ TCGA-UCEC : chr [1:2] "Corpus uteri" "Uterus, NOS"
##   ..$ TCGA-LGG  : chr "Brain"
##   ..$ TCGA-SARC : chr [1:13] "Kidney" "Other and unspecified parts of tongue" "Bones, joints and articular cartilage of limbs" "Colon" ...
##   ..$ TCGA-PAAD : chr "Pancreas"
##   .. [list output truncated]
##  $ dbgap_accession_number: logi [1:10] NA NA NA NA NA NA ...
##  $ project_id            : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-LGG" "TCGA-SARC" ...
##  $ disease_type          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Germ Cell Neoplasms" "Granular Cell Tumors and Alveolar Soft Part Sarcomas" "Immunoproliferative Diseases" "Plasma Cell Tumors" ...
##   ..$ TCGA-UCEC : chr [1:4] "Epithelial Neoplasms, NOS" "Cystic, Mucinous and Serous Neoplasms" "Adenomas and Adenocarcinomas" "Not Reported"
##   ..$ TCGA-LGG  : chr "Gliomas"
##   ..$ TCGA-SARC : chr [1:6] "Nerve Sheath Tumors" "Myomatous Neoplasms" "Fibromatous Neoplasms" "Lipomatous Neoplasms" ...
##   ..$ TCGA-PAAD : chr [1:4] "Cystic, Mucinous and Serous Neoplasms" "Ductal and Lobular Neoplasms" "Adenomas and Adenocarcinomas" "Epithelial Neoplasms, NOS"
##   .. [list output truncated]
##   [list output truncated]
##  - attr(*, "row.names")= int [1:10] 1 2 3 4 5 6 7 8 9 10
##  - attr(*, "class")= chr [1:3] "GDCprojectsResults" "GDCResults" "list"</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">projResults</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "id"                     "primary_site"           "dbgap_accession_number"
## [4] "project_id"             "disease_type"           "name"                  
## [7] "releasable"             "state"                  "released"</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># or listviewer::jsonedit(clinResults)</span></code></pre></div>
</div>
</div>
<div id="basic-design" class="section level3">
<h3 class="hasAnchor">
<a href="#basic-design" class="anchor"></a>Basic design</h3>
<p>This package design is meant to have some similarities to the “hadleyverse” approach of dplyr. Roughly, the functionality for finding and accessing files and metadata can be divided into:</p>
<ol style="list-style-type: decimal">
<li>Simple query constructors based on GDC API endpoints.</li>
<li>A set of verbs that when applied, adjust filtering, field selection, and faceting (fields for aggregation) and result in a new query object (an endomorphism)</li>
<li>A set of verbs that take a query and return results from the GDC</li>
</ol>
<p>In addition, there are exhiliary functions for asking the GDC API for information about available and default fields, slicing BAM files, and downloading actual data files. Here is an overview of functionality<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<ul>
<li>Creating a query
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">annotations()</a></code></li>
</ul>
</li>
<li>Manipulating a query
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code></li>
</ul>
</li>
<li>Introspection on the GDC API fields
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/mapping.html">mapping()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/grep_fields.html">grep_fields()</a></code></li>
<li><code>field_picker()</code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_values.html">available_values()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_expand.html">available_expand()</a></code></li>
</ul>
</li>
<li>Executing an API call to retrieve query results
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/response.html">response()</a></code></li>
</ul>
</li>
<li>Raw data file downloads
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_client.html">gdc_client()</a></code></li>
</ul>
</li>
<li>Summarizing and aggregating field values (faceting)
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code></li>
</ul>
</li>
<li>Authentication
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_token.html">gdc_token()</a></code></li>
</ul>
</li>
<li>BAM file slicing
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/slicing.html">slicing()</a></code></li>
</ul>
</li>
</ul>
</div>
<div id="usage" class="section level3">
<h3 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h3>
<p>There are two main classes of operations when working with the NCI GDC.</p>
<ol style="list-style-type: decimal">
<li>
<a href="#querying-metadata">Querying metadata and finding data files</a> (e.g., finding all gene expression quantifications data files for all colon cancer patients).</li>
<li>
<a href="#datafile-access-and-download">Transferring raw or processed data</a> from the GDC to another computer (e.g., downloading raw or processed data)</li>
</ol>
<p>Both classes of operation are reviewed in detail in the following sections.</p>
</div>
</div>
<div id="querying-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#querying-metadata" class="anchor"></a>Querying metadata</h2>
<p>Vast amounts of metadata about cases (patients, basically), files, projects, and so-called annotations are available via the NCI GDC API. Typically, one will want to query metadata to either focus in on a set of files for download or transfer <em>or</em> to perform so-called aggregations (pivot-tables, facets, similar to the R <code><a href="https://rdrr.io/r/base/table.html">table()</a></code> functionality).</p>
<p>Querying metadata starts with <a href="#creating-a-query">creating a “blank” query</a>. One will often then want to <a href="#filtering"><code>filter</code></a> the query to limit results prior to <a href="#retrieving-results">retrieving results</a>. The GenomicDataCommons package has <a href="#fields-and-values">helper functions for listing fields</a> that are available for filtering.</p>
<p>In addition to fetching results, the GDC API allows <a href="#facets-and-aggregation">faceting, or aggregating,</a>, useful for compiling reports, generating dashboards, or building user interfaces to GDC data (see GDC web query interface for a non-R-based example).</p>
<div id="creating-a-query" class="section level4">
<h4 class="hasAnchor">
<a href="#creating-a-query" class="anchor"></a>Creating a query</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> package accesses the <em>same</em> API as the <em>GDC</em> website. Therefore, a useful approach, particularly for beginning users is to examine the filters available on the <a href="https://portal.gdc.cancer.gov/repository">GDC repository pages</a> to find appropriate filtering criteria. From there, converting those checkboxes to a GenomicDataCommons <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">query()</a></code> is relatively straightforward. Note that only a small subset of the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code> are available by default on the website.</p>
<div class="figure">
<img src="figures/gdc_filtering.png" alt=""><p class="caption">A screenshot of an example query of the GDC repository portal.</p>
</div>
<p>A query of the GDC starts its life in R. Queries follow the four metadata endpoints available at the GDC. In particular, there are four convenience functions that each create <code>GDCQuery</code> objects (actually, specific subclasses of <code>GDCQuery</code>):</p>
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">annotations()</a></code></li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pquery</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>The <code>pquery</code> object is now an object of (S3) class, <code>GDCQuery</code> (and <code>gdc_projects</code> and <code>list</code>). The object contains the following elements:</p>
<ul>
<li>fields: This is a character vector of the fields that will be returned when we <a href="#retrieving-results">retrieve data</a>. If no fields are specified to, for example, the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code> function, the default fields from the GDC are used (see <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code>)</li>
<li>filters: This will contain results after calling the <a href="#filtering"><code>filter()</code> method</a> and will be used to filter results on <a href="#retrieving-results">retrieval</a>.</li>
<li>facets: A character vector of field names that will be used for <a href="#facets-and-aggregation">aggregating data</a> in a call to <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code>.</li>
<li>archive: One of either “default” or <a href="https://gdc-portal.nci.nih.gov/legacy-archive/">“legacy”</a>.</li>
<li>token: A character(1) token from the GDC. See <a href="#authentication">the authentication section</a> for details, but note that, in general, the token is not necessary for metadata query and retrieval, only for actual data download.</li>
</ul>
<p>Looking at the actual object (get used to using <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code>!), note that the query contains no results.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 5
##  $ fields : chr [1:10] "dbgap_accession_number" "disease_type" "intended_release_date" "name" ...
##  $ filters: NULL
##  $ facets : NULL
##  $ legacy : logi FALSE
##  $ expand : NULL
##  - attr(*, "class")= chr [1:3] "gdc_projects" "GDCQuery" "list"</code></pre>
</div>
<div id="retrieving-results" class="section level4">
<h4 class="hasAnchor">
<a href="#retrieving-results" class="anchor"></a>Retrieving results</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#size-and-from">[ GDC pagination documentation ]</a></p>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#sort">[ GDC sorting documentation ]</a></p>
<p>With a query object available, the next step is to retrieve results from the GDC. The GenomicDataCommons package. The most basic type of results we can get is a simple <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> of records available that satisfy the filter criteria. Note that we have not set any filters, so a <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> here will represent all the project records publicly available at the GDC in the “default” archive"</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pcount</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span>
<span class="co"># or</span>
<span class="va">pcount</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">pcount</span></code></pre></div>
<pre><code>## [1] 68</code></pre>
<p>The <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> method will fetch actual results.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">presults</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>These results are returned from the GDC in <a href="http://www.json.org/">JSON</a> format and converted into a (potentially nested) list in R. The <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code> method is useful for taking a quick glimpse of the data.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 9
##  $ id                    : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-LGG" "TCGA-SARC" ...
##  $ primary_site          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Testis" "Gallbladder" "Unknown" "Other and unspecified parts of biliary tract" ...
##   ..$ TCGA-UCEC : chr [1:2] "Corpus uteri" "Uterus, NOS"
##   ..$ TCGA-LGG  : chr "Brain"
##   ..$ TCGA-SARC : chr [1:13] "Kidney" "Other and unspecified parts of tongue" "Bones, joints and articular cartilage of limbs" "Colon" ...
##   ..$ TCGA-PAAD : chr "Pancreas"
##   ..$ TCGA-ESCA : chr [1:2] "Esophagus" "Stomach"
##   ..$ TCGA-PRAD : chr "Prostate gland"
##   ..$ GENIE-VICC: chr [1:46] "Testis" "Unknown" "Other and unspecified parts of biliary tract" "Adrenal gland" ...
##   ..$ TCGA-LAML : chr "Hematopoietic and reticuloendothelial systems"
##   ..$ TCGA-KIRC : chr "Kidney"
##  $ dbgap_accession_number: logi [1:10] NA NA NA NA NA NA ...
##  $ project_id            : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-LGG" "TCGA-SARC" ...
##  $ disease_type          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Germ Cell Neoplasms" "Granular Cell Tumors and Alveolar Soft Part Sarcomas" "Immunoproliferative Diseases" "Plasma Cell Tumors" ...
##   ..$ TCGA-UCEC : chr [1:4] "Epithelial Neoplasms, NOS" "Cystic, Mucinous and Serous Neoplasms" "Adenomas and Adenocarcinomas" "Not Reported"
##   ..$ TCGA-LGG  : chr "Gliomas"
##   ..$ TCGA-SARC : chr [1:6] "Nerve Sheath Tumors" "Myomatous Neoplasms" "Fibromatous Neoplasms" "Lipomatous Neoplasms" ...
##   ..$ TCGA-PAAD : chr [1:4] "Cystic, Mucinous and Serous Neoplasms" "Ductal and Lobular Neoplasms" "Adenomas and Adenocarcinomas" "Epithelial Neoplasms, NOS"
##   ..$ TCGA-ESCA : chr [1:3] "Cystic, Mucinous and Serous Neoplasms" "Squamous Cell Neoplasms" "Adenomas and Adenocarcinomas"
##   ..$ TCGA-PRAD : chr [1:3] "Cystic, Mucinous and Serous Neoplasms" "Ductal and Lobular Neoplasms" "Adenomas and Adenocarcinomas"
##   ..$ GENIE-VICC: chr [1:41] "Germ Cell Neoplasms" "Acinar Cell Neoplasms" "Synovial-like Neoplasms" "Plasma Cell Tumors" ...
##   ..$ TCGA-LAML : chr "Myeloid Leukemias"
##   ..$ TCGA-KIRC : chr "Adenomas and Adenocarcinomas"
##  $ name                  : chr [1:10] "AACR Project GENIE - Contributed by Memorial Sloan Kettering Cancer Center" "Uterine Corpus Endometrial Carcinoma" "Brain Lower Grade Glioma" "Sarcoma" ...
##  $ releasable            : logi [1:10] FALSE TRUE TRUE TRUE TRUE TRUE ...
##  $ state                 : chr [1:10] "open" "open" "open" "open" ...
##  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...
##  - attr(*, "row.names")= int [1:10] 1 2 3 4 5 6 7 8 9 10
##  - attr(*, "class")= chr [1:3] "GDCprojectsResults" "GDCResults" "list"</code></pre>
<p>A default of only 10 records are returned. We can use the <code>size</code> and <code>from</code> arguments to <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> to either page through results or to change the number of results. Finally, there is a convenience method, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code> that will simply fetch all the available results given a query. Note that <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code> may take a long time and return HUGE result sets if not used carefully. Use of a combination of <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> and <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> to get a sense of the expected data size is probably warranted before calling <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">presults</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 68</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># includes all records</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Extracting subsets of results or manipulating the results into a more conventional R data structure is not easily generalizable. However, the <a href="https://github.com/hadley/purrr">purrr</a>, <a href="https://renkun.me/rlist/">rlist</a>, and <a href="https://cran.r-project.org/web/packages/data.tree/vignettes/data.tree.html">data.tree</a> packages are all potentially of interest for manipulating complex, nested list structures. For viewing the results in an interactive viewer, consider the <a href="https://github.com/timelyportfolio/listviewer">listviewer</a> package.</p>
</div>
<div id="fields-and-values" class="section level4">
<h4 class="hasAnchor">
<a href="#fields-and-values" class="anchor"></a>Fields and Values</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#fields">[ GDC <code>fields</code> documentation ]</a></p>
<p>Central to querying and retrieving data from the GDC is the ability to specify which fields to return, filtering by fields and values, and faceting or aggregating. The GenomicDataCommons package includes two simple functions, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code> and <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code>. Each can operate on a character(1) endpoint name (“cases,” “files,” “annotations,” or “projects”) or a <code>GDCQuery</code> object.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "access"                         "acl"                           
##  [3] "average_base_quality"           "average_insert_size"           
##  [5] "average_read_length"            "channel"                       
##  [7] "chip_id"                        "chip_position"                 
##  [9] "contamination"                  "contamination_error"           
## [11] "created_datetime"               "data_category"                 
## [13] "data_format"                    "data_type"                     
## [15] "error_type"                     "experimental_strategy"         
## [17] "file_autocomplete"              "file_id"                       
## [19] "file_name"                      "file_size"                     
## [21] "imaging_date"                   "magnification"                 
## [23] "md5sum"                         "mean_coverage"                 
## [25] "msi_score"                      "msi_status"                    
## [27] "pairs_on_diff_chr"              "plate_name"                    
## [29] "plate_well"                     "platform"                      
## [31] "proportion_base_mismatch"       "proportion_coverage_10x"       
## [33] "proportion_coverage_10X"        "proportion_coverage_30x"       
## [35] "proportion_coverage_30X"        "proportion_reads_duplicated"   
## [37] "proportion_reads_mapped"        "proportion_targets_no_coverage"
## [39] "read_pair_number"               "revision"                      
## [41] "stain_type"                     "state"                         
## [43] "state_comment"                  "submitter_id"                  
## [45] "tags"                           "total_reads"                   
## [47] "tumor_ploidy"                   "tumor_purity"                  
## [49] "type"                           "updated_datetime"</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The number of fields available for files endpoint</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 943</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The first few fields available for files endpoint</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "access"                      "acl"                        
## [3] "analysis.analysis_id"        "analysis.analysis_type"     
## [5] "analysis.created_datetime"   "analysis.input_files.access"</code></pre>
<p>The fields to be returned by a query can be specified following a similar paradigm to that of the dplyr package. The <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> function is a verb that resets the fields slot of a <code>GDCQuery</code>; note that this is not quite analogous to the dplyr <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> verb that limits from already-present fields. We <em>completely replace</em> the fields when using <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> on a <code>GDCQuery</code>.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default fields here</span>
<span class="va">qcases</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">qcases</span><span class="op">$</span><span class="va">fields</span></code></pre></div>
<pre><code>##  [1] "aliquot_ids"              "analyte_ids"             
##  [3] "case_autocomplete"        "case_id"                 
##  [5] "consent_type"             "created_datetime"        
##  [7] "days_to_consent"          "days_to_lost_to_followup"
##  [9] "diagnosis_ids"            "disease_type"            
## [11] "index_date"               "lost_to_followup"        
## [13] "portion_ids"              "primary_site"            
## [15] "sample_ids"               "slide_ids"               
## [17] "state"                    "submitter_aliquot_ids"   
## [19] "submitter_analyte_ids"    "submitter_diagnosis_ids" 
## [21] "submitter_id"             "submitter_portion_ids"   
## [23] "submitter_sample_ids"     "submitter_slide_ids"     
## [25] "updated_datetime"</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set up query to use ALL available fields</span>
<span class="co"># Note that checking of fields is done by select()</span>
<span class="va">qcases</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'cases'</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">qcases</span><span class="op">$</span><span class="va">fields</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "case_id"                       "aliquot_ids"                  
## [3] "analyte_ids"                   "annotations.annotation_id"    
## [5] "annotations.case_id"           "annotations.case_submitter_id"</code></pre>
<p>Finding fields of interest is such a common operation that the GenomicDataCommons includes the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/grep_fields.html">grep_fields()</a></code> function and the <code>field_picker()</code> widget. See the appropriate help pages for details.</p>
</div>
<div id="facets-and-aggregation" class="section level4">
<h4 class="hasAnchor">
<a href="#facets-and-aggregation" class="anchor"></a>Facets and aggregation</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC <code>facet</code> documentation ]</a></p>
<p>The GDC API offers a feature known as aggregation or faceting. By specifying one or more fields (of appropriate type), the GDC can return to us a count of the number of records matching each potential value. This is similar to the R <code>table</code> method. Multiple fields can be returned at once, but the GDC API does not have a cross-tabulation feature; all aggregations are only on one field at a time. Results of <code>aggregation()</code> calls come back as a list of data.frames (actually, tibbles).</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># total number of files of a specific type</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'type'</span>,<span class="st">'data_type'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">res</span><span class="op">$</span><span class="va">type</span></code></pre></div>
<pre><code>##    doc_count                           key
## 1     151077    annotated_somatic_mutation
## 2      88599       simple_somatic_mutation
## 3      82177                 aligned_reads
## 4      61693               gene_expression
## 5      58116           copy_number_segment
## 6      45419          copy_number_estimate
## 7      30072                   slide_image
## 8      29990              mirna_expression
## 9      25591        biospecimen_supplement
## 10     12931           clinical_supplement
## 11     12359        methylation_beta_value
## 12     11444          structural_variation
## 13      4359   aggregated_somatic_mutation
## 14      4317       masked_somatic_mutation
## 15        54 secondary_expression_analysis</code></pre>
<p>Using <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code> is an also easy way to learn the contents of individual fields and forms the basis for faceted search pages.</p>
</div>
<div id="filtering" class="section level4">
<h4 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query">[ GDC <code>filtering</code> documentation ]</a></p>
<p>The GenomicDataCommons package uses a form of non-standard evaluation to specify R-like queries that are then translated into an R list. That R list is, upon calling a method that fetches results from the GDC API, translated into the appropriate JSON string. The R expression uses the formula interface as suggested by Hadley Wickham in his <a href="https://cran.r-project.org/web/packages/dplyr/vignettes/nse.html">vignette on non-standard evaluation</a></p>
<blockquote>
<p>It’s best to use a formula because a formula captures both the expression to evaluate and the environment where the evaluation occurs. This is important if the expression is a mixture of variables in a data frame and objects in the local environment [for example].</p>
</blockquote>
<p>For the user, these details will not be too important except to note that a filter expression must begin with a “~.”</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">qfiles</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># all files</span></code></pre></div>
<pre><code>## [1] 618198</code></pre>
<p>To limit the file type, we can refer back to the <a href="#facets-and-aggregation">section on faceting</a> to see the possible values for the file field “type.” For example, to filter file results to only “gene_expression” files, we simply specify a filter.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span><span class="op">)</span>
<span class="co"># here is what the filter looks like after translation</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">get_filter</a></span><span class="op">(</span><span class="va">qfiles</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 2
##  $ op     : 'scalar' chr "="
##  $ content:List of 2
##   ..$ field: chr "type"
##   ..$ value: chr "gene_expression"</code></pre>
<p>What if we want to create a filter based on the project (‘TCGA-OVCA,’ for example)? Well, we have a couple of possible ways to discover available fields. The first is based on base R functionality and some intuition.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">'pro'</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span>, value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "analysis.input_files.proportion_base_mismatch"                  
##  [2] "analysis.input_files.proportion_coverage_10x"                   
##  [3] "analysis.input_files.proportion_coverage_10X"                   
##  [4] "analysis.input_files.proportion_coverage_30x"                   
##  [5] "analysis.input_files.proportion_coverage_30X"                   
##  [6] "analysis.input_files.proportion_reads_duplicated"               
##  [7] "analysis.input_files.proportion_reads_mapped"                   
##  [8] "analysis.input_files.proportion_targets_no_coverage"            
##  [9] "cases.diagnoses.international_prognostic_index"                 
## [10] "cases.diagnoses.progression_or_recurrence"                      
## [11] "cases.follow_ups.days_to_progression"                           
## [12] "cases.follow_ups.days_to_progression_free"                      
## [13] "cases.follow_ups.procedures_performed"                          
## [14] "cases.follow_ups.progression_or_recurrence"                     
## [15] "cases.follow_ups.progression_or_recurrence_anatomic_site"       
## [16] "cases.follow_ups.progression_or_recurrence_type"                
## [17] "cases.project.dbgap_accession_number"                           
## [18] "cases.project.disease_type"                                     
## [19] "cases.project.intended_release_date"                            
## [20] "cases.project.name"                                             
## [21] "cases.project.primary_site"                                     
## [22] "cases.project.program.dbgap_accession_number"                   
## [23] "cases.project.program.name"                                     
## [24] "cases.project.program.program_id"                               
## [25] "cases.project.project_id"                                       
## [26] "cases.project.releasable"                                       
## [27] "cases.project.released"                                         
## [28] "cases.project.state"                                            
## [29] "cases.samples.days_to_sample_procurement"                       
## [30] "cases.samples.method_of_sample_procurement"                     
## [31] "cases.samples.portions.slides.number_proliferating_cells"       
## [32] "cases.samples.portions.slides.prostatic_chips_positive_count"   
## [33] "cases.samples.portions.slides.prostatic_chips_total_count"      
## [34] "cases.samples.portions.slides.prostatic_involvement_percent"    
## [35] "cases.tissue_source_site.project"                               
## [36] "downstream_analyses.output_files.proportion_base_mismatch"      
## [37] "downstream_analyses.output_files.proportion_coverage_10x"       
## [38] "downstream_analyses.output_files.proportion_coverage_10X"       
## [39] "downstream_analyses.output_files.proportion_coverage_30x"       
## [40] "downstream_analyses.output_files.proportion_coverage_30X"       
## [41] "downstream_analyses.output_files.proportion_reads_duplicated"   
## [42] "downstream_analyses.output_files.proportion_reads_mapped"       
## [43] "downstream_analyses.output_files.proportion_targets_no_coverage"
## [44] "index_files.proportion_base_mismatch"                           
## [45] "index_files.proportion_coverage_10x"                            
## [46] "index_files.proportion_coverage_10X"                            
## [47] "index_files.proportion_coverage_30x"                            
## [48] "index_files.proportion_coverage_30X"                            
## [49] "index_files.proportion_reads_duplicated"                        
## [50] "index_files.proportion_reads_mapped"                            
## [51] "index_files.proportion_targets_no_coverage"                     
## [52] "proportion_base_mismatch"                                       
## [53] "proportion_coverage_10x"                                        
## [54] "proportion_coverage_10X"                                        
## [55] "proportion_coverage_30x"                                        
## [56] "proportion_coverage_30X"                                        
## [57] "proportion_reads_duplicated"                                    
## [58] "proportion_reads_mapped"                                        
## [59] "proportion_targets_no_coverage"</code></pre>
<p>Interestingly, the project information is “nested” inside the case. We don’t need to know that detail other than to know that we now have a few potential guesses for where our information might be in the files records. We need to know where because we need to construct the appropriate filter.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet</a></span><span class="op">(</span><span class="st">'cases.project.project_id'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## $cases.project.project_id
##    doc_count                   key
## 1      36134                 FM-AD
## 2      33766             TCGA-BRCA
## 3      42178               CPTAC-3
## 4      36470             GENIE-MSK
## 5      18358             TCGA-LUAD
## 6      17277             TCGA-UCEC
## 7      16340             TCGA-HNSC
## 8      16344               TCGA-OV
## 9      15445             TCGA-THCA
## 10     29433         MMRF-COMMPASS
## 11     16368             TCGA-LUSC
## 12     15795              TCGA-LGG
## 13     28464            GENIE-DFCI
## 14     16255             TCGA-KIRC
## 15     15296             TCGA-PRAD
## 16     15338             TCGA-COAD
## 17     13089              TCGA-GBM
## 18     20772         TARGET-ALL-P2
## 19     13674             TCGA-SKCM
## 20     13739             TCGA-STAD
## 21     12513             TCGA-BLCA
## 22     11578             TCGA-LIHC
## 23      9201             TCGA-CESC
## 24      9137             TCGA-KIRP
## 25      8002             TCGA-SARC
## 26      7772            TARGET-AML
## 27      5671             TCGA-PAAD
## 28      5657             TCGA-ESCA
## 29      5378             TCGA-PCPG
## 30      5269             TCGA-READ
## 31      5796            TARGET-NBL
## 32      8981     BEATAML1.0-COHORT
## 33      8958               CPTAC-2
## 34      4605             TCGA-TGCT
## 35      4814             TCGA-LAML
## 36      3691             TCGA-THYM
## 37      6036             HCMI-CMDC
## 38      5941               CMI-MBC
## 39      5550         CGCI-HTMCP-CC
## 40      2736              TCGA-ACC
## 41      2457             TCGA-KICH
## 42      2677             TARGET-WT
## 43      4805          NCICCR-DLBCL
## 44      2518             TCGA-MESO
## 45      2340              TCGA-UVM
## 46      3113             TARGET-OS
## 47      3982         TARGET-ALL-P3
## 48      3857             GENIE-MDA
## 49      3833            GENIE-VICC
## 50      3320             GENIE-JHU
## 51      1765              TCGA-UCS
## 52      1426             TCGA-CHOL
## 53      2632             GENIE-UHN
## 54      1325             TCGA-DLBC
## 55      2477            CGCI-BLGSP
## 56      1049             TARGET-RT
## 57      1038            GENIE-GRCC
## 58       994            WCDT-MCRPC
## 59       934               CMI-ASC
## 60       801             GENIE-NKI
## 61       798              OHSU-CNL
## 62       703   ORGANOID-PANCREATIC
## 63       570               CMI-MPC
## 64       417           CTSP-DLBCL1
## 65       223 BEATAML1.0-CRENOLANIB
## 66       169           TARGET-CCSK
## 67       133         TARGET-ALL-P1
## 68        21        VAREPOP-APOLLO</code></pre>
<p>We note that <code>cases.project.project_id</code> looks like it is a good fit. We also note that <code>TCGA-OV</code> is the correct project_id, not <code>TCGA-OVCA</code>. Note that <em>unlike with dplyr and friends, the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter()</a></code> method here <strong>replaces</strong> the filter and does not build on any previous filters</em>.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span> <span class="op">&amp;</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">get_filter</a></span><span class="op">(</span><span class="va">qfiles</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 2
##  $ op     : 'scalar' chr "and"
##  $ content:List of 2
##   ..$ :List of 2
##   .. ..$ op     : 'scalar' chr "="
##   .. ..$ content:List of 2
##   .. .. ..$ field: chr "cases.project.project_id"
##   .. .. ..$ value: chr "TCGA-OV"
##   ..$ :List of 2
##   .. ..$ op     : 'scalar' chr "="
##   .. ..$ content:List of 2
##   .. .. ..$ field: chr "type"
##   .. .. ..$ value: chr "gene_expression"</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qfiles</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1137</code></pre>
<p>Asking for a <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> of results given these new filter criteria gives <code>r qfiles %&gt;% count()</code> results. Generating a manifest for bulk downloads is as simple as asking for the manifest from the current query.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">manifest_df</span> <span class="op">=</span> <span class="va">qfiles</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 5
##   id                  filename                    md5                size state 
##   &lt;chr&gt;               &lt;chr&gt;                       &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt; 
## 1 4b25720c-2fad-4055… 51b8dd7e-1686-46bb-9a29-b0… 19af96c6c5d4e11… 538565 relea…
## 2 5e8d1839-40ab-432b… 9f223240-1020-4450-a6ec-a6… 968c18169c965ee… 568707 relea…
## 3 691ec8c8-17b3-4c26… 0315439e-694c-4b14-a27c-29… 2cd3130425a20de… 252344 relea…
## 4 18863638-3369-4ca3… 4d337bb6-8e53-4142-8c22-43… bda0aeee9382d72… 249487 relea…
## 5 81c89a1f-19d6-43df… ce511378-d8f8-494e-a07d-2e… 6d82eb3ca6b62b2… 559993 relea…
## 6 fe36a6b6-6401-4fc5… 968544ba-7990-41e8-a0ee-52… b382541b102c7e1… 526190 relea…</code></pre>
<p>Note that we might still not be quite there. Looking at filenames, there are suspiciously named files that might include “FPKM,” “FPKM-UQ,” or “counts.” Another round of <code>grep</code> and <code>available_fields</code>, looking for “type” turned up that the field “analysis.workflow_type” has the appropriate filter criteria.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span> <span class="op">&amp;</span>
                            <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span> <span class="op">&amp;</span>
                            <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'HTSeq - Counts'</span><span class="op">)</span>
<span class="va">manifest_df</span> <span class="op">=</span> <span class="va">qfiles</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 379</code></pre>
<p>The GDC Data Transfer Tool can be used (from R, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer()</a></code> or from the command-line) to orchestrate high-performance, restartable transfers of all the files in the manifest. See <a href="bulk-downloads">the bulk downloads section</a> for details.</p>
</div>
<div id="authentication" class="section level3">
<h3 class="hasAnchor">
<a href="#authentication" class="anchor"></a>Authentication</h3>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC authentication documentation ]</a></p>
<p>The GDC offers both “controlled-access” and “open” data. As of this writing, only data stored as files is “controlled-access”; that is, metadata accessible via the GDC is all “open” data and some files are “open” and some are “controlled-access.” Controlled-access data are only available after <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data">going through the process of obtaining access.</a></p>
<p>After controlled-access to one or more datasets has been granted, logging into the GDC web portal will allow you to <a href="https://docs.gdc.cancer.gov/Data_Portal/Users_Guide/Authentication/#gdc-authentication-tokens">access a GDC authentication token</a>, which can be downloaded and then used to access available controlled-access data via the GenomicDataCommons package.</p>
<p>The GenomicDataCommons uses authentication tokens only for downloading data (see <code>transfer</code> and <code>gdcdata</code> documentation). The package includes a helper function, <code>gdc_token</code>, that looks for the token to be stored in one of three ways (resolved in this order):</p>
<ol style="list-style-type: decimal">
<li>As a string stored in the environment variable, <code>GDC_TOKEN</code>
</li>
<li>As a file, stored in the file named by the environment variable, <code>GDC_TOKEN_FILE</code>
</li>
<li>In a file in the user home directory, called <code>.gdc_token</code>
</li>
</ol>
<p>As a concrete example:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">token</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_token.html">gdc_token</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer</a></span><span class="op">(</span><span class="va">...</span>,token<span class="op">=</span><span class="va">token</span><span class="op">)</span>
<span class="co"># or</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer</a></span><span class="op">(</span><span class="va">...</span>,token<span class="op">=</span><span class="fu">get_token</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="datafile-access-and-download" class="section level3">
<h3 class="hasAnchor">
<a href="#datafile-access-and-download" class="anchor"></a>Datafile access and download</h3>
<p>The <code>gdcdata</code> function takes a character vector of one or more file ids. A simple way of producing such a vector is to produce a <code>manifest</code> data frame and then pass in the first column, which will contain file ids.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fnames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>,progress<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Note that for controlled-access data, a GDC <a href="#authentication">authentication token</a> is required. Using the <code>BiocParallel</code> package may be useful for downloading in parallel, particularly for large numbers of smallish files.</p>
<p>The bulk download functionality is only efficient (as of v1.2.0 of the GDC Data Transfer Tool) for relatively large files, so use this approach only when transferring BAM files or larger VCF files, for example. Otherwise, consider using the approach shown above, perhaps in parallel.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fnames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, access_method <span class="op">=</span> <span class="st">'client'</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="benjamin-haibe-kains" class="section level1">
<h1 class="hasAnchor">
<a href="#benjamin-haibe-kains" class="anchor"></a>Benjamin Haibe-Kains</h1>
<div id="pharmacogenomics" class="section level2">
<h2 class="hasAnchor">
<a href="#pharmacogenomics" class="anchor"></a>Pharmacogenomics</h2>
<p>Pharmacogenomics holds great promise for the development of biomarkers of drug response and the design of new therapeutic options, which are key challenges in precision medicine. However, such data are scattered and lack standards for efficient access and analysis, consequently preventing the realization of the full potential of pharmacogenomics. To address these issues, we implemented <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em>, an easy-to-use, open source package for integrative analysis of multiple pharmacogenomic datasets. <code>PharmacoGx</code> provides a unified framework for downloading and analyzing large pharmacogenomic datasets which are extensively curated to ensure maximum overlap and consistency.</p>
<p>Examples of <code>PharmacoGx</code> usage in biomedical research can be found in the following publications:</p>
<ul>
<li>
<a href="https://academic.oup.com/bioinformatics/article/32/8/1244/1744214/PharmacoGx-an-R-package-for-analysis-of-large">Smirnov et al. PharmacoGx: an R package for analysis of large pharmacogenomic datasets." Bioinformatics (2015): 1244-1246</a>.</li>
<li><a href="https://doi.org/10.1093/jamia/ocx062">Yao et al., Tissue specificity of in vitro drug sensitivity, JAMIA (2017)</a></li>
<li><a href="https://www.nature.com/articles/s41467-017-01153-8">Safikhani et al., Gene isoforms as expression-based biomarkers predictive of drug response in vitro, Nature Communications (2017)</a></li>
<li><a href="http://cancerres.aacrjournals.org/content/77/11/3057.long">El-Hachem et al., Integrative Cancer Pharmacogenomics to Infer Large-Scale Drug Taxonomy, Cancer Research (2017)</a></li>
<li><a href="https://cancerres.aacrjournals.org/content/78/17/5124">Singh, M. et al. Therapeutic Targeting of the Premetastatic Stage in Human Lung-to-Brain Metastasis. Cancer Research (2018).</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1006743">Knowles, D. A., Bouchard, G. &amp; Plevritis, S. Sparse discriminative latent characteristics for predicting cancer drug sensitivity from genomic features. PLOS Computational Biology (2019)</a></li>
<li><a href="doi:10.1038/s43018-019-0018-6">Corsello, S. M. et al. Discovering the anticancer potential of non-oncology drugs by systematic viability profiling, Nat Cancer (2020)</a></li>
</ul>
<div id="getting-started" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h3>
<p>Let us first load the <code>PharmacoGx</code> library.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PharmacoGx</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: no DISPLAY variable so Tk is not available</code></pre>
<p>We can now access large-scale preclinical pharmacogenomic datasets that have been fully curated for ease of use.</p>
</div>
<div id="overview-of-pharmacogx-datasets-pharmacosets" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-of-pharmacogx-datasets-pharmacosets" class="anchor"></a>Overview of PharmacoGx datasets (PharmacoSets)</h3>
<p>To efficiently store and analyze large pharmacogenomic datasets, we developed the <code>PharmacoSet</code> class (also referred to as <code>PSet</code>), which acts as a data container storing pharmacological and molecular data along with experimental metadata (detailed structure provided in Supplementary materials). This class enables efficient implementation of curated annotations for cell lines, drug compounds and molecular features, which facilitates comparisons between different datasets stored as <code>PharmacoSet</code> objects.</p>
<p>We have made the PharmacoSet objects of the curated datasets available for download using functions provided in the package. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function. Any of the PharmacoSets in the table can then be downloaded by calling <em>downloadPSet</em>, which saves the datasets into a directory of the users choice, and returns the data into the R session.</p>
<div class="figure">
<img src="figures/PharmacoSet_structure.png" alt=""><p class="caption">Overview of PharmacoGx</p>
</div>
<p>To get a list of all the available <code>PharmacoSets</code> in PharmacoGx, we can use the availablePSets` function, which returns a table providing key information for each dataset.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">psets</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html">availablePSets</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">7</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##   Dataset Name             Date Created         PSet Name      version
## 1         CCLE 2020-06-24T14:39:26.588Z         CCLE_2015         2015
## 2       CTRPv2 2020-06-24T14:39:26.588Z       CTRPv2_2015         2015
## 3         FIMM 2020-06-24T14:39:26.588Z         FIMM_2016         2016
## 4         gCSI 2020-06-24T14:39:26.588Z         gCSI_2017         2017
## 5         GDSC 2020-06-24T14:39:26.588Z GDSC_2020(v2-8.2) 2020(v2-8.2)
## 6         GDSC 2020-06-24T14:39:26.588Z GDSC_2020(v1-8.2) 2020(v1-8.2)
## 7         GRAY 2021-02-23T14:39:26.588Z         GRAY_2017         2017
## 8    UHNBreast 2020-06-24T14:39:26.588Z    UHNBreast_2019         2019
##          type                    DOI
## 1 sensitivity 10.5281/zenodo.3905462
## 2 sensitivity 10.5281/zenodo.3905470
## 3 sensitivity 10.5281/zenodo.3905448
## 4 sensitivity 10.5281/zenodo.3905452
## 5 sensitivity 10.5281/zenodo.3905481
## 6 sensitivity 10.5281/zenodo.3905485
## 7 sensitivity 10.5281/zenodo.4557735
## 8        both 10.5281/zenodo.3905460
##                                                           Download
## 1      https://zenodo.org/record/3905462/files/CCLE.rds?download=1
## 2    https://zenodo.org/record/3905470/files/CTRPv2.rds?download=1
## 3      https://zenodo.org/record/3905448/files/FIMM.rds?download=1
## 4      https://zenodo.org/record/3905452/files/gCSI.rds?download=1
## 5     https://zenodo.org/record/3905481/files/GDSC2.rds?download=1
## 6     https://zenodo.org/record/3905485/files/GDSC1.rds?download=1
## 7  https://zenodo.org/record/4557735/files/GRAY2017.rds?download=1
## 8 https://zenodo.org/record/3905460/files/UHNBreast.rds?download=1</code></pre>
<p>There are currently 8 datasets available in PharmacoGx, including sensitivity datasets and perturbation datasets (see below).</p>
<div id="drug-sensitivity-datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#drug-sensitivity-datasets" class="anchor"></a>Drug Sensitivity Datasets</h4>
<p>Drug sensitivity datasets refer to pharmacogenomic data where cancer cells are molecularly profiled at baseline (before drug treatment), and the effect of drug treatment on cell viability is measured using a pharmacological assay (e.g., Cell Titer-Glo). These datasets can be used for biomarker discovery by correlating the molecular features of cancer cells to their response to drugs of interest.</p>
<div class="figure">
<img src="figures/Drug_sensitivity.png" alt=""><p class="caption">Schematic view of the drug sensitivity datasets.</p>
</div>
<p>Notably, the Genomics of Drug Sensitivity in Cancer <a href="https://www.cancerrxgene.org/">GDSC</a> and the Cancer Cell Line Encyclopedia <a href="https://portals.broadinstitute.org/ccle">CCLE</a> are large drug sensitivity datasets published in seminal studies in Nature, <a href="https://www.nature.com/articles/nature11005">Garnett et al., https://www.nature.com/articles/nature11005, Nature (2012)</a> and <a href="https://www.nature.com/articles/nature11003">Barretina et al., The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity, Nature (2012)</a>, respectively.</p>
</div>
<div id="drug-perturbation-datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#drug-perturbation-datasets" class="anchor"></a>Drug Perturbation Datasets</h4>
<p>Drug perturbation datasets refer to pharmacogenomic data where gene expression profiles are measured before and after short-term or medium term (e.g., 6h, 24h) drug treatment to identify genes that are up- and down-regulated due to the drug treatment. These datasets can be to classify drug (drug taxonomy), infer their mechanism of action, or find drugs with similar effects (drug repurposing).</p>
<div class="figure">
<img src="figures/Drug_perturbation.png" alt=""><p class="caption">Schematic view of drug perturbation datasets</p>
</div>
<p>Large drug perturbation data have been generated within the Connectivity Map Project <a href="https://clue.io/cmap">CAMP</a>, with CMAPv2 and CMAPv3 available from <code>PharmacoGx</code>, published in <a href="http://science.sciencemag.org/content/313/5795/1929">Lamb et al., The Connectivity Map: Using Gene-Expression Signatures to Connect Small Molecules, Genes, and Disease, Science (2006)</a> and <a href="https://www.cell.com/cell/abstract/S0092-8674(17)31309-0">Subramanian et al., A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles, Cell (2017)</a>, respectively.</p>
</div>
</div>
<div id="exploring-other-treatment-types" class="section level3">
<h3 class="hasAnchor">
<a href="#exploring-other-treatment-types" class="anchor"></a>Exploring Other Treatment Types</h3>
<p>In addition to PharmacoGx, there is a suite of packages in Bioconductor for exploring public high throughput screening data. For Sensitivity datasets, <em><a href="https://bioconductor.org/packages/3.12/Xeva">Xeva</a></em> provides access to public drug screening datasets in Patient Derived Xenograft models, including providing access to the Novartis PDX Encyclopedia dataset, published in <a href="https://www.nature.com/articles/nm.3954">Gao, H. et al. High-throughput screening using patient-derived tumor xenografts to predict clinical trial drug response, Nature Medicine.</a>. Additionally, <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> is currently available in the development branch of Bioconductor, providing access to cell line screening data for response to radiation treatment, featuring data from <a href="https://www.nature.com/articles/ncomms11428">Yard, B. D. et al. A genetic basis for the variation in the vulnerability of cancer to DNA damage. Nature Communications 7, 11428 (2016).</a>.</p>
<p>The <a href="http://bhklab.ca/bioc2020workshop/articles/GxWorkshop.html">Biomarker Discovery from High Throughput Screening Datasets</a> workshop at Bioc2020 goes into depth about using these packages for accessing and exploring public sensitivity datasets.</p>
<p>In addition, Bioconductor also includes the <em><a href="https://bioconductor.org/packages/3.12/ToxicoGx">ToxicoGx</a></em> package, which provides access to Sensitivity and Perturbation datasets characterizing <em>in vitro</em> response of human tissue to toxicant exposure, currently providing access to data from the <a href="https://pubmed.ncbi.nlm.nih.gov/25313160/">TGGates</a> and <a href="https://ntp.niehs.nih.gov/data/drugmatrix/">DrugMatrix</a> datasets. More information about the ToxicoGx package can be found at the <a href="https://doi.org/10.7490/f1000research.1118037.1">Bioc2020 poster presentation on the package</a>.</p>
</div>
</div>
</div>
<div id="levi-waldron" class="section level1">
<h1 class="hasAnchor">
<a href="#levi-waldron" class="anchor"></a>Levi Waldron</h1>
<div id="accessing-the-cancer-genome-atlas-tcga" class="section level2">
<h2 class="hasAnchor">
<a href="#accessing-the-cancer-genome-atlas-tcga" class="anchor"></a>Accessing The Cancer Genome Atlas (TCGA)</h2>
<p>We summarize two approaches to accessing TCGA data:</p>
<ol style="list-style-type: decimal">
<li>
<em><a href="https://bioconductor.org/packages/3.12/TCGAbiolinks">TCGAbiolinks</a></em>:
<ol style="list-style-type: lower-alpha">
<li>data access through <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em>
</li>
<li>provides data both from the legacy Firehose pipeline used by the TCGA publications (alignments based on hg18 and hg19 builds<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>), and the GDC harmonized GRCh38 pipeline<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</li>
<li>downloads files from the Genomic Data Commons, and provides conversion to <code>(Ranged)SummarizedExperiment</code> where possible</li>
</ol>
</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedTCGAData">curatedTCGAData</a></em>:
<ol style="list-style-type: lower-alpha">
<li>data access through <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>
</li>
<li>provides data from the legacy Firehose pipeline<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>
</li>
<li>provides individual assays as <code>(Ranged)SummarizedExperiment</code> and <code>RaggedExperiment</code>, integrates multiple assays within and across cancer types using <em><a href="https://bioconductor.org/packages/3.12/MultiAssayExperiment">MultiAssayExperiment</a></em>
</li>
</ol>
</li>
</ol>
<div id="tcgabiolinks" class="section level3">
<h3 class="hasAnchor">
<a href="#tcgabiolinks" class="anchor"></a>TCGAbiolinks</h3>
<p>We demonstrate here generating a <code>RangedSummarizedExperiment</code> for RNA-seq data from adrenocortical carcinoma (ACC). For additional information and options, see the TCGAbiolinks vignettes<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<p>Load packages:</p>
<p>Search for matching data:</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks">TCGAbiolinks</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCquery.html">GDCquery</a></span><span class="op">(</span>project <span class="op">=</span> <span class="st">"TCGA-ACC"</span>,
                           data.category <span class="op">=</span> <span class="st">"Gene expression"</span>,
                           data.type <span class="op">=</span> <span class="st">"Gene expression quantification"</span>,
                           platform <span class="op">=</span> <span class="st">"Illumina HiSeq"</span>, 
                           file.type  <span class="op">=</span> <span class="st">"normalized_results"</span>,
                           experimental.strategy <span class="op">=</span> <span class="st">"RNA-Seq"</span>,
                           legacy <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Download data and convert it to <code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#gdcdir &lt;- file.path("Waldron_PublicData", "GDCdata")</span>
<span class="co">#GDCdownload(query, method = "api", files.per.chunk = 10,</span>
<span class="co">#            directory = gdcdir)</span>
<span class="co">#ACCse &lt;- GDCprepare(query, directory = gdcdir)</span>
<span class="co">#ACCse</span></code></pre></div>
</div>
<div id="curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects" class="anchor"></a>curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</h3>
<p><em><a href="https://bioconductor.org/packages/3.12/curatedTCGAData">curatedTCGAData</a></em> does not interface with the Genomic Data Commons, but downloads data from Bioconductor’s <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>.</p>
<p>References: * <a href="https://waldronlab.io/MultiAssayWorkshop/articles/curatedTCGAData_ref.html" class="uri">https://waldronlab.io/MultiAssayWorkshop/articles/curatedTCGAData_ref.html</a> : available datasets and data types. * <a href="https://waldronlab.io/MultiAssayWorkshop/articles/TCGAutilsCheatsheet.html" class="uri">https://waldronlab.io/MultiAssayWorkshop/articles/TCGAutilsCheatsheet.html</a> : quick-reference for related utilities in TCGAutils package * <a href="https://waldronlab.io/MultiAssayWorkshop/articles/Ramos_MultiAssayExperiment.html" class="uri">https://waldronlab.io/MultiAssayWorkshop/articles/Ramos_MultiAssayExperiment.html</a>: workshop for MultiAssayExperiment</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/">MultiAssayExperiment</a></span><span class="op">)</span></code></pre></div>
<p>By default, the <code><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData()</a></code> function will only show available datasets, and not download anything. The arguments are shown here only for demonstration, the same result is obtained with no arguments:</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span><span class="op">(</span>diseaseCode <span class="op">=</span> <span class="st">"*"</span>, assays <span class="op">=</span> <span class="st">"*"</span><span class="op">)</span></code></pre></div>
<pre><code>## using temporary cache /tmp/Rtmpea0AIg/BiocFileCache</code></pre>
<pre><code>## snapshotDate(): 2020-10-27</code></pre>
<pre><code>## See '?curatedTCGAData' for 'diseaseCode' and 'assays' inputs</code></pre>
<pre><code>##      ah_id                                      title file_size
## 1    EH558                        ACC_CNASNP-20160128    0.8 Mb
## 2    EH559                        ACC_CNVSNP-20160128    0.2 Mb
## 3    EH561              ACC_GISTIC_AllByGene-20160128    0.3 Mb
## 4   EH2115                  ACC_GISTIC_Peaks-20160128      0 Mb
## 5    EH562      ACC_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 6   EH2116            ACC_Methylation-20160128_assays  236.4 Mb
## 7   EH2117                ACC_Methylation-20160128_se    6.1 Mb
## 8    EH565                  ACC_miRNASeqGene-20160128    0.1 Mb
## 9    EH566                      ACC_Mutation-20160128    0.7 Mb
## 10   EH567               ACC_RNASeq2GeneNorm-20160128      4 Mb
## 11   EH568                     ACC_RPPAArray-20160128    0.1 Mb
## 12   EH570                       BLCA_CNASeq-20160128    0.3 Mb
## 13   EH571                       BLCA_CNASNP-20160128    4.3 Mb
## 14   EH572                       BLCA_CNVSNP-20160128    1.1 Mb
## 15   EH574             BLCA_GISTIC_AllByGene-20160128    0.6 Mb
## 16  EH2118                 BLCA_GISTIC_Peaks-20160128      0 Mb
## 17   EH575     BLCA_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 18  EH2119           BLCA_Methylation-20160128_assays 1283.4 Mb
## 19  EH2120               BLCA_Methylation-20160128_se    6.1 Mb
## 20   EH578                 BLCA_miRNASeqGene-20160128    0.4 Mb
## 21   EH579                     BLCA_Mutation-20160128    3.7 Mb
## 22   EH580              BLCA_RNASeq2GeneNorm-20160128   21.9 Mb
## 23   EH581                   BLCA_RNASeqGene-20160128    2.4 Mb
## 24   EH582                    BLCA_RPPAArray-20160128    0.5 Mb
## 25   EH584                       BRCA_CNASeq-20160128      0 Mb
## 26   EH585                       BRCA_CNASNP-20160128    9.8 Mb
## 27   EH586                       BRCA_CNVSNP-20160128    2.8 Mb
## 28   EH588             BRCA_GISTIC_AllByGene-20160128    1.3 Mb
## 29  EH2121                 BRCA_GISTIC_Peaks-20160128      0 Mb
## 30   EH589     BRCA_GISTIC_ThresholdedByGene-20160128    0.4 Mb
## 31  EH2122  BRCA_Methylation_methyl27-20160128_assays   63.2 Mb
## 32  EH2123      BRCA_Methylation_methyl27-20160128_se    0.4 Mb
## 33  EH2124 BRCA_Methylation_methyl450-20160128_assays 2613.2 Mb
## 34  EH2125     BRCA_Methylation_methyl450-20160128_se    6.1 Mb
## 35   EH593                 BRCA_miRNASeqGene-20160128    0.6 Mb
## 36   EH594                    BRCA_mRNAArray-20160128   27.3 Mb
## 37   EH595                     BRCA_Mutation-20160128    4.5 Mb
## 38   EH596              BRCA_RNASeq2GeneNorm-20160128   64.5 Mb
## 39   EH597                   BRCA_RNASeqGene-20160128     30 Mb
## 40   EH598                    BRCA_RPPAArray-20160128    1.6 Mb
## 41   EH600                       CESC_CNASeq-20160128    0.1 Mb
## 42   EH601                       CESC_CNASNP-20160128    2.3 Mb
## 43   EH602                       CESC_CNVSNP-20160128    0.6 Mb
## 44   EH604             CESC_GISTIC_AllByGene-20160128    0.5 Mb
## 45  EH2126                 CESC_GISTIC_Peaks-20160128      0 Mb
## 46   EH605     CESC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 47  EH2127           CESC_Methylation-20160128_assays  921.2 Mb
## 48  EH2128               CESC_Methylation-20160128_se    6.1 Mb
## 49   EH608                 CESC_miRNASeqGene-20160128    0.3 Mb
## 50   EH609                     CESC_Mutation-20160128      2 Mb
## 51   EH610              CESC_RNASeq2GeneNorm-20160128   16.1 Mb
## 52   EH611                    CESC_RPPAArray-20160128    0.3 Mb
## 53   EH613                       CHOL_CNASNP-20160128    0.4 Mb
## 54   EH614                       CHOL_CNVSNP-20160128    0.1 Mb
## 55   EH616             CHOL_GISTIC_AllByGene-20160128    0.3 Mb
## 56  EH2129                 CHOL_GISTIC_Peaks-20160128      0 Mb
## 57   EH617     CHOL_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 58  EH2130           CHOL_Methylation-20160128_assays  132.1 Mb
## 59  EH2131               CHOL_Methylation-20160128_se    6.1 Mb
## 60   EH620                 CHOL_miRNASeqGene-20160128    0.1 Mb
## 61   EH621                     CHOL_Mutation-20160128    0.2 Mb
## 62   EH622              CHOL_RNASeq2GeneNorm-20160128    2.4 Mb
## 63   EH623                    CHOL_RPPAArray-20160128    0.1 Mb
## 64   EH625                       COAD_CNASeq-20160128    0.3 Mb
## 65   EH626                       COAD_CNASNP-20160128    3.9 Mb
## 66   EH627                       COAD_CNVSNP-20160128    0.9 Mb
## 67   EH629             COAD_GISTIC_AllByGene-20160128    0.5 Mb
## 68  EH2132                 COAD_GISTIC_Peaks-20160128      0 Mb
## 69   EH630     COAD_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 70  EH2133  COAD_Methylation_methyl27-20160128_assays   37.2 Mb
## 71  EH2134      COAD_Methylation_methyl27-20160128_se    0.4 Mb
## 72  EH2135 COAD_Methylation_methyl450-20160128_assays  983.8 Mb
## 73  EH2136     COAD_Methylation_methyl450-20160128_se    6.1 Mb
## 74   EH634                 COAD_miRNASeqGene-20160128    0.2 Mb
## 75   EH635                    COAD_mRNAArray-20160128    8.1 Mb
## 76   EH636                     COAD_Mutation-20160128    1.2 Mb
## 77   EH637              COAD_RNASeq2GeneNorm-20160128    8.8 Mb
## 78   EH638                   COAD_RNASeqGene-20160128    0.4 Mb
## 79   EH639                    COAD_RPPAArray-20160128    0.6 Mb
## 80   EH641                       DLBC_CNASNP-20160128    0.4 Mb
## 81   EH642                       DLBC_CNVSNP-20160128    0.1 Mb
## 82   EH644             DLBC_GISTIC_AllByGene-20160128    0.2 Mb
## 83  EH2137                 DLBC_GISTIC_Peaks-20160128      0 Mb
## 84   EH645     DLBC_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 85  EH2138           DLBC_Methylation-20160128_assays  141.8 Mb
## 86  EH2139               DLBC_Methylation-20160128_se    6.1 Mb
## 87   EH648                 DLBC_miRNASeqGene-20160128    0.1 Mb
## 88   EH649                     DLBC_Mutation-20160128    0.7 Mb
## 89   EH650              DLBC_RNASeq2GeneNorm-20160128    2.5 Mb
## 90   EH651                    DLBC_RPPAArray-20160128    0.1 Mb
## 91   EH653                       ESCA_CNASeq-20160128    0.2 Mb
## 92   EH654                       ESCA_CNASNP-20160128    1.8 Mb
## 93   EH655                       ESCA_CNVSNP-20160128    0.7 Mb
## 94   EH657             ESCA_GISTIC_AllByGene-20160128    0.5 Mb
## 95  EH2140                 ESCA_GISTIC_Peaks-20160128      0 Mb
## 96   EH658     ESCA_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 97  EH2141           ESCA_Methylation-20160128_assays  596.9 Mb
## 98  EH2142               ESCA_Methylation-20160128_se    6.1 Mb
## 99   EH661                 ESCA_miRNASeqGene-20160128    0.2 Mb
## 100  EH662                     ESCA_Mutation-20160128    2.8 Mb
## 101  EH663              ESCA_RNASeq2GeneNorm-20160128   10.8 Mb
## 102  EH664                   ESCA_RNASeqGene-20160128      8 Mb
## 103  EH665                    ESCA_RPPAArray-20160128    0.2 Mb
## 104  EH667            GBM_CNACGH_CGH_hg_244a-20160128    0.7 Mb
## 105  EH668     GBM_CNACGH_CGH_hg_415k_g4124a-20160128    0.5 Mb
## 106  EH669                        GBM_CNASNP-20160128    5.2 Mb
## 107  EH670                        GBM_CNVSNP-20160128    1.5 Mb
## 108  EH672              GBM_GISTIC_AllByGene-20160128    0.7 Mb
## 109 EH2143                  GBM_GISTIC_Peaks-20160128      0 Mb
## 110  EH673      GBM_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 111 EH2144   GBM_Methylation_methyl27-20160128_assays   52.4 Mb
## 112 EH2145       GBM_Methylation_methyl27-20160128_se    0.4 Mb
## 113 EH2146  GBM_Methylation_methyl450-20160128_assays  455.1 Mb
## 114 EH2147      GBM_Methylation_methyl450-20160128_se    6.1 Mb
## 115  EH677                    GBM_miRNAArray-20160128    2.1 Mb
## 116  EH678                  GBM_miRNASeqGene-20160128      0 Mb
## 117  EH679                GBM_mRNAArray_huex-20160128   56.3 Mb
## 118 EH2148         GBM_mRNAArray_TX_g4502a_1-20160128   22.5 Mb
## 119  EH680           GBM_mRNAArray_TX_g4502a-20160128    5.7 Mb
## 120  EH681      GBM_mRNAArray_TX_ht_hg_u133a-20160128   44.7 Mb
## 121  EH682                      GBM_Mutation-20160128    2.1 Mb
## 122  EH683               GBM_RNASeq2GeneNorm-20160128    8.6 Mb
## 123  EH684                     GBM_RPPAArray-20160128    0.4 Mb
## 124  EH686                       HNSC_CNASeq-20160128    0.3 Mb
## 125  EH687                       HNSC_CNASNP-20160128    4.2 Mb
## 126  EH688                       HNSC_CNVSNP-20160128    1.1 Mb
## 127  EH690             HNSC_GISTIC_AllByGene-20160128    0.6 Mb
## 128 EH2149                 HNSC_GISTIC_Peaks-20160128      0 Mb
## 129  EH691     HNSC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 130 EH2150           HNSC_Methylation-20160128_assays 1714.7 Mb
## 131 EH2151               HNSC_Methylation-20160128_se    6.1 Mb
## 132  EH694                 HNSC_miRNASeqGene-20160128    0.4 Mb
## 133  EH695                     HNSC_Mutation-20160128    4.8 Mb
## 134  EH696              HNSC_RNASeq2GeneNorm-20160128   29.5 Mb
## 135  EH697                   HNSC_RNASeqGene-20160128   10.3 Mb
## 136  EH698                    HNSC_RPPAArray-20160128    0.3 Mb
## 137  EH700                       KICH_CNASNP-20160128    0.5 Mb
## 138  EH701                       KICH_CNVSNP-20160128    0.1 Mb
## 139  EH703             KICH_GISTIC_AllByGene-20160128    0.2 Mb
## 140 EH2152                 KICH_GISTIC_Peaks-20160128      0 Mb
## 141  EH704     KICH_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 142 EH2153           KICH_Methylation-20160128_assays    195 Mb
## 143 EH2154               KICH_Methylation-20160128_se    6.1 Mb
## 144  EH707                 KICH_miRNASeqGene-20160128    0.1 Mb
## 145  EH708                     KICH_Mutation-20160128    0.1 Mb
## 146  EH709              KICH_RNASeq2GeneNorm-20160128    4.9 Mb
## 147  EH710                    KICH_RPPAArray-20160128    0.1 Mb
## 148  EH712                       KIRC_CNASNP-20160128    4.1 Mb
## 149  EH713                       KIRC_CNVSNP-20160128    0.8 Mb
## 150  EH715             KIRC_GISTIC_AllByGene-20160128    0.5 Mb
## 151 EH2155                 KIRC_GISTIC_Peaks-20160128      0 Mb
## 152  EH716     KIRC_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 153 EH2156  KIRC_Methylation_methyl27-20160128_assays   76.8 Mb
## 154 EH2157      KIRC_Methylation_methyl27-20160128_se    0.4 Mb
## 155 EH2158 KIRC_Methylation_methyl450-20160128_assays 1418.8 Mb
## 156 EH2159     KIRC_Methylation_methyl450-20160128_se    6.1 Mb
## 157  EH720                 KIRC_miRNASeqGene-20160128    0.2 Mb
## 158  EH721                    KIRC_mRNAArray-20160128    3.5 Mb
## 159  EH722                     KIRC_Mutation-20160128    0.4 Mb
## 160  EH723              KIRC_RNASeq2GeneNorm-20160128   32.2 Mb
## 161  EH724                   KIRC_RNASeqGene-20160128   18.2 Mb
## 162  EH725                    KIRC_RPPAArray-20160128    0.8 Mb
## 163  EH727                       KIRP_CNASNP-20160128    2.6 Mb
## 164  EH728                       KIRP_CNVSNP-20160128    0.5 Mb
## 165  EH730             KIRP_GISTIC_AllByGene-20160128    0.3 Mb
## 166 EH2160                 KIRP_GISTIC_Peaks-20160128      0 Mb
## 167  EH731     KIRP_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 168 EH2161  KIRP_Methylation_methyl27-20160128_assays    3.9 Mb
## 169 EH2162      KIRP_Methylation_methyl27-20160128_se    0.4 Mb
## 170 EH2163 KIRP_Methylation_methyl450-20160128_assays    948 Mb
## 171 EH2164     KIRP_Methylation_methyl450-20160128_se    6.1 Mb
## 172  EH735                 KIRP_miRNASeqGene-20160128    0.3 Mb
## 173  EH736                    KIRP_mRNAArray-20160128    0.9 Mb
## 174  EH737                     KIRP_Mutation-20160128    0.5 Mb
## 175  EH738              KIRP_RNASeq2GeneNorm-20160128   16.7 Mb
## 176  EH739                   KIRP_RNASeqGene-20160128    0.6 Mb
## 177  EH740                    KIRP_RPPAArray-20160128    0.3 Mb
## 178  EH742                       LAML_CNASNP-20160128    7.6 Mb
## 179  EH743                       LAML_CNVSNP-20160128    0.3 Mb
## 180 EH2538             LAML_GISTIC_AllByGene-20160128    0.3 Mb
## 181 EH2539                 LAML_GISTIC_Peaks-20160128      0 Mb
## 182 EH2540     LAML_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 183 EH2166  LAML_Methylation_methyl27-20160128_assays   35.6 Mb
## 184 EH2167      LAML_Methylation_methyl27-20160128_se    0.4 Mb
## 185 EH2168 LAML_Methylation_methyl450-20160128_assays  572.8 Mb
## 186 EH2169     LAML_Methylation_methyl450-20160128_se    6.1 Mb
## 187  EH748                     LAML_Mutation-20160128    0.2 Mb
## 188  EH749              LAML_RNASeq2GeneNorm-20160128    8.1 Mb
## 189  EH750                   LAML_RNASeqGene-20160128    5.4 Mb
## 190  EH752                        LGG_CNASeq-20160128    0.1 Mb
## 191  EH753                        LGG_CNASNP-20160128    3.3 Mb
## 192  EH754                        LGG_CNVSNP-20160128    0.8 Mb
## 193  EH756              LGG_GISTIC_AllByGene-20160128    0.5 Mb
## 194 EH2170                  LGG_GISTIC_Peaks-20160128      0 Mb
## 195  EH757      LGG_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 196 EH2171            LGG_Methylation-20160128_assays 1564.1 Mb
## 197 EH2172                LGG_Methylation-20160128_se    6.1 Mb
## 198  EH760                  LGG_miRNASeqGene-20160128    0.4 Mb
## 199  EH761                     LGG_mRNAArray-20160128    1.7 Mb
## 200  EH762                      LGG_Mutation-20160128    0.2 Mb
## 201  EH763               LGG_RNASeq2GeneNorm-20160128   28.3 Mb
## 202  EH764                     LGG_RPPAArray-20160128    0.6 Mb
## 203  EH766                       LIHC_CNASNP-20160128    3.1 Mb
## 204  EH767                       LIHC_CNVSNP-20160128      1 Mb
## 205  EH769             LIHC_GISTIC_AllByGene-20160128    0.6 Mb
## 206 EH2173                 LIHC_GISTIC_Peaks-20160128      0 Mb
## 207  EH770     LIHC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 208 EH2174           LIHC_Methylation-20160128_assays 1267.8 Mb
## 209 EH2175               LIHC_Methylation-20160128_se    6.1 Mb
## 210  EH773                 LIHC_miRNASeqGene-20160128    0.3 Mb
## 211  EH774                     LIHC_Mutation-20160128    0.9 Mb
## 212  EH775              LIHC_RNASeq2GeneNorm-20160128   20.6 Mb
## 213  EH776                   LIHC_RNASeqGene-20160128    0.9 Mb
## 214  EH777                    LIHC_RPPAArray-20160128    0.3 Mb
## 215  EH779                       LUAD_CNASeq-20160128    3.8 Mb
## 216  EH780                       LUAD_CNASNP-20160128    4.2 Mb
## 217  EH781                       LUAD_CNVSNP-20160128    1.2 Mb
## 218  EH783             LUAD_GISTIC_AllByGene-20160128    0.7 Mb
## 219 EH2176                 LUAD_GISTIC_Peaks-20160128      0 Mb
## 220  EH784     LUAD_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 221 EH2177  LUAD_Methylation_methyl27-20160128_assays   16.4 Mb
## 222 EH2178      LUAD_Methylation_methyl27-20160128_se    0.4 Mb
## 223 EH2179 LUAD_Methylation_methyl450-20160128_assays 1452.7 Mb
## 224 EH2180     LUAD_Methylation_methyl450-20160128_se    6.1 Mb
## 225  EH788                 LUAD_miRNASeqGene-20160128    0.4 Mb
## 226  EH789                    LUAD_mRNAArray-20160128    1.6 Mb
## 227  EH790                     LUAD_Mutation-20160128    6.7 Mb
## 228  EH791              LUAD_RNASeq2GeneNorm-20160128   29.8 Mb
## 229  EH792                   LUAD_RNASeqGene-20160128    5.6 Mb
## 230  EH793                    LUAD_RPPAArray-20160128    0.6 Mb
## 231  EH795                       LUSC_CNACGH-20160128    0.7 Mb
## 232  EH796                       LUSC_CNASNP-20160128    4.7 Mb
## 233  EH797                       LUSC_CNVSNP-20160128    1.4 Mb
## 234  EH799             LUSC_GISTIC_AllByGene-20160128    0.7 Mb
## 235 EH2181                 LUSC_GISTIC_Peaks-20160128      0 Mb
## 236  EH800     LUSC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 237 EH2182  LUSC_Methylation_methyl27-20160128_assays   29.5 Mb
## 238 EH2183      LUSC_Methylation_methyl27-20160128_se    0.4 Mb
## 239 EH2184 LUSC_Methylation_methyl450-20160128_assays 1218.4 Mb
## 240 EH2185     LUSC_Methylation_methyl450-20160128_se    6.1 Mb
## 241  EH804                 LUSC_miRNASeqGene-20160128    0.3 Mb
## 242  EH805               LUSC_mRNAArray_huex-20160128   14.8 Mb
## 243  EH806          LUSC_mRNAArray_TX_g4502a-20160128    7.2 Mb
## 244  EH807     LUSC_mRNAArray_TX_ht_hg_u133a-20160128   11.4 Mb
## 245  EH808                     LUSC_Mutation-20160128    5.7 Mb
## 246  EH809              LUSC_RNASeq2GeneNorm-20160128   29.2 Mb
## 247  EH810                   LUSC_RNASeqGene-20160128    8.4 Mb
## 248  EH811                    LUSC_RPPAArray-20160128    0.5 Mb
## 249  EH813                       MESO_CNASNP-20160128    0.8 Mb
## 250  EH814                       MESO_CNVSNP-20160128    0.2 Mb
## 251  EH816             MESO_GISTIC_AllByGene-20160128    0.3 Mb
## 252 EH2186                 MESO_GISTIC_Peaks-20160128      0 Mb
## 253  EH817     MESO_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 254 EH2187           MESO_Methylation-20160128_assays  257.2 Mb
## 255 EH2188               MESO_Methylation-20160128_se    6.1 Mb
## 256  EH820                 MESO_miRNASeqGene-20160128    0.1 Mb
## 257  EH821              MESO_RNASeq2GeneNorm-20160128    4.6 Mb
## 258  EH822                    MESO_RPPAArray-20160128    0.1 Mb
## 259  EH824             OV_CNACGH_CGH_hg_244a-20160128    1.1 Mb
## 260  EH825      OV_CNACGH_CGH_hg_415k_g4124a-20160128    2.2 Mb
## 261  EH826                         OV_CNASNP-20160128    8.2 Mb
## 262  EH827                         OV_CNVSNP-20160128    2.7 Mb
## 263  EH829               OV_GISTIC_AllByGene-20160128    1.2 Mb
## 264 EH2189                   OV_GISTIC_Peaks-20160128      0 Mb
## 265  EH830       OV_GISTIC_ThresholdedByGene-20160128    0.4 Mb
## 266 EH2190    OV_Methylation_methyl27-20160128_assays  108.7 Mb
## 267 EH2191        OV_Methylation_methyl27-20160128_se    0.4 Mb
## 268 EH2192   OV_Methylation_methyl450-20160128_assays   29.6 Mb
## 269 EH2193       OV_Methylation_methyl450-20160128_se    6.1 Mb
## 270  EH834                     OV_miRNAArray-20160128    3.2 Mb
## 271  EH835                   OV_miRNASeqGene-20160128    0.3 Mb
## 272  EH836                 OV_mRNAArray_huex-20160128   75.2 Mb
## 273 EH2194          OV_mRNAArray_TX_g4502a_1-20160128   25.2 Mb
## 274  EH837            OV_mRNAArray_TX_g4502a-20160128    1.9 Mb
## 275  EH838       OV_mRNAArray_TX_ht_hg_u133a-20160128   44.6 Mb
## 276  EH839                       OV_Mutation-20160128    0.5 Mb
## 277  EH840                OV_RNASeq2GeneNorm-20160128   16.7 Mb
## 278  EH841                     OV_RNASeqGene-20160128   10.4 Mb
## 279  EH842                      OV_RPPAArray-20160128    0.7 Mb
## 280  EH844                       PAAD_CNASNP-20160128    1.7 Mb
## 281  EH845                       PAAD_CNVSNP-20160128    0.4 Mb
## 282  EH847             PAAD_GISTIC_AllByGene-20160128    0.4 Mb
## 283 EH2195                 PAAD_GISTIC_Peaks-20160128      0 Mb
## 284  EH848     PAAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 285 EH2196           PAAD_Methylation-20160128_assays  575.8 Mb
## 286 EH2197               PAAD_Methylation-20160128_se    6.1 Mb
## 287  EH851                 PAAD_miRNASeqGene-20160128    0.2 Mb
## 288  EH852                     PAAD_Mutation-20160128    6.8 Mb
## 289  EH853              PAAD_RNASeq2GeneNorm-20160128    9.7 Mb
## 290  EH854                    PAAD_RPPAArray-20160128    0.2 Mb
## 291  EH856                       PCPG_CNASNP-20160128    2.6 Mb
## 292  EH857                       PCPG_CNVSNP-20160128    0.3 Mb
## 293  EH859             PCPG_GISTIC_AllByGene-20160128    0.3 Mb
## 294 EH2198                 PCPG_GISTIC_Peaks-20160128      0 Mb
## 295  EH860     PCPG_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 296 EH2199           PCPG_Methylation-20160128_assays  552.9 Mb
## 297 EH2200               PCPG_Methylation-20160128_se    6.1 Mb
## 298  EH863                 PCPG_miRNASeqGene-20160128    0.2 Mb
## 299  EH864                     PCPG_Mutation-20160128    0.6 Mb
## 300  EH865              PCPG_RNASeq2GeneNorm-20160128    9.6 Mb
## 301  EH866                    PCPG_RPPAArray-20160128    0.1 Mb
## 302  EH868                       PRAD_CNASeq-20160128    0.2 Mb
## 303  EH869                       PRAD_CNASNP-20160128    4.9 Mb
## 304  EH870                       PRAD_CNVSNP-20160128    1.2 Mb
## 305  EH872             PRAD_GISTIC_AllByGene-20160128    0.6 Mb
## 306 EH2201                 PRAD_GISTIC_Peaks-20160128      0 Mb
## 307  EH873     PRAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 308 EH2202           PRAD_Methylation-20160128_assays 1622.5 Mb
## 309 EH2203               PRAD_Methylation-20160128_se    6.1 Mb
## 310  EH876                 PRAD_miRNASeqGene-20160128    0.4 Mb
## 311  EH877                     PRAD_Mutation-20160128    1.4 Mb
## 312  EH878              PRAD_RNASeq2GeneNorm-20160128   28.9 Mb
## 313  EH879                    PRAD_RPPAArray-20160128    0.5 Mb
## 314  EH881                       READ_CNASeq-20160128    0.4 Mb
## 315  EH882                       READ_CNASNP-20160128    1.4 Mb
## 316  EH883                       READ_CNVSNP-20160128    0.4 Mb
## 317  EH885             READ_GISTIC_AllByGene-20160128    0.3 Mb
## 318 EH2204                 READ_GISTIC_Peaks-20160128      0 Mb
## 319  EH886     READ_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 320 EH2205  READ_Methylation_methyl27-20160128_assays   13.5 Mb
## 321 EH2206      READ_Methylation_methyl27-20160128_se    0.4 Mb
## 322 EH2207 READ_Methylation_methyl450-20160128_assays  313.3 Mb
## 323 EH2208     READ_Methylation_methyl450-20160128_se    6.1 Mb
## 324  EH890                 READ_miRNASeqGene-20160128    0.1 Mb
## 325  EH891                    READ_mRNAArray-20160128    3.5 Mb
## 326  EH892                     READ_Mutation-20160128    0.4 Mb
## 327  EH893              READ_RNASeq2GeneNorm-20160128    3.5 Mb
## 328  EH894                   READ_RNASeqGene-20160128    2.1 Mb
## 329  EH895                    READ_RPPAArray-20160128    0.2 Mb
## 330  EH897                       SARC_CNASNP-20160128    3.1 Mb
## 331  EH898                       SARC_CNVSNP-20160128    1.1 Mb
## 332  EH900             SARC_GISTIC_AllByGene-20160128    0.6 Mb
## 333 EH2209                 SARC_GISTIC_Peaks-20160128      0 Mb
## 334  EH901     SARC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 335 EH2210           SARC_Methylation-20160128_assays  794.3 Mb
## 336 EH2211               SARC_Methylation-20160128_se    6.1 Mb
## 337  EH904                 SARC_miRNASeqGene-20160128    0.2 Mb
## 338  EH905                     SARC_Mutation-20160128    1.1 Mb
## 339  EH906              SARC_RNASeq2GeneNorm-20160128   13.6 Mb
## 340  EH907                    SARC_RPPAArray-20160128    0.3 Mb
## 341 EH1029                       SKCM_CNASeq-20160128    0.3 Mb
## 342 EH1030                       SKCM_CNASNP-20160128    3.8 Mb
## 343 EH1031                       SKCM_CNVSNP-20160128    1.1 Mb
## 344 EH2541             SKCM_GISTIC_AllByGene-20160128    0.5 Mb
## 345 EH2542                 SKCM_GISTIC_Peaks-20160128      0 Mb
## 346 EH2543     SKCM_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 347 EH2213           SKCM_Methylation-20160128_assays 1403.5 Mb
## 348 EH2214               SKCM_Methylation-20160128_se    6.1 Mb
## 349 EH1033                 SKCM_miRNASeqGene-20160128    0.4 Mb
## 350 EH1034                     SKCM_Mutation-20160128   26.7 Mb
## 351 EH1035              SKCM_RNASeq2GeneNorm-20160128   24.5 Mb
## 352 EH1036                    SKCM_RPPAArray-20160128    0.6 Mb
## 353  EH920                       STAD_CNASeq-20160128    0.3 Mb
## 354  EH921                       STAD_CNASNP-20160128    3.8 Mb
## 355  EH922                       STAD_CNVSNP-20160128    1.2 Mb
## 356  EH924             STAD_GISTIC_AllByGene-20160128    0.7 Mb
## 357 EH2215                 STAD_GISTIC_Peaks-20160128      0 Mb
## 358  EH925     STAD_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 359 EH2216  STAD_Methylation_methyl27-20160128_assays   13.5 Mb
## 360 EH2217      STAD_Methylation_methyl27-20160128_se    0.4 Mb
## 361 EH2218 STAD_Methylation_methyl450-20160128_assays 1172.9 Mb
## 362 EH2219     STAD_Methylation_methyl450-20160128_se    6.1 Mb
## 363  EH929                 STAD_miRNASeqGene-20160128    0.3 Mb
## 364  EH930                     STAD_Mutation-20160128   13.2 Mb
## 365  EH931              STAD_RNASeq2GeneNorm-20160128   23.9 Mb
## 366  EH932                   STAD_RNASeqGene-20160128    1.6 Mb
## 367  EH933                    STAD_RPPAArray-20160128    0.5 Mb
## 368  EH935                       TGCT_CNASNP-20160128    1.2 Mb
## 369  EH936                       TGCT_CNVSNP-20160128    0.3 Mb
## 370  EH938             TGCT_GISTIC_AllByGene-20160128    0.3 Mb
## 371 EH2220                 TGCT_GISTIC_Peaks-20160128      0 Mb
## 372  EH939     TGCT_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 373 EH2221           TGCT_Methylation-20160128_assays  411.3 Mb
## 374 EH2222               TGCT_Methylation-20160128_se    6.1 Mb
## 375  EH942                 TGCT_miRNASeqGene-20160128    0.2 Mb
## 376  EH943                     TGCT_Mutation-20160128    0.5 Mb
## 377  EH944              TGCT_RNASeq2GeneNorm-20160128    7.4 Mb
## 378  EH945                    TGCT_RPPAArray-20160128    0.2 Mb
## 379  EH947                       THCA_CNASeq-20160128      0 Mb
## 380  EH948                       THCA_CNASNP-20160128    3.1 Mb
## 381  EH949                       THCA_CNVSNP-20160128    0.5 Mb
## 382  EH951             THCA_GISTIC_AllByGene-20160128    0.3 Mb
## 383 EH2223                 THCA_GISTIC_Peaks-20160128      0 Mb
## 384  EH952     THCA_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 385 EH2224           THCA_Methylation-20160128_assays 1674.6 Mb
## 386 EH2225               THCA_Methylation-20160128_se    6.1 Mb
## 387  EH955                 THCA_miRNASeqGene-20160128    0.5 Mb
## 388  EH956                     THCA_Mutation-20160128    0.9 Mb
## 389  EH957              THCA_RNASeq2GeneNorm-20160128   30.1 Mb
## 390  EH958                   THCA_RNASeqGene-20160128    0.2 Mb
## 391  EH959                    THCA_RPPAArray-20160128    0.3 Mb
## 392  EH961                       THYM_CNASNP-20160128    0.9 Mb
## 393  EH962                       THYM_CNVSNP-20160128    0.2 Mb
## 394  EH964             THYM_GISTIC_AllByGene-20160128    0.2 Mb
## 395 EH2226                 THYM_GISTIC_Peaks-20160128      0 Mb
## 396  EH965     THYM_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 397 EH2227           THYM_Methylation-20160128_assays  372.2 Mb
## 398 EH2228               THYM_Methylation-20160128_se    6.1 Mb
## 399  EH968                 THYM_miRNASeqGene-20160128    0.1 Mb
## 400  EH969                     THYM_Mutation-20160128    0.2 Mb
## 401  EH970              THYM_RNASeq2GeneNorm-20160128    6.4 Mb
## 402  EH971                    THYM_RPPAArray-20160128    0.2 Mb
## 403  EH973                       UCEC_CNASeq-20160128    0.3 Mb
## 404  EH974                       UCEC_CNASNP-20160128    5.4 Mb
## 405  EH975                       UCEC_CNVSNP-20160128    1.3 Mb
## 406  EH977             UCEC_GISTIC_AllByGene-20160128    0.7 Mb
## 407 EH2229                 UCEC_GISTIC_Peaks-20160128      0 Mb
## 408  EH978     UCEC_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 409 EH2230  UCEC_Methylation_methyl27-20160128_assays   21.7 Mb
## 410 EH2231      UCEC_Methylation_methyl27-20160128_se    0.4 Mb
## 411 EH2232 UCEC_Methylation_methyl450-20160128_assays   1377 Mb
## 412 EH2233     UCEC_Methylation_methyl450-20160128_se    6.1 Mb
## 413  EH982                 UCEC_miRNASeqGene-20160128    0.4 Mb
## 414  EH983                    UCEC_mRNAArray-20160128    2.6 Mb
## 415  EH984                     UCEC_Mutation-20160128    4.6 Mb
## 416  EH985              UCEC_RNASeq2GeneNorm-20160128   18.4 Mb
## 417  EH986                   UCEC_RNASeqGene-20160128    7.7 Mb
## 418  EH987                    UCEC_RPPAArray-20160128    0.7 Mb
## 419  EH989                        UCS_CNASNP-20160128    0.5 Mb
## 420  EH990                        UCS_CNVSNP-20160128    0.2 Mb
## 421  EH992              UCS_GISTIC_AllByGene-20160128    0.3 Mb
## 422 EH2234                  UCS_GISTIC_Peaks-20160128      0 Mb
## 423  EH993      UCS_GISTIC_ThresholdedByGene-20160128    0.3 Mb
## 424 EH2235            UCS_Methylation-20160128_assays  168.3 Mb
## 425 EH2236                UCS_Methylation-20160128_se    6.1 Mb
## 426  EH996                  UCS_miRNASeqGene-20160128    0.1 Mb
## 427  EH997                      UCS_Mutation-20160128    1.2 Mb
## 428  EH998               UCS_RNASeq2GeneNorm-20160128    3.1 Mb
## 429  EH999                     UCS_RPPAArray-20160128    0.1 Mb
## 430 EH1001                        UVM_CNASeq-20160128    0.1 Mb
## 431 EH1002                        UVM_CNASNP-20160128    0.6 Mb
## 432 EH1003                        UVM_CNVSNP-20160128    0.1 Mb
## 433 EH1005              UVM_GISTIC_AllByGene-20160128    0.3 Mb
## 434 EH2237                  UVM_GISTIC_Peaks-20160128      0 Mb
## 435 EH1006      UVM_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 436 EH2238            UVM_Methylation-20160128_assays  236.4 Mb
## 437 EH2239                UVM_Methylation-20160128_se    6.1 Mb
## 438 EH1009                  UVM_miRNASeqGene-20160128    0.1 Mb
## 439 EH1010                      UVM_Mutation-20160128    0.9 Mb
## 440 EH1011               UVM_RNASeq2GeneNorm-20160128      4 Mb
## 441 EH1012                     UVM_RPPAArray-20160128      0 Mb
##                     rdataclass rdatadateadded rdatadateremoved
## 1             RaggedExperiment     2017-10-10             &lt;NA&gt;
## 2             RaggedExperiment     2017-10-10             &lt;NA&gt;
## 3         SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 4   RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 5         SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 6         SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 7             RaggedExperiment     2019-01-09             &lt;NA&gt;
## 8         SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 9             RaggedExperiment     2017-10-10             &lt;NA&gt;
## 10        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 11        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 12            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 13            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 14            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 15        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 16  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 17        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 18        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 19            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 20        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 21            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 22        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 23        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 24        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 25            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 26            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 27            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 28        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 29  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 30        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 31        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 32        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 33            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 34        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 35        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 36        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 37            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 38        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 39        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 40        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 41            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 42            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 43            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 44        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 45  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 46        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 47        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 48            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 49        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 50            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 51        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 52        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 53            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 54            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 55        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 56  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 57        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 58        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 59            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 60        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 61            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 62        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 63        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 64            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 65            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 66            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 67        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 68  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 69        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 70        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 71        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 72            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 73        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 74        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 75        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 76            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 77        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 78        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 79        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 80            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 81            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 82        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 83  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 84        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 85        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 86            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 87        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 88            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 89        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 90        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 91            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 92            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 93            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 94        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 95  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 96        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 97        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 98            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 99        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 100           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 101       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 102       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 103       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 104           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 105           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 106           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 107           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 108       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 109 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 110       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 111       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 112       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 113       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 114       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 115       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 116       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 117       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 118       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 119       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 120       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 121           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 122       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 123       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 124           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 125           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 126           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 127       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 128 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 129       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 130       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 131           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 132       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 133           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 134       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 135       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 136       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 137           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 138           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 139       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 140 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 141       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 142       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 143           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 144       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 145           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 146       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 147       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 148           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 149           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 150       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 151 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 152       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 153       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 154       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 155           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 156       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 157       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 158       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 159           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 160       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 161       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 162       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 163           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 164           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 165       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 166 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 167       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 168       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 169       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 170           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 171       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 172       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 173       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 174           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 175       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 176       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 177       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 178           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 179           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 180       SummarizedExperiment     2019-04-29             &lt;NA&gt;
## 181 RangedSummarizedExperiment     2019-04-29             &lt;NA&gt;
## 182       SummarizedExperiment     2019-04-29             &lt;NA&gt;
## 183           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 184       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 185       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 186                  DataFrame     2019-01-09             &lt;NA&gt;
## 187           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 188       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 189       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 190           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 191           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 192           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 193       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 194 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 195       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 196       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 197       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 198       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 199       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 200           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 201       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 202       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 203           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 204           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 205       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 206 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 207       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 208       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 209           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 210       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 211           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 212       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 213       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 214       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 215           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 216           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 217           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 218       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 219 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 220       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 221       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 222       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 223           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 224       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 225       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 226       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 227           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 228       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 229       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 230       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 231           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 232           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 233           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 234       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 235 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 236       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 237       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 238       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 239       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 240       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 241       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 242       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 243       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 244       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 245           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 246       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 247       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 248       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 249           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 250           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 251       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 252 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 253       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 254       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 255       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 256       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 257       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 258       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 259           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 260           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 261           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 262           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 263       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 264 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 265       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 266       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 267       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 268       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 269       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 270       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 271       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 272       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 273       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 274       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 275       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 276           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 277       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 278       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 279       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 280           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 281           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 282       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 283 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 284       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 285       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 286           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 287       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 288           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 289       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 290       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 291           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 292           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 293       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 294 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 295       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 296       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 297           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 298       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 299           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 300       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 301       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 302           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 303           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 304           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 305       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 306 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 307       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 308       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 309           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 310       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 311           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 312       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 313       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 314           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 315           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 316           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 317       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 318 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 319       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 320       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 321       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 322           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 323       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 324       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 325       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 326           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 327       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 328       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 329       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 330           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 331           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 332       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 333 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 334       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 335       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 336           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 337       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 338           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 339       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 340       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 341           RaggedExperiment     2017-10-16             &lt;NA&gt;
## 342           RaggedExperiment     2017-10-16             &lt;NA&gt;
## 343           RaggedExperiment     2017-10-16             &lt;NA&gt;
## 344       SummarizedExperiment     2019-04-29             &lt;NA&gt;
## 345 RangedSummarizedExperiment     2019-04-29             &lt;NA&gt;
## 346       SummarizedExperiment     2019-04-29             &lt;NA&gt;
## 347       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 348           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 349       SummarizedExperiment     2017-10-16             &lt;NA&gt;
## 350           RaggedExperiment     2017-10-16             &lt;NA&gt;
## 351       SummarizedExperiment     2017-10-16             &lt;NA&gt;
## 352       SummarizedExperiment     2017-10-16             &lt;NA&gt;
## 353           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 354           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 355           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 356       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 357 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 358       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 359       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 360           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 361       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 362       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 363       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 364           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 365       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 366       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 367       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 368           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 369           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 370       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 371 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 372       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 373       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 374           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 375       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 376           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 377       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 378       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 379           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 380           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 381           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 382       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 383 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 384       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 385       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 386           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 387       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 388           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 389       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 390       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 391       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 392           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 393           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 394       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 395 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 396       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 397       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 398           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 399       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 400           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 401       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 402       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 403           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 404           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 405           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 406       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 407 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 408       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 409       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 410       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 411           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 412       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 413       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 414       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 415           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 416       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 417       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 418       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 419           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 420           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 421       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 422 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 423       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 424       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 425           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 426       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 427           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 428       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 429       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 430           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 431           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 432           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 433       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 434 RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 435       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 436       SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 437           RaggedExperiment     2019-01-09             &lt;NA&gt;
## 438       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 439           RaggedExperiment     2017-10-10             &lt;NA&gt;
## 440       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 441       SummarizedExperiment     2017-10-10             &lt;NA&gt;</code></pre>
<p>Check potential files to be downloaded for adrenocortical carcinoma (ACC):</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span><span class="op">(</span>diseaseCode <span class="op">=</span> <span class="st">"ACC"</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2020-10-27</code></pre>
<pre><code>## See '?curatedTCGAData' for 'diseaseCode' and 'assays' inputs</code></pre>
<pre><code>##     ah_id                                 title file_size
## 1   EH558                   ACC_CNASNP-20160128    0.8 Mb
## 2   EH559                   ACC_CNVSNP-20160128    0.2 Mb
## 3   EH561         ACC_GISTIC_AllByGene-20160128    0.3 Mb
## 4  EH2115             ACC_GISTIC_Peaks-20160128      0 Mb
## 5   EH562 ACC_GISTIC_ThresholdedByGene-20160128    0.2 Mb
## 6  EH2116       ACC_Methylation-20160128_assays  236.4 Mb
## 7  EH2117           ACC_Methylation-20160128_se    6.1 Mb
## 8   EH565             ACC_miRNASeqGene-20160128    0.1 Mb
## 9   EH566                 ACC_Mutation-20160128    0.7 Mb
## 10  EH567          ACC_RNASeq2GeneNorm-20160128      4 Mb
## 11  EH568                ACC_RPPAArray-20160128    0.1 Mb
##                    rdataclass rdatadateadded rdatadateremoved
## 1            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 2            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 3        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 4  RangedSummarizedExperiment     2019-01-09             &lt;NA&gt;
## 5        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 6        SummarizedExperiment     2019-01-09             &lt;NA&gt;
## 7            RaggedExperiment     2019-01-09             &lt;NA&gt;
## 8        SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 9            RaggedExperiment     2017-10-10             &lt;NA&gt;
## 10       SummarizedExperiment     2017-10-10             &lt;NA&gt;
## 11       SummarizedExperiment     2017-10-10             &lt;NA&gt;</code></pre>
<p>Actually download the reverse phase protein array (RPPA) and RNA-seq data for ACC</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ACCmae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span><span class="op">(</span><span class="st">"ACC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"RPPAArray"</span>, <span class="st">"RNASeq2GeneNorm"</span><span class="op">)</span>, 
                          dry.run<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">ACCmae</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 2 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 2:
##  [1] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns
##  [2] ACC_RPPAArray-20160128: SummarizedExperiment with 192 rows and 46 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
<p><strong>Note</strong>. Data will be downloaded the first time the above command is run; subsequent times it will be loaded from local cache.</p>
<p>This object contains 822 columns of clinical, pathological, specimen, and subtypes data in its <code>colData</code>, merged from all available data levels (1-4) of the Firehose pipeline:</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">ACCmae</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1]  79 822</code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">ACCmae</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "patientID"             "years_to_birth"        "vital_status"         
## [4] "days_to_death"         "days_to_last_followup" "tumor_tissue_site"</code></pre>
<p>See the <em><a href="https://bioconductor.org/packages/3.12/MultiAssayExperiment">MultiAssayExperiment</a></em> vignette <span class="citation">(Ramos et al. 2017)</span> and the <a href="https://waldronlab.io/MultiAssayWorkshop/">Workflow for Multi-omics Analysis with MultiAssayExperiment</a> workshop (<a href="https://waldronlab.io/MultiAssayWorkshop/" class="uri">https://waldronlab.io/MultiAssayWorkshop/</a>) for details on using this object.</p>
<div id="subtype-information" class="section level4">
<h4 class="hasAnchor">
<a href="#subtype-information" class="anchor"></a>Subtype information</h4>
<p>Some cancer datasets contain associated subtype information within the clinical datasets provided. This subtype information is included in the metadata of <code>colData</code> of the <code>MultiAssayExperiment</code> object. To obtain these variable names, run the <code>metadata</code> function on the <code>colData</code> of the object such as:</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">ACCmae</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="st">"subtypes"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##         ACC_annotations   ACC_subtype
## 1            Patient_ID        SAMPLE
## 2 histological_subtypes     Histology
## 3         mrna_subtypes       C1A/C1B
## 4         mrna_subtypes       mRNA_K4
## 5                  cimp    MethyLevel
## 6     microrna_subtypes miRNA cluster</code></pre>
</div>
</div>
</div>
<div id="cbioportaldata" class="section level2">
<h2 class="hasAnchor">
<a href="#cbioportaldata" class="anchor"></a>cBioPortalData</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/cBioPortalData">cBioPortalData</a></em> accesses <em>most</em> data from the Cancer Biology Portal cbioportal.org within R/Bioconductor using <em><a href="https://bioconductor.org/packages/3.12/MultiAssayExperiment">MultiAssayExperiment</a></em>. It provides two main functions:</p>
<ul>
<li>
<code>cBioDataPack</code> for pre-packaged full datasets</li>
<li>
<code>cBioPortalData</code> for API-query data</li>
</ul>
<p>Interactive workshop: <a href="https://waldronlab.io/MultiAssayWorkshop/tutorials/cBioPortalData_Intro.html" class="uri">https://waldronlab.io/MultiAssayWorkshop/tutorials/cBioPortalData_Intro.html</a></p>
</div>
<div id="recount-reproducible-rna-seq-analysis-using-recount2" class="section level2">
<h2 class="hasAnchor">
<a href="#recount-reproducible-rna-seq-analysis-using-recount2" class="anchor"></a>recount: Reproducible RNA-seq Analysis Using recount2</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em><span class="citation">(Collado-Torres et al. 2017)</span> package provides uniformly processed <code>RangedSummarizedExperiment</code> objects at the gene, exon, or exon-exon junctions level, the raw counts, the phenotype metadata used, the urls to sample coverage bigWig files and mean coverage bigWig file, for every study available. The <code>RangedSummarizedExperiment</code> objects can be used for differential expression analysis. These are also accessible through a web interface.<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p>
<p><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> provides a search function:</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/leekgroup/recount">recount</a></span><span class="op">)</span>
<span class="va">project_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search</a></span><span class="op">(</span><span class="st">'GSE32465'</span><span class="op">)</span></code></pre></div>
<p>It is not an ExperimentHub package, so downloading and serializing is slightly more involved in involves two steps: first, download the gene-level RangedSummarizedExperiment data:</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span><span class="op">)</span></code></pre></div>
<pre><code>## 2021-04-29 12:21:06 downloading file rse_gene.Rdata to SRP009615</code></pre>
<p>followed by loading the data</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span>, <span class="st">'rse_gene.Rdata'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="curateddata-packages-for-standardized-cancer-transcriptomes" class="section level2">
<h2 class="hasAnchor">
<a href="#curateddata-packages-for-standardized-cancer-transcriptomes" class="anchor"></a>curated*Data packages for standardized cancer transcriptomes</h2>
<p>There are focused databases of cancer microarray data for several cancer types, which can be useful for researchers of those cancer types or for methodological development:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedOvarianData">curatedOvarianData</a></em><span class="citation">(Ganzfried et al. 2013)</span>: Clinically Annotated Data for the Ovarian Cancer Transcriptome (data available with additional options through the <em><a href="https://bioconductor.org/packages/3.12/MetaGxOvarian">MetaGxOvarian</a></em> package).</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedBladderData">curatedBladderData</a></em>: Clinically Annotated Data for the Bladder Cancer Transcriptome</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedCRCData">curatedCRCData</a></em>: Clinically Annotated Data for the Colorectal Cancer Transcriptome</li>
</ul>
<p>These provide data from the Gene Expression Omnibus and other sources, but use a formally vocabulary for clinicopathological data and use a common pipeline for preprocessing of microarray data (for Affymetrix, other for other platforms the processed data are provided as processed by original study authors), merging probesets, and mapping to gene symbols. The pipeline is described by <span class="citation">Ganzfried et al. (2013)</span>.</p>
</div>
<div id="microbiome-data" class="section level2">
<h2 class="hasAnchor">
<a href="#microbiome-data" class="anchor"></a>Microbiome data</h2>
<p>Bioconductor provides curated resources of microbiome data. Most microbiome data are generated either by targeted amplicon sequencing (usually of variable regions of the 16S ribosomal RNA gene) or by metagenomic shotgun sequencing (MGX). These two approaches are analyzed by different sequence analysis tools, but downstream statistical and ecological analysis can involve any of the following types of data:</p>
<ul>
<li>taxonomic abundance at different levels of the taxonomic hierarchy</li>
<li>phylogenetic distances and the phylogenetic tree of life</li>
<li>metabolic potential of the microbiome</li>
<li>abundance of microbial genes and gene families</li>
</ul>
<p>A review of types and properties of microbiome data is provided by <span class="citation">(Morgan and Huttenhower 2012)</span>.</p>
<div id="curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub" class="section level3">
<h3 class="hasAnchor">
<a href="#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub" class="anchor"></a>curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</h3>
<p><code>curatedMetagenomicData</code><span class="citation">(Pasolli et al. 2017)</span> provides 6 types of processed data for &gt;30 publicly available whole-metagenome shotgun sequencing datasets (obtained from the Sequence Read Archive):</p>
<ol style="list-style-type: decimal">
<li>Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level</li>
<li>Presence of unique, clade-specific markers</li>
<li>Abundance of unique, clade-specific markers</li>
<li>Abundance of gene families</li>
<li>Metabolic pathway coverage</li>
<li>Metabolic pathway abundance</li>
</ol>
<p>Types 1-3 are generated by <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a>; 4-6 are generated by <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a>.</p>
<p>Currently, <code>curatedMetagenomicData</code> provides:</p>
<ul>
<li>10199 samples from 57 datasets, primarily of the human gut but including body sites profiled in the Human Microbiome Project</li>
<li>Processed data from whole-metagenome shotgun metagenomics, with manually-curated metadata, as integrated and documented Bioconductor ExpressionSet objects</li>
<li>~80 fields of specimen metadata from original papers, supplementary files, and websites, with manual curation to standardize annotations</li>
<li>Processing of data through the <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a> pipeline for taxonomic abundance, and <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a> pipeline for metabolic analysis</li>
<li>These represent ~100TB of raw sequencing data, but the processed data provided are much smaller.</li>
</ul>
<p>These datasets are documented in the <a href="https://bioconductor.org/packages/devel/data/experiment/manuals/curatedMetagenomicData/man/curatedMetagenomicData.pdf">reference manual</a>.</p>
<p>This is an <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em> package, and its main workhorse function is <code><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData()</a></code>:</p>
<p>The manually curated metadata for all available samples are provided in a single table <code>combined_metadata</code>:</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></span><span class="op">)</span>
<span class="op">?</span><span class="va">combined_metadata</span>
<span class="fu"><a href="https://rdrr.io/r/utils/View.html">View</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">combined_metadata</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The main function provides a <code>list</code> of <code>ExpressionSet</code> objects:</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">oral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BritoIL_2016.metaphlan_bugs_list.oralcavity"</span>,
          <span class="st">"Castro-NallarE_2015.metaphlan_bugs_list.oralcavity"</span><span class="op">)</span>
<span class="va">esl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="va">oral</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">esl</span></code></pre></div>
<pre><code>## List of length 2
## names(2): BritoIL_2016.metaphlan_bugs_list.oralcavity ...</code></pre>
<p>These <code>ExpressionSet</code> objects can also be converted to <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> object for ecological analysis and differential abundance analysis using the <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> package, using the <code><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq()</a></code> function:</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq</a></span><span class="op">(</span> <span class="va">esl</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, phylogenetictree <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: phyloseq</code></pre>
<pre><code>## Warning: `data_frame()` was deprecated in tibble 1.1.0.
## Please use `tibble()` instead.</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 535 taxa and 140 samples ]
## sample_data() Sample Data:       [ 140 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 535 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 535 tips and 534 internal nodes ]</code></pre>
<p>See the documentation of <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> for more on ecological and differential abundance analysis of the microbiome.</p>
</div>
<div id="hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project" class="section level3">
<h3 class="hasAnchor">
<a href="#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project" class="anchor"></a>HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</h3>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/HMP16SData">HMP16SData</a></span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2020-10-27</code></pre>
<p><em><a href="https://bioconductor.org/packages/3.12/HMP16SData">HMP16SData</a></em><span class="citation">(Schiffer et al. 2018)</span> is a Bioconductor ExperimentData package of the Human Microbiome Project (HMP) 16S rRNA sequencing data. Taxonomic count data files are provided as downloaded from the HMP Data Analysis and Coordination Center from its QIIME pipeline. Processed data is provided as <code>SummarizedExperiment</code> class objects via <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>. Like other ExperimentHub-based packages, a convenience function does downloading, automatic local caching, and serializing of a Bioconductor data class. This returns taxonomic counts from the V1-3 variable region of the 16S rRNA gene, along with the unrestricted participant data and phylogenetic tree.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/HMP16SData/man/V13.html">V13</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 43140 2898 
## metadata(2): experimentData phylogeneticTree
## assays(1): 16SrRNA
## rownames(43140): OTU_97.1 OTU_97.10 ... OTU_97.9997 OTU_97.9999
## rowData names(7): CONSENSUS_LINEAGE SUPERKINGDOM ... FAMILY GENUS
## colnames(2898): 700013549 700014386 ... 700114963 700114965
## colData names(7): RSID VISITNO ... HMP_BODY_SUBSITE SRS_SAMPLE_ID</code></pre>
<p>This can also be converted to <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> for ecological and differential abundance analysis; see the <em><a href="https://bioconductor.org/packages/3.12/HMP16SData">HMP16SData</a></em> vignette for details.</p>
</div>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Collado-Torres2017-ww" class="csl-entry">
Collado-Torres, Leonardo, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, and Jeffrey T Leek. 2017. <span>“Reproducible <span class="nocase">RNA-seq</span> Analysis Using Recount2.”</span> <em>Nature Biotechnology</em> 35 (4): 319–21. <a href="https://doi.org/10.1038/nbt.3838">https://doi.org/10.1038/nbt.3838</a>.
</div>
<div id="ref-Sean2007-cv" class="csl-entry">
Davis, Sean R., and Paul S Meltzer. 2007. <span>“<span>GEOquery</span>: A Bridge Between the Gene Expression Omnibus (<span>GEO</span>) and <span>BioConductor</span>.”</span> <em>Bioinformatics</em> 23 (14): 1846–47. <a href="https://doi.org/10.1093/bioinformatics/btm254">https://doi.org/10.1093/bioinformatics/btm254</a>.
</div>
<div id="ref-Ganzfried2013-mc" class="csl-entry">
Ganzfried, Benjamin Frederick, Markus Riester, Benjamin Haibe-Kains, Thomas Risch, Svitlana Tyekucheva, Ina Jazic, Xin Victoria Wang, et al. 2013. <span>“<span class="nocase">curatedOvarianData</span>: Clinically Annotated Data for the Ovarian Cancer Transcriptome.”</span> <em>Database: The Journal of Biological Databases and Curation</em> 2013 (April): bat013. <a href="https://doi.org/10.1093/database/bat013">https://doi.org/10.1093/database/bat013</a>.
</div>
<div id="ref-Morgan2012-mq" class="csl-entry">
Morgan, Xochitl C, and Curtis Huttenhower. 2012. <span>“Chapter 12: Human Microbiome Analysis.”</span> <em>PLoS Computational Biology</em> 8 (12): e1002808. <a href="https://doi.org/10.1371/journal.pcbi.1002808">https://doi.org/10.1371/journal.pcbi.1002808</a>.
</div>
<div id="ref-Pasolli2017-gf" class="csl-entry">
Pasolli, Edoardo, Lucas Schiffer, Paolo Manghi, Audrey Renson, Valerie Obenchain, Duy Tin Truong, Francesco Beghini, et al. 2017. <span>“Accessible, Curated Metagenomic Data Through <span>ExperimentHub</span>.”</span> <em>Nature Methods</em> 14 (11): 1023–24. <a href="https://doi.org/10.1038/nmeth.4468">https://doi.org/10.1038/nmeth.4468</a>.
</div>
<div id="ref-Ramos2017-nq" class="csl-entry">
Ramos, Marcel, Lucas Schiffer, Angela Re, Rimsha Azhar, Azfar Basunia, Carmen Rodriguez, Tiffany Chan, et al. 2017. <span>“Software for the Integration of Multiomics Experiments in Bioconductor.”</span> <em>Cancer Research</em> 77 (21): e39–42. <a href="https://doi.org/10.1158/0008-5472.CAN-17-0344">https://doi.org/10.1158/0008-5472.CAN-17-0344</a>.
</div>
<div id="ref-Schiffer2018-qy" class="csl-entry">
Schiffer, Lucas, Rimsha Azhar, Lori Shepherd, Marcel Ramos, Ludwig Geistlinger, Curtis Huttenhower, Jennifer B Dowd, Nicola Segata, and Levi Waldron. 2018. <span>“<span>HMP16SData</span>: Efficient Access to the Human Microbiome Project Through Bioconductor.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/299115">https://doi.org/10.1101/299115</a>.
</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>See individual function and methods documentation for specific details.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing" class="uri">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0" class="uri">https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing" class="uri">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html</a><a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p><a href="https://jhubiostatistics.shinyapps.io/recount/" class="uri">https://jhubiostatistics.shinyapps.io/recount/</a><a href="#fnref6" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sean Davis, Benjamin Haibe-Kains, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
