<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • PublicDataResources</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="PublicDataResources">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PublicDataResources</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/PublicDataResources.html">Workshop</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    External Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html">PharmacoGx</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/bioc/vignettes/GEOquery/inst/doc/GEOquery.html">GEOquery</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/devel/bioc/html/TCGAbiolinks.html">TCGAbiolinks</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicDataCommons/inst/doc/overview.html">GenomicDataCommons</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/curatedTCGAData/inst/doc/curatedTCGAData.html">curatedTCGAData</a>
    </li>
    <li>
      <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/TCGAutils/inst/doc/TCGAutils.html">TCGAutils</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/curatedMetagenomicData/inst/doc/curatedMetagenomicData.html">curatedMetagenomicData</a>
    </li>
    <li>
      <a href="https://bioconductor.org/packages/release/data/experiment/html/HMP16SData.html">HMP16SData</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/PublicDataResources">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://hub.docker.com/repository/docker/waldronlab/publicdataresources">
    <span class="fa fa-docker"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>PublicDataResources.Rmd</code></div>

    </div>

    
    
<div id="public-data-resources-and-bioconductor" class="section level1">
<h1 class="hasAnchor">
<a href="#public-data-resources-and-bioconductor" class="anchor"></a>Public data resources and Bioconductor</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>If you are using the Docker container provided with this workshop, everything (including the workshop) is already installed. Otherwise, you will need to install the following packages.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">required_pkgs</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
  <span class="st">"GEOquery"</span>,
  <span class="st">"GenomicDataCommons"</span>,
  <span class="st">"curatedTCGAData"</span>,
  <span class="st">"recount"</span>,
  <span class="st">"curatedMetagenomicData"</span>,
  <span class="st">"phyloseq"</span>,
  <span class="st">"HMP16SData"</span>,
  <span class="st">"PharmacoGx"</span>)
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="no">required_pkgs</span>)</pre></body></html></div>
</div>
<div id="geoquery" class="section level2">
<h2 class="hasAnchor">
<a href="#geoquery" class="anchor"></a>GEOquery</h2>
<p><span class="citation">(Davis and Meltzer 2007)</span></p>
<p>The NCBI Gene Expression Omnibus (GEO) serves as a public repository for a wide range of high-throughput experimental data. These data include single and dual channel microarray-based experiments measuring mRNA, genomic DNA, and protein abundance, as well as non-array techniques such as serial analysis of gene expression (SAGE), mass spectrometry proteomic data, and high-throughput sequencing data. The <em><a href="https://bioconductor.org/packages/3.12/GEOquery">GEOquery</a></em> package <span class="citation">(Davis and Meltzer 2007)</span> forms a bridge between this public repository and the analysis capabilities in Bioconductor.</p>
<div id="overview-of-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-of-geo" class="anchor"></a>Overview of GEO</h3>
<p>At the most basic level of organization of GEO, there are four basic entity types. The first three (Sample, Platform, and Series) are supplied by users; the fourth, the dataset, is compiled and curated by GEO staff from the user-submitted data. See <a href="https://www.ncbi.nlm.nih.gov/geo/">the GEO home page</a> for more information.</p>
<div id="platforms" class="section level4">
<h4 class="hasAnchor">
<a href="#platforms" class="anchor"></a>Platforms</h4>
<p>A Platform record describes the list of elements on the array (e.g., cDNAs, oligonucleotide probesets, ORFs, antibodies) or the list of elements that may be detected and quantified in that experiment (e.g., SAGE tags, peptides). Each Platform record is assigned a unique and stable GEO accession number (GPLxxx). A Platform may reference many Samples that have been submitted by multiple submitters.</p>
</div>
<div id="samples" class="section level4">
<h4 class="hasAnchor">
<a href="#samples" class="anchor"></a>Samples</h4>
<p>A Sample record describes the conditions under which an individual Sample was handled, the manipulations it underwent, and the abundance measurement of each element derived from it. Each Sample record is assigned a unique and stable GEO accession number (GSMxxx). A Sample entity must reference only one Platform and may be included in multiple Series.</p>
</div>
<div id="series" class="section level4">
<h4 class="hasAnchor">
<a href="#series" class="anchor"></a>Series</h4>
<p>A Series record defines a set of related Samples considered to be part of a group, how the Samples are related, and if and how they are ordered. A Series provides a focal point and description of the experiment as a whole. Series records may also contain tables describing extracted data, summary conclusions, or analyses. Each Series record is assigned a unique and stable GEO accession number (GSExxx). Series records are available in a couple of formats which are handled by GEOquery independently. The smaller and new GSEMatrix files are quite fast to parse; a simple flag is used by GEOquery to choose to use GSEMatrix files (see below).</p>
</div>
<div id="datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h4>
<p>GEO DataSets (GDSxxx) are curated sets of GEO Sample data. There are hundreds of GEO datasets available, but GEO discontinued creating GDS records several years ago. We mention them here for completeness only.</p>
</div>
</div>
<div id="getting-started-using-geoquery" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started-using-geoquery" class="anchor"></a>Getting Started using GEOquery</h3>
<p>Getting data from GEO is really quite easy. There is only one command that is needed, <code>getGEO</code>. This one function interprets its input to determine how to get the data from GEO and then parse the data into useful R data structures.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GEOquery</span>)</pre></body></html></div>
<p>With the library loaded, we are free to access any GEO accession.</p>
</div>
<div id="use-case-mds-plot-of-cancer-data" class="section level3">
<h3 class="hasAnchor">
<a href="#use-case-mds-plot-of-cancer-data" class="anchor"></a>Use case: MDS plot of cancer data</h3>
<p>The data we are going to access are from <a href="https://doi.org/10.1158/1055-9965.EPI-17-0461">this paper</a>.</p>
<blockquote>
<p>Background: The tumor microenvironment is an important factor in cancer immunotherapy response. To further understand how a tumor affects the local immune system, we analyzed immune gene expression differences between matching normal and tumor tissue.Methods: We analyzed public and new gene expression data from solid cancers and isolated immune cell populations. We also determined the correlation between CD8, FoxP3 IHC, and our gene signatures.Results: We observed that regulatory T cells (Tregs) were one of the main drivers of immune gene expression differences between normal and tumor tissue. A tumor-specific CD8 signature was slightly lower in tumor tissue compared with normal of most (12 of 16) cancers, whereas a Treg signature was higher in tumor tissue of all cancers except liver. Clustering by Treg signature found two groups in colorectal cancer datasets. The high Treg cluster had more samples that were consensus molecular subtype 1/4, right-sided, and microsatellite-instable, compared with the low Treg cluster. Finally, we found that the correlation between signature and IHC was low in our small dataset, but samples in the high Treg cluster had significantly more CD8+ and FoxP3+ cells compared with the low Treg cluster.Conclusions: Treg gene expression is highly indicative of the overall tumor immune environment.Impact: In comparison with the consensus molecular subtype and microsatellite status, the Treg signature identifies more colorectal tumors with high immune activation that may benefit from cancer immunotherapy.</p>
</blockquote>
<p>In this little exercise, we will:</p>
<ol style="list-style-type: decimal">
<li>Access public omics data using the GEOquery package</li>
<li>Convert the public omics data to a <code>SummarizedExperiment</code> object.</li>
<li>Perform a simple unsupervised analysis to visualize these public data.</li>
</ol>
<p>Use the <a href="https://bioconductor.org/packages/GEOquery">GEOquery</a> package to fetch data about <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE103512">GSE103512</a>.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">gse</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html">getGEO</a></span>(<span class="st">"GSE103512"</span>)<span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<pre><code>## Warning: 102 parsing failures.
##   row     col           expected    actual         file
## 54614 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54615 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54616 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54617 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## 54618 SPOT_ID 1/0/T/F/TRUE/FALSE --Control literal data
## ..... ....... .................. ......... ............
## See problems(...) for more details.</code></pre>
<p>Note that <code>getGEO</code>, when used to retrieve <em>GSE</em> records, returns a list. The members of the list each represent one <em>GEO Platform</em>, since each <em>GSE</em> record can contain multiple related datasets (eg., gene expression and DNA methylation). In this case, the list is of length one, but it is still necessary to grab the first elment.</p>
<p>The first step–a detail–is to convert from the older Bioconductor data structure (GEOquery was written in 2007), the <code>ExpressionSet</code>, to the newer <code>SummarizedExperiment</code>. One line suffices.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SummarizedExperiment</span>)
<span class="no">se</span> <span class="kw">=</span> <span class="fu">as</span>(<span class="no">gse</span>, <span class="st">"SummarizedExperiment"</span>)</pre></body></html></div>
<p>Examine two variables of interest, cancer type and tumor/normal status.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">se</span>),<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">`cancer.type.ch1`</span>,<span class="no">`normal.ch1`</span>))</pre></body></html></div>
<pre><code>##                normal.ch1
## cancer.type.ch1 no yes
##           BC    65  10
##           CRC   57  12
##           NSCLC 60   9
##           PCA   60   7</code></pre>
<p>Filter gene expression by variance to find most informative genes.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">sds</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span>(<span class="no">se</span>, <span class="st">'exprs'</span>),<span class="fl">1</span>,<span class="no">sd</span>)
<span class="no">dat</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span>(<span class="no">se</span>, <span class="st">'exprs'</span>)[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">sds</span>,<span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">TRUE</span>)[<span class="fl">1</span>:<span class="fl">500</span>],]</pre></body></html></div>
<p>Perform <a href="https://en.wikipedia.org/wiki/Multidimensional_scaling">multidimensional scaling</a> and prepare for plotting. We will be using ggplot2, so we need to make a data.frame before plotting.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">mdsvals</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cmdscale.html">cmdscale</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="no">dat</span>)))
<span class="no">mdsvals</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">mdsvals</span>)
<span class="no">mdsvals</span>$<span class="no">Type</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">se</span>)[,<span class="st">'cancer.type.ch1'</span>])
<span class="no">mdsvals</span>$<span class="no">Normal</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">se</span>)[,<span class="st">'normal.ch1'</span>])
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">mdsvals</span>)</pre></body></html></div>
<pre><code>##                   V1       V2 Type Normal
## GSM2772660  8.531331 18.57115   BC     no
## GSM2772661  8.991591 13.63764   BC     no
## GSM2772662 10.788973 13.48403   BC     no
## GSM2772663  3.127105 19.13529   BC     no
## GSM2772664 13.056599 13.88711   BC     no
## GSM2772665  7.903717 13.24731   BC     no</code></pre>
<p>And do the plot.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">mdsvals</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">V1</span>,<span class="kw">y</span><span class="kw">=</span><span class="no">V2</span>,<span class="kw">shape</span><span class="kw">=</span><span class="no">Normal</span>,<span class="kw">color</span><span class="kw">=</span><span class="no">Type</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>( <span class="kw">alpha</span><span class="kw">=</span><span class="fl">0.6</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">text</span><span class="kw">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">18</span>))</pre></body></html></div>
<p><img src="PublicDataResources_files/figure-html/mdsplot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="accessing-raw-data-from-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-raw-data-from-geo" class="anchor"></a>Accessing Raw Data from GEO</h3>
<p>NCBI GEO accepts (but has not always required) raw data such as .CEL files, .CDF files, images, etc. It is also not uncommon for some RNA-seq or other sequencing datasets to supply <em>only</em> raw data (with accompanying sample information, of course), necessitating Sometimes, it is useful to get quick access to such data. A single function, <code>getGEOSuppFiles</code>, can take as an argument a GEO accession and will download all the raw data associate with that accession. By default, the function will create a directory in the current working directory to store the raw data for the chosen GEO accession.</p>
</div>
</div>
<div id="genomicdatacommons" class="section level2">
<h2 class="hasAnchor">
<a href="#genomicdatacommons" class="anchor"></a>GenomicDataCommons</h2>
<p>From the <a href="https://gdc.nci.nih.gov/about-gdc">Genomic Data Commons (GDC) website</a>:</p>
<blockquote>
<p>The National Cancer Institute’s (NCI’s) Genomic Data Commons (GDC) is a data sharing platform that promotes precision medicine in oncology. It is not just a database or a tool; it is an expandable knowledge network supporting the import and standardization of genomic and clinical data from cancer research programs. The GDC contains NCI-generated data from some of the largest and most comprehensive cancer genomic datasets, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Therapies (TARGET). For the first time, these datasets have been harmonized using a common set of bioinformatics pipelines, so that the data can be directly compared. As a growing knowledge system for cancer, the GDC also enables researchers to submit data, and harmonizes these data for import into the GDC. As more researchers add clinical and genomic data to the GDC, it will become an even more powerful tool for making discoveries about the molecular basis of cancer that may lead to better care for patients.</p>
</blockquote>
<p>The <a href="https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components">data model for the GDC is complex</a>, but it worth a quick overview and a graphical representation is included here.</p>
<div class="figure">
<img src="figures/all_nodes_040318.png" alt=""><p class="caption">The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details.</p>
</div>
<p>The GDC API exposes these nodes and edges in a somewhat simplified set of <a href="https://en.wikipedia.org/wiki/Representational_state_transfer">RESTful</a> endpoints.</p>
<div id="quickstart" class="section level3">
<h3 class="hasAnchor">
<a href="#quickstart" class="anchor"></a>Quickstart</h3>
<p>This quickstart section is just meant to show basic functionality. More details of functionality are included further on in this vignette and in function-specific help.</p>
<p>To report bugs or problems, either <a href="https://github.com/Bioconductor/GenomicDataCommons/issues">submit a new issue</a> or submit a <code><a href="https://rdrr.io/r/utils/bug.report.html">bug.report(package='GenomicDataCommons')</a></code> from within R (which will redirect you to the new issue on GitHub).</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicDataCommons</span>)</pre></body></html></div>
<div id="check-connectivity-and-status" class="section level4">
<h4 class="hasAnchor">
<a href="#check-connectivity-and-status" class="anchor"></a>Check connectivity and status</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> package relies on having network connectivity. In addition, the NCI GDC API must also be operational and not under maintenance. Checking <code>status</code> can be used to check this connectivity and functionality.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="kw pkg">GenomicDataCommons</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/status.html">status</a></span>()</pre></body></html></div>
<pre><code>## $commit
## [1] "0137629ad19be1839669a25a741a165f90c7276c"
## 
## $data_release
## [1] "Data Release 25.0 - July 22, 2020"
## 
## $status
## [1] "OK"
## 
## $tag
## [1] "2.3.1"
## 
## $version
## [1] 1</code></pre>
</div>
<div id="find-data" class="section level4">
<h4 class="hasAnchor">
<a href="#find-data" class="anchor"></a>Find data</h4>
<p>The following code builds a <code>manifest</code> that can be used to guide the download of raw data. Here, filtering finds gene expression files quantified as raw counts using <code>HTSeq</code> from ovarian cancer patients.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">ge_manifest</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span>( ~ <span class="no">cases.project.project_id</span> <span class="kw">==</span> <span class="st">'TCGA-OV'</span> <span class="kw">&amp;</span>
                <span class="no">type</span> <span class="kw">==</span> <span class="st">'gene_expression'</span> <span class="kw">&amp;</span>
                <span class="no">analysis.workflow_type</span> <span class="kw">==</span> <span class="st">'HTSeq - Counts'</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span>()</pre></body></html></div>
</div>
<div id="download-data" class="section level4">
<h4 class="hasAnchor">
<a href="#download-data" class="anchor"></a>Download data</h4>
<p>After the 379 gene expression files specified in the query above. Using multiple processes to do the download very significantly speeds up the transfer in many cases. On a standard 1Gb connection, the following completes in about 30 seconds. The first time the data are downloaded, R will ask to create a cache directory (see <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_cache.html">?gdc_cache</a></code> for details of setting and interacting with the cache). Resulting downloaded files will be stored in the cache directory. Future access to the same files will be directly from the cache, alleviating multiple downloads.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">fnames</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">ge_manifest</span>$<span class="no">id</span>[<span class="fl">1</span>:<span class="fl">20</span>],<span class="no">gdcdata</span>)</pre></body></html></div>
<p>If the download had included controlled-access data, the download above would have needed to include a <code>token</code>. Details are available in <a href="#authentication">the authentication section below</a>.</p>
</div>
<div id="metadata-queries" class="section level4">
<h4 class="hasAnchor">
<a href="#metadata-queries" class="anchor"></a>Metadata queries</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> can access the significant clinical, demographic, biospecimen, and annotation information contained in the NCI GDC.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">expands</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"diagnoses"</span>,<span class="st">"annotations"</span>,
             <span class="st">"demographic"</span>,<span class="st">"exposures"</span>)
<span class="no">projResults</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results</a></span>(<span class="kw">size</span><span class="kw">=</span><span class="fl">10</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">projResults</span>,<span class="kw">list.len</span><span class="kw">=</span><span class="fl">5</span>)</pre></body></html></div>
<pre><code>## List of 9
##  $ dbgap_accession_number: logi [1:10] NA NA NA NA NA NA ...
##  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...
##  $ id                    : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-ACC" "TCGA-LGG" ...
##  $ disease_type          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Germ Cell Neoplasms" "Granular Cell Tumors and Alveolar Soft Part Sarcomas" "Immunoproliferative Diseases" "Plasma Cell Tumors" ...
##   ..$ TCGA-UCEC : chr [1:4] "Epithelial Neoplasms, NOS" "Cystic, Mucinous and Serous Neoplasms" "Adenomas and Adenocarcinomas" "Not Reported"
##   ..$ TCGA-ACC  : chr "Adenomas and Adenocarcinomas"
##   ..$ TCGA-LGG  : chr "Gliomas"
##   ..$ TCGA-SARC : chr [1:6] "Nerve Sheath Tumors" "Myomatous Neoplasms" "Fibromatous Neoplasms" "Lipomatous Neoplasms" ...
##   .. [list output truncated]
##  $ state                 : chr [1:10] "open" "open" "open" "open" ...
##   [list output truncated]
##  - attr(*, "row.names")= int [1:10] 1 2 3 4 5 6 7 8 9 10
##  - attr(*, "class")= chr [1:3] "GDCprojectsResults" "GDCResults" "list"</code></pre>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">projResults</span>)</pre></body></html></div>
<pre><code>## [1] "dbgap_accession_number" "released"               "id"                    
## [4] "disease_type"           "state"                  "name"                  
## [7] "releasable"             "project_id"             "primary_site"</code></pre>
<div class="sourceCode" id="cb21"><html><body><pre class="r"># or listviewer::jsonedit(clinResults)</pre></body></html></div>
</div>
</div>
<div id="basic-design" class="section level3">
<h3 class="hasAnchor">
<a href="#basic-design" class="anchor"></a>Basic design</h3>
<p>This package design is meant to have some similarities to the “hadleyverse” approach of dplyr. Roughly, the functionality for finding and accessing files and metadata can be divided into:</p>
<ol style="list-style-type: decimal">
<li>Simple query constructors based on GDC API endpoints.</li>
<li>A set of verbs that when applied, adjust filtering, field selection, and faceting (fields for aggregation) and result in a new query object (an endomorphism)</li>
<li>A set of verbs that take a query and return results from the GDC</li>
</ol>
<p>In addition, there are exhiliary functions for asking the GDC API for information about available and default fields, slicing BAM files, and downloading actual data files. Here is an overview of functionality<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<ul>
<li>Creating a query
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">annotations()</a></code></li>
</ul>
</li>
<li>Manipulating a query
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code></li>
</ul>
</li>
<li>Introspection on the GDC API fields
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/mapping.html">mapping()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/grep_fields.html">grep_fields()</a></code></li>
<li><code>field_picker()</code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_values.html">available_values()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_expand.html">available_expand()</a></code></li>
</ul>
</li>
<li>Executing an API call to retrieve query results
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/response.html">response()</a></code></li>
</ul>
</li>
<li>Raw data file downloads
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_client.html">gdc_client()</a></code></li>
</ul>
</li>
<li>Summarizing and aggregating field values (faceting)
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code></li>
</ul>
</li>
<li>Authentication
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_token.html">gdc_token()</a></code></li>
</ul>
</li>
<li>BAM file slicing
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/slicing.html">slicing()</a></code></li>
</ul>
</li>
</ul>
</div>
<div id="usage" class="section level3">
<h3 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h3>
<p>There are two main classes of operations when working with the NCI GDC.</p>
<ol style="list-style-type: decimal">
<li>
<a href="#querying-metadata">Querying metadata and finding data files</a> (e.g., finding all gene expression quantifications data files for all colon cancer patients).</li>
<li>
<a href="#datafile-access-and-download">Transferring raw or processed data</a> from the GDC to another computer (e.g., downloading raw or processed data)</li>
</ol>
<p>Both classes of operation are reviewed in detail in the following sections.</p>
</div>
</div>
<div id="querying-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#querying-metadata" class="anchor"></a>Querying metadata</h2>
<p>Vast amounts of metadata about cases (patients, basically), files, projects, and so-called annotations are available via the NCI GDC API. Typically, one will want to query metadata to either focus in on a set of files for download or transfer <em>or</em> to perform so-called aggregations (pivot-tables, facets, similar to the R <code><a href="https://rdrr.io/r/base/table.html">table()</a></code> functionality).</p>
<p>Querying metadata starts with <a href="#creating-a-query">creating a “blank” query</a>. One will often then want to <a href="#filtering"><code>filter</code></a> the query to limit results prior to <a href="#retrieving-results">retrieving results</a>. The GenomicDataCommons package has <a href="#fields-and-values">helper functions for listing fields</a> that are available for filtering.</p>
<p>In addition to fetching results, the GDC API allows <a href="#facets-and-aggregation">faceting, or aggregating,</a>, useful for compiling reports, generating dashboards, or building user interfaces to GDC data (see GDC web query interface for a non-R-based example).</p>
<div id="creating-a-query" class="section level4">
<h4 class="hasAnchor">
<a href="#creating-a-query" class="anchor"></a>Creating a query</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em> package accesses the <em>same</em> API as the <em>GDC</em> website. Therefore, a useful approach, particularly for beginning users is to examine the filters available on the <a href="https://portal.gdc.cancer.gov/repository">GDC repository pages</a> to find appropriate filtering criteria. From there, converting those checkboxes to a GenomicDataCommons <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">query()</a></code> is relatively straightforward. Note that only a small subset of the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code> are available by default on the website.</p>
<div class="figure">
<img src="figures/gdc_filtering.png" alt=""><p class="caption">A screenshot of an example query of the GDC repository portal.</p>
</div>
<p>A query of the GDC starts its life in R. Queries follow the four metadata endpoints available at the GDC. In particular, there are four convenience functions that each create <code>GDCQuery</code> objects (actually, specific subclasses of <code>GDCQuery</code>):</p>
<ul>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">annotations()</a></code></li>
</ul>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">pquery</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects</a></span>()</pre></body></html></div>
<p>The <code>pquery</code> object is now an object of (S3) class, <code>GDCQuery</code> (and <code>gdc_projects</code> and <code>list</code>). The object contains the following elements:</p>
<ul>
<li>fields: This is a character vector of the fields that will be returned when we <a href="#retrieving-results">retrieve data</a>. If no fields are specified to, for example, the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">projects()</a></code> function, the default fields from the GDC are used (see <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code>)</li>
<li>filters: This will contain results after calling the <a href="#filtering"><code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter()</a></code> method</a> and will be used to filter results on <a href="#retrieving-results">retrieval</a>.</li>
<li>facets: A character vector of field names that will be used for <a href="#facets-and-aggregation">aggregating data</a> in a call to <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code>.</li>
<li>archive: One of either “default” or <a href="https://gdc-portal.nci.nih.gov/legacy-archive/">“legacy”</a>.</li>
<li>token: A character(1) token from the GDC. See <a href="#authentication">the authentication section</a> for details, but note that, in general, the token is not necessary for metadata query and retrieval, only for actual data download.</li>
</ul>
<p>Looking at the actual object (get used to using <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code>!), note that the query contains no results.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">pquery</span>)</pre></body></html></div>
<pre><code>## List of 5
##  $ fields : chr [1:10] "dbgap_accession_number" "disease_type" "intended_release_date" "name" ...
##  $ filters: NULL
##  $ facets : NULL
##  $ legacy : logi FALSE
##  $ expand : NULL
##  - attr(*, "class")= chr [1:3] "gdc_projects" "GDCQuery" "list"</code></pre>
</div>
<div id="retrieving-results" class="section level4">
<h4 class="hasAnchor">
<a href="#retrieving-results" class="anchor"></a>Retrieving results</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#size-and-from">[ GDC pagination documentation ]</a></p>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#sort">[ GDC sorting documentation ]</a></p>
<p>With a query object available, the next step is to retrieve results from the GDC. The GenomicDataCommons package. The most basic type of results we can get is a simple <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> of records available that satisfy the filter criteria. Note that we have not set any filters, so a <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> here will represent all the project records publicly available at the GDC in the “default” archive"</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">pcount</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span>(<span class="no">pquery</span>)
<span class="co"># or</span>
<span class="no">pcount</span> <span class="kw">=</span> <span class="no">pquery</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span>()
<span class="no">pcount</span></pre></body></html></div>
<pre><code>## [1] 65</code></pre>
<p>The <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> method will fetch actual results.</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="no">presults</span> <span class="kw">=</span> <span class="no">pquery</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results</a></span>()</pre></body></html></div>
<p>These results are returned from the GDC in <a href="http://www.json.org/">JSON</a> format and converted into a (potentially nested) list in R. The <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code> method is useful for taking a quick glimpse of the data.</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">presults</span>)</pre></body></html></div>
<pre><code>## List of 9
##  $ id                    : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-ACC" "TCGA-LGG" ...
##  $ primary_site          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Testis" "Gallbladder" "Unknown" "Other and unspecified parts of biliary tract" ...
##   ..$ TCGA-UCEC : chr [1:2] "Corpus uteri" "Uterus, NOS"
##   ..$ TCGA-ACC  : chr "Adrenal gland"
##   ..$ TCGA-LGG  : chr "Brain"
##   ..$ TCGA-SARC : chr [1:13] "Kidney" "Other and unspecified parts of tongue" "Bones, joints and articular cartilage of limbs" "Colon" ...
##   ..$ TCGA-PAAD : chr "Pancreas"
##   ..$ TCGA-ESCA : chr [1:2] "Esophagus" "Stomach"
##   ..$ TCGA-LUAD : chr "Bronchus and lung"
##   ..$ TCGA-PRAD : chr "Prostate gland"
##   ..$ GENIE-VICC: chr [1:46] "Testis" "Unknown" "Other and unspecified parts of biliary tract" "Adrenal gland" ...
##  $ state                 : chr [1:10] "open" "open" "open" "open" ...
##  $ dbgap_accession_number: logi [1:10] NA NA NA NA NA NA ...
##  $ disease_type          :List of 10
##   ..$ GENIE-MSK : chr [1:49] "Germ Cell Neoplasms" "Granular Cell Tumors and Alveolar Soft Part Sarcomas" "Immunoproliferative Diseases" "Plasma Cell Tumors" ...
##   ..$ TCGA-UCEC : chr [1:4] "Epithelial Neoplasms, NOS" "Cystic, Mucinous and Serous Neoplasms" "Adenomas and Adenocarcinomas" "Not Reported"
##   ..$ TCGA-ACC  : chr "Adenomas and Adenocarcinomas"
##   ..$ TCGA-LGG  : chr "Gliomas"
##   ..$ TCGA-SARC : chr [1:6] "Nerve Sheath Tumors" "Myomatous Neoplasms" "Fibromatous Neoplasms" "Lipomatous Neoplasms" ...
##   ..$ TCGA-PAAD : chr [1:4] "Cystic, Mucinous and Serous Neoplasms" "Ductal and Lobular Neoplasms" "Adenomas and Adenocarcinomas" "Epithelial Neoplasms, NOS"
##   ..$ TCGA-ESCA : chr [1:3] "Cystic, Mucinous and Serous Neoplasms" "Squamous Cell Neoplasms" "Adenomas and Adenocarcinomas"
##   ..$ TCGA-LUAD : chr [1:3] "Cystic, Mucinous and Serous Neoplasms" "Acinar Cell Neoplasms" "Adenomas and Adenocarcinomas"
##   ..$ TCGA-PRAD : chr [1:3] "Cystic, Mucinous and Serous Neoplasms" "Ductal and Lobular Neoplasms" "Adenomas and Adenocarcinomas"
##   ..$ GENIE-VICC: chr [1:41] "Germ Cell Neoplasms" "Acinar Cell Neoplasms" "Synovial-like Neoplasms" "Plasma Cell Tumors" ...
##  $ name                  : chr [1:10] "AACR Project GENIE - Contributed by Memorial Sloan Kettering Cancer Center" "Uterine Corpus Endometrial Carcinoma" "Adrenocortical Carcinoma" "Brain Lower Grade Glioma" ...
##  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...
##  $ releasable            : logi [1:10] FALSE TRUE TRUE TRUE TRUE TRUE ...
##  $ project_id            : chr [1:10] "GENIE-MSK" "TCGA-UCEC" "TCGA-ACC" "TCGA-LGG" ...
##  - attr(*, "row.names")= int [1:10] 1 2 3 4 5 6 7 8 9 10
##  - attr(*, "class")= chr [1:3] "GDCprojectsResults" "GDCResults" "list"</code></pre>
<p>A default of only 10 records are returned. We can use the <code>size</code> and <code>from</code> arguments to <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> to either page through results or to change the number of results. Finally, there is a convenience method, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code> that will simply fetch all the available results given a query. Note that <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code> may take a long time and return HUGE result sets if not used carefully. Use of a combination of <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> and <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results.html">results()</a></code> to get a sense of the expected data size is probably warranted before calling <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all()</a></code></p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span>(<span class="no">presults</span>))</pre></body></html></div>
<pre><code>## [1] 10</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">presults</span> <span class="kw">=</span> <span class="no">pquery</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/results_all.html">results_all</a></span>()
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span>(<span class="no">presults</span>))</pre></body></html></div>
<pre><code>## [1] 65</code></pre>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="co"># includes all records</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/ids.html">ids</a></span>(<span class="no">presults</span>)) <span class="kw">==</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span>(<span class="no">pquery</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>Extracting subsets of results or manipulating the results into a more conventional R data structure is not easily generalizable. However, the <a href="https://github.com/hadley/purrr">purrr</a>, <a href="https://renkun.me/rlist/">rlist</a>, and <a href="https://cran.r-project.org/web/packages/data.tree/vignettes/data.tree.html">data.tree</a> packages are all potentially of interest for manipulating complex, nested list structures. For viewing the results in an interactive viewer, consider the <a href="https://github.com/timelyportfolio/listviewer">listviewer</a> package.</p>
</div>
<div id="fields-and-values" class="section level4">
<h4 class="hasAnchor">
<a href="#fields-and-values" class="anchor"></a>Fields and Values</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#fields">[ GDC <code>fields</code> documentation ]</a></p>
<p>Central to querying and retrieving data from the GDC is the ability to specify which fields to return, filtering by fields and values, and faceting or aggregating. The GenomicDataCommons package includes two simple functions, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields()</a></code> and <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields()</a></code>. Each can operate on a character(1) endpoint name (“cases”, “files”, “annotations”, or “projects”) or a <code>GDCQuery</code> object.</p>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/default_fields.html">default_fields</a></span>(<span class="st">'files'</span>)</pre></body></html></div>
<pre><code>##  [1] "access"                         "acl"                           
##  [3] "average_base_quality"           "average_insert_size"           
##  [5] "average_read_length"            "channel"                       
##  [7] "chip_id"                        "chip_position"                 
##  [9] "contamination"                  "contamination_error"           
## [11] "created_datetime"               "data_category"                 
## [13] "data_format"                    "data_type"                     
## [15] "error_type"                     "experimental_strategy"         
## [17] "file_autocomplete"              "file_id"                       
## [19] "file_name"                      "file_size"                     
## [21] "imaging_date"                   "magnification"                 
## [23] "md5sum"                         "mean_coverage"                 
## [25] "msi_score"                      "msi_status"                    
## [27] "origin"                         "pairs_on_diff_chr"             
## [29] "plate_name"                     "plate_well"                    
## [31] "platform"                       "proportion_base_mismatch"      
## [33] "proportion_coverage_10x"        "proportion_coverage_10X"       
## [35] "proportion_coverage_30x"        "proportion_coverage_30X"       
## [37] "proportion_reads_duplicated"    "proportion_reads_mapped"       
## [39] "proportion_targets_no_coverage" "read_pair_number"              
## [41] "revision"                       "stain_type"                    
## [43] "state"                          "state_comment"                 
## [45] "submitter_id"                   "tags"                          
## [47] "total_reads"                    "tumor_ploidy"                  
## [49] "tumor_purity"                   "type"                          
## [51] "updated_datetime"</code></pre>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="co"># The number of fields available for files endpoint</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span>(<span class="st">'files'</span>))</pre></body></html></div>
<pre><code>## [1] 945</code></pre>
<div class="sourceCode" id="cb40"><html><body><pre class="r"><span class="co"># The first few fields available for files endpoint</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span>(<span class="st">'files'</span>))</pre></body></html></div>
<pre><code>## [1] "access"                      "acl"                        
## [3] "analysis.analysis_id"        "analysis.analysis_type"     
## [5] "analysis.created_datetime"   "analysis.input_files.access"</code></pre>
<p>The fields to be returned by a query can be specified following a similar paradigm to that of the dplyr package. The <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> function is a verb that resets the fields slot of a <code>GDCQuery</code>; note that this is not quite analogous to the dplyr <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> verb that limits from already-present fields. We <em>completely replace</em> the fields when using <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select()</a></code> on a <code>GDCQuery</code>.</p>
<div class="sourceCode" id="cb42"><html><body><pre class="r"><span class="co"># Default fields here</span>
<span class="no">qcases</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases</a></span>()
<span class="no">qcases</span>$<span class="no">fields</span></pre></body></html></div>
<pre><code>##  [1] "aliquot_ids"              "analyte_ids"             
##  [3] "case_autocomplete"        "case_id"                 
##  [5] "consent_type"             "created_datetime"        
##  [7] "days_to_consent"          "days_to_index"           
##  [9] "days_to_lost_to_followup" "diagnosis_ids"           
## [11] "disease_type"             "index_date"              
## [13] "lost_to_followup"         "portion_ids"             
## [15] "primary_site"             "sample_ids"              
## [17] "slide_ids"                "state"                   
## [19] "submitter_aliquot_ids"    "submitter_analyte_ids"   
## [21] "submitter_diagnosis_ids"  "submitter_id"            
## [23] "submitter_portion_ids"    "submitter_sample_ids"    
## [25] "submitter_slide_ids"      "updated_datetime"</code></pre>
<div class="sourceCode" id="cb44"><html><body><pre class="r"><span class="co"># set up query to use ALL available fields</span>
<span class="co"># Note that checking of fields is done by select()</span>
<span class="no">qcases</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">cases</a></span>() <span class="kw">%&gt;%</span> <span class="kw pkg">GenomicDataCommons</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/select.html">select</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span>(<span class="st">'cases'</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">qcases</span>$<span class="no">fields</span>)</pre></body></html></div>
<pre><code>## [1] "case_id"                       "aliquot_ids"                  
## [3] "analyte_ids"                   "annotations.annotation_id"    
## [5] "annotations.case_id"           "annotations.case_submitter_id"</code></pre>
<p>Finding fields of interest is such a common operation that the GenomicDataCommons includes the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/grep_fields.html">grep_fields()</a></code> function and the <code>field_picker()</code> widget. See the appropriate help pages for details.</p>
</div>
<div id="facets-and-aggregation" class="section level4">
<h4 class="hasAnchor">
<a href="#facets-and-aggregation" class="anchor"></a>Facets and aggregation</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC <code>facet</code> documentation ]</a></p>
<p>The GDC API offers a feature known as aggregation or faceting. By specifying one or more fields (of appropriate type), the GDC can return to us a count of the number of records matching each potential value. This is similar to the R <code>table</code> method. Multiple fields can be returned at once, but the GDC API does not have a cross-tabulation feature; all aggregations are only on one field at a time. Results of <code>aggregation()</code> calls come back as a list of data.frames (actually, tibbles).</p>
<div class="sourceCode" id="cb46"><html><body><pre class="r"><span class="co"># total number of files of a specific type</span>
<span class="no">res</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'type'</span>,<span class="st">'data_type'</span>)) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations</a></span>()
<span class="no">res</span>$<span class="no">type</span></pre></body></html></div>
<pre><code>##    doc_count                         key
## 1     142950  annotated_somatic_mutation
## 2      81911     simple_somatic_mutation
## 3      78009               aligned_reads
## 4      58290             gene_expression
## 5      56840         copy_number_segment
## 6      44143        copy_number_estimate
## 7      30072                 slide_image
## 8      29996            mirna_expression
## 9      25510      biospecimen_supplement
## 10     12870         clinical_supplement
## 11     12359      methylation_beta_value
## 12      5426        structural_variation
## 13      3547 aggregated_somatic_mutation
## 14      3505     masked_somatic_mutation</code></pre>
<p>Using <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations()</a></code> is an also easy way to learn the contents of individual fields and forms the basis for faceted search pages.</p>
</div>
<div id="filtering" class="section level4">
<h4 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query">[ GDC <code>filtering</code> documentation ]</a></p>
<p>The GenomicDataCommons package uses a form of non-standard evaluation to specify R-like queries that are then translated into an R list. That R list is, upon calling a method that fetches results from the GDC API, translated into the appropriate JSON string. The R expression uses the formula interface as suggested by Hadley Wickham in his <a href="https://cran.r-project.org/web/packages/dplyr/vignettes/nse.html">vignette on non-standard evaluation</a></p>
<blockquote>
<p>It’s best to use a formula because a formula captures both the expression to evaluate and the environment where the evaluation occurs. This is important if the expression is a mixture of variables in a data frame and objects in the local environment [for example].</p>
</blockquote>
<p>For the user, these details will not be too important except to note that a filter expression must begin with a “~”.</p>
<div class="sourceCode" id="cb48"><html><body><pre class="r"><span class="no">qfiles</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>()
<span class="no">qfiles</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span>() <span class="co"># all files</span></pre></body></html></div>
<pre><code>## [1] 585428</code></pre>
<p>To limit the file type, we can refer back to the <a href="#facets-and-aggregation">section on faceting</a> to see the possible values for the file field “type”. For example, to filter file results to only “gene_expression” files, we simply specify a filter.</p>
<div class="sourceCode" id="cb50"><html><body><pre class="r"><span class="no">qfiles</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span>(~ <span class="no">type</span> <span class="kw">==</span> <span class="st">'gene_expression'</span>)
<span class="co"># here is what the filter looks like after translation</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">get_filter</a></span>(<span class="no">qfiles</span>))</pre></body></html></div>
<pre><code>## List of 2
##  $ op     : 'scalar' chr "="
##  $ content:List of 2
##   ..$ field: chr "type"
##   ..$ value: chr "gene_expression"</code></pre>
<p>What if we want to create a filter based on the project (‘TCGA-OVCA’, for example)? Well, we have a couple of possible ways to discover available fields. The first is based on base R functionality and some intuition.</p>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">'pro'</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/available_fields.html">available_fields</a></span>(<span class="st">'files'</span>), <span class="kw">value</span><span class="kw">=</span><span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>##  [1] "analysis.input_files.proportion_base_mismatch"                  
##  [2] "analysis.input_files.proportion_coverage_10x"                   
##  [3] "analysis.input_files.proportion_coverage_10X"                   
##  [4] "analysis.input_files.proportion_coverage_30x"                   
##  [5] "analysis.input_files.proportion_coverage_30X"                   
##  [6] "analysis.input_files.proportion_reads_duplicated"               
##  [7] "analysis.input_files.proportion_reads_mapped"                   
##  [8] "analysis.input_files.proportion_targets_no_coverage"            
##  [9] "cases.diagnoses.international_prognostic_index"                 
## [10] "cases.diagnoses.progression_or_recurrence"                      
## [11] "cases.follow_ups.days_to_progression"                           
## [12] "cases.follow_ups.days_to_progression_free"                      
## [13] "cases.follow_ups.procedures_performed"                          
## [14] "cases.follow_ups.progression_or_recurrence"                     
## [15] "cases.follow_ups.progression_or_recurrence_anatomic_site"       
## [16] "cases.follow_ups.progression_or_recurrence_type"                
## [17] "cases.project.dbgap_accession_number"                           
## [18] "cases.project.disease_type"                                     
## [19] "cases.project.intended_release_date"                            
## [20] "cases.project.name"                                             
## [21] "cases.project.primary_site"                                     
## [22] "cases.project.program.dbgap_accession_number"                   
## [23] "cases.project.program.name"                                     
## [24] "cases.project.program.program_id"                               
## [25] "cases.project.project_id"                                       
## [26] "cases.project.releasable"                                       
## [27] "cases.project.released"                                         
## [28] "cases.project.state"                                            
## [29] "cases.samples.days_to_sample_procurement"                       
## [30] "cases.samples.method_of_sample_procurement"                     
## [31] "cases.samples.portions.slides.number_proliferating_cells"       
## [32] "cases.samples.portions.slides.prostatic_chips_positive_count"   
## [33] "cases.samples.portions.slides.prostatic_chips_total_count"      
## [34] "cases.samples.portions.slides.prostatic_involvement_percent"    
## [35] "cases.tissue_source_site.project"                               
## [36] "downstream_analyses.output_files.proportion_base_mismatch"      
## [37] "downstream_analyses.output_files.proportion_coverage_10x"       
## [38] "downstream_analyses.output_files.proportion_coverage_10X"       
## [39] "downstream_analyses.output_files.proportion_coverage_30x"       
## [40] "downstream_analyses.output_files.proportion_coverage_30X"       
## [41] "downstream_analyses.output_files.proportion_reads_duplicated"   
## [42] "downstream_analyses.output_files.proportion_reads_mapped"       
## [43] "downstream_analyses.output_files.proportion_targets_no_coverage"
## [44] "index_files.proportion_base_mismatch"                           
## [45] "index_files.proportion_coverage_10x"                            
## [46] "index_files.proportion_coverage_10X"                            
## [47] "index_files.proportion_coverage_30x"                            
## [48] "index_files.proportion_coverage_30X"                            
## [49] "index_files.proportion_reads_duplicated"                        
## [50] "index_files.proportion_reads_mapped"                            
## [51] "index_files.proportion_targets_no_coverage"                     
## [52] "proportion_base_mismatch"                                       
## [53] "proportion_coverage_10x"                                        
## [54] "proportion_coverage_10X"                                        
## [55] "proportion_coverage_30x"                                        
## [56] "proportion_coverage_30X"                                        
## [57] "proportion_reads_duplicated"                                    
## [58] "proportion_reads_mapped"                                        
## [59] "proportion_targets_no_coverage"</code></pre>
<p>Interestingly, the project information is “nested” inside the case. We don’t need to know that detail other than to know that we now have a few potential guesses for where our information might be in the files records. We need to know where because we need to construct the appropriate filter.</p>
<div class="sourceCode" id="cb54"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/faceting.html">facet</a></span>(<span class="st">'cases.project.project_id'</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/aggregations.html">aggregations</a></span>()</pre></body></html></div>
<pre><code>## $cases.project.project_id
##                      key doc_count
## 1                  FM-AD     36134
## 2              TCGA-BRCA     33766
## 3              GENIE-MSK     36470
## 4              TCGA-LUAD     18162
## 5              TCGA-UCEC     17277
## 6              TCGA-HNSC     16340
## 7                TCGA-OV     16344
## 8              TCGA-THCA     15445
## 9          MMRF-COMMPASS     29433
## 10             TCGA-LUSC     16368
## 11              TCGA-LGG     15795
## 12            GENIE-DFCI     28464
## 13             TCGA-KIRC     16255
## 14             TCGA-PRAD     15296
## 15             TCGA-COAD     15338
## 16              TCGA-GBM     13089
## 17         TARGET-ALL-P2     20772
## 18             TCGA-SKCM     13674
## 19             TCGA-STAD     13739
## 20               CPTAC-3     25005
## 21             TCGA-BLCA     12513
## 22             TCGA-LIHC     11578
## 23             TCGA-CESC      9201
## 24             TCGA-KIRP      9137
## 25             TCGA-SARC      8002
## 26            TARGET-AML      7772
## 27             TCGA-PAAD      5671
## 28             TCGA-ESCA      5657
## 29             TCGA-PCPG      5378
## 30               CPTAC-2      9978
## 31             TCGA-READ      5269
## 32            TARGET-NBL      5796
## 33     BEATAML1.0-COHORT      8981
## 34             TCGA-TGCT      4605
## 35             TCGA-LAML      4814
## 36             TCGA-THYM      3691
## 37              TCGA-ACC      2736
## 38             TCGA-KICH      2457
## 39             TARGET-WT      2677
## 40          NCICCR-DLBCL      4805
## 41             TCGA-MESO      2518
## 42              TCGA-UVM      2340
## 43             TARGET-OS      3113
## 44         TARGET-ALL-P3      3982
## 45             GENIE-MDA      3857
## 46            GENIE-VICC      3833
## 47             GENIE-JHU      3320
## 48              TCGA-UCS      1765
## 49             TCGA-CHOL      1426
## 50             GENIE-UHN      2632
## 51             TCGA-DLBC      1325
## 52         CGCI-HTMCP-CC      1968
## 53            CGCI-BLGSP      1782
## 54             TARGET-RT      1049
## 55             HCMI-CMDC      1337
## 56            GENIE-GRCC      1038
## 57            WCDT-MCRPC       994
## 58             GENIE-NKI       801
## 59              OHSU-CNL       798
## 60   ORGANOID-PANCREATIC       703
## 61           CTSP-DLBCL1       417
## 62 BEATAML1.0-CRENOLANIB       223
## 63           TARGET-CCSK       169
## 64         TARGET-ALL-P1       133
## 65        VAREPOP-APOLLO        21</code></pre>
<p>We note that <code>cases.project.project_id</code> looks like it is a good fit. We also note that <code>TCGA-OV</code> is the correct project_id, not <code>TCGA-OVCA</code>. Note that <em>unlike with dplyr and friends, the <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter()</a></code> method here <strong>replaces</strong> the filter and does not build on any previous filters</em>.</p>
<div class="sourceCode" id="cb56"><html><body><pre class="r"><span class="no">qfiles</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span>( ~ <span class="no">cases.project.project_id</span> <span class="kw">==</span> <span class="st">'TCGA-OV'</span> <span class="kw">&amp;</span> <span class="no">type</span> <span class="kw">==</span> <span class="st">'gene_expression'</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">get_filter</a></span>(<span class="no">qfiles</span>))</pre></body></html></div>
<pre><code>## List of 2
##  $ op     : 'scalar' chr "and"
##  $ content:List of 2
##   ..$ :List of 2
##   .. ..$ op     : 'scalar' chr "="
##   .. ..$ content:List of 2
##   .. .. ..$ field: chr "cases.project.project_id"
##   .. .. ..$ value: chr "TCGA-OV"
##   ..$ :List of 2
##   .. ..$ op     : 'scalar' chr "="
##   .. ..$ content:List of 2
##   .. .. ..$ field: chr "type"
##   .. .. ..$ value: chr "gene_expression"</code></pre>
<div class="sourceCode" id="cb58"><html><body><pre class="r"><span class="no">qfiles</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count</a></span>()</pre></body></html></div>
<pre><code>## [1] 1137</code></pre>
<p>Asking for a <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/count.html">count()</a></code> of results given these new filter criteria gives <code>r qfiles %&gt;% count()</code> results. Generating a manifest for bulk downloads is as simple as asking for the manifest from the current query.</p>
<div class="sourceCode" id="cb60"><html><body><pre class="r"><span class="no">manifest_df</span> <span class="kw">=</span> <span class="no">qfiles</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span>()
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">manifest_df</span>)</pre></body></html></div>
<pre><code>## # A tibble: 6 x 5
##   id                  filename                    md5                size state 
##   &lt;chr&gt;               &lt;chr&gt;                       &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt; 
## 1 15466fa6-cfd8-4c8d… 1e0f7f1f-e4aa-43e6-a6c1-bc… 0f27753af670df5… 556114 relea…
## 2 14859d69-d6bd-4415… 655c5a8b-a87a-482e-95cf-15… de20d380965fbe9… 547656 relea…
## 3 3c03ea30-96aa-4d8c… 4b449f26-ee8f-47d1-9d93-de… ed6aea70a3e9c19… 535690 relea…
## 4 8ea31edd-b2c9-4ff7… 451aa144-4d34-4858-b81e-34… 8023f9a07fc7b80… 552054 relea…
## 5 f37c892c-8440-4498… bd727430-2539-4897-a754-2c… d2fb61d663f1024… 563204 relea…
## 6 a1bacdbd-60f6-472f… 4722ba18-27f6-480d-975b-59… e6835e5ac38301a… 264266 relea…</code></pre>
<p>Note that we might still not be quite there. Looking at filenames, there are suspiciously named files that might include “FPKM”, “FPKM-UQ”, or “counts”. Another round of <code>grep</code> and <code>available_fields</code>, looking for “type” turned up that the field “analysis.workflow_type” has the appropriate filter criteria.</p>
<div class="sourceCode" id="cb62"><html><body><pre class="r"><span class="no">qfiles</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>() <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span>( ~ <span class="no">cases.project.project_id</span> <span class="kw">==</span> <span class="st">'TCGA-OV'</span> <span class="kw">&amp;</span>
                            <span class="no">type</span> <span class="kw">==</span> <span class="st">'gene_expression'</span> <span class="kw">&amp;</span>
                            <span class="no">analysis.workflow_type</span> <span class="kw">==</span> <span class="st">'HTSeq - Counts'</span>)
<span class="no">manifest_df</span> <span class="kw">=</span> <span class="no">qfiles</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span>()
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">manifest_df</span>)</pre></body></html></div>
<pre><code>## [1] 379</code></pre>
<p>The GDC Data Transfer Tool can be used (from R, <code><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer()</a></code> or from the command-line) to orchestrate high-performance, restartable transfers of all the files in the manifest. See <a href="bulk-downloads">the bulk downloads section</a> for details.</p>
</div>
<div id="authentication" class="section level3">
<h3 class="hasAnchor">
<a href="#authentication" class="anchor"></a>Authentication</h3>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC authentication documentation ]</a></p>
<p>The GDC offers both “controlled-access” and “open” data. As of this writing, only data stored as files is “controlled-access”; that is, metadata accessible via the GDC is all “open” data and some files are “open” and some are “controlled-access”. Controlled-access data are only available after <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data">going through the process of obtaining access.</a></p>
<p>After controlled-access to one or more datasets has been granted, logging into the GDC web portal will allow you to <a href="https://docs.gdc.cancer.gov/Data_Portal/Users_Guide/Authentication/#gdc-authentication-tokens">access a GDC authentication token</a>, which can be downloaded and then used to access available controlled-access data via the GenomicDataCommons package.</p>
<p>The GenomicDataCommons uses authentication tokens only for downloading data (see <code>transfer</code> and <code>gdcdata</code> documentation). The package includes a helper function, <code>gdc_token</code>, that looks for the token to be stored in one of three ways (resolved in this order):</p>
<ol style="list-style-type: decimal">
<li>As a string stored in the environment variable, <code>GDC_TOKEN</code>
</li>
<li>As a file, stored in the file named by the environment variable, <code>GDC_TOKEN_FILE</code>
</li>
<li>In a file in the user home directory, called <code>.gdc_token</code>
</li>
</ol>
<p>As a concrete example:</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="no">token</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdc_token.html">gdc_token</a></span>()
<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer</a></span>(<span class="no">...</span>,<span class="kw">token</span><span class="kw">=</span><span class="no">token</span>)
<span class="co"># or</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/transfer.html">transfer</a></span>(<span class="no">...</span>,<span class="kw">token</span><span class="kw">=</span><span class="fu">get_token</span>())</pre></body></html></div>
</div>
<div id="datafile-access-and-download" class="section level3">
<h3 class="hasAnchor">
<a href="#datafile-access-and-download" class="anchor"></a>Datafile access and download</h3>
<p>The <code>gdcdata</code> function takes a character vector of one or more file ids. A simple way of producing such a vector is to produce a <code>manifest</code> data frame and then pass in the first column, which will contain file ids.</p>
<div class="sourceCode" id="cb65"><html><body><pre class="r"><span class="no">fnames</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata</a></span>(<span class="no">manifest_df</span>$<span class="no">id</span>[<span class="fl">1</span>:<span class="fl">2</span>],<span class="kw">progress</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<p>Note that for controlled-access data, a GDC <a href="#authentication">authentication token</a> is required. Using the <code>BiocParallel</code> package may be useful for downloading in parallel, particularly for large numbers of smallish files.</p>
<p>The bulk download functionality is only efficient (as of v1.2.0 of the GDC Data Transfer Tool) for relatively large files, so use this approach only when transferring BAM files or larger VCF files, for example. Otherwise, consider using the approach shown above, perhaps in parallel.</p>
<div class="sourceCode" id="cb66"><html><body><pre class="r"><span class="no">fnames</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata</a></span>(<span class="no">manifest_df</span>$<span class="no">id</span>[<span class="fl">3</span>:<span class="fl">10</span>], <span class="kw">access_method</span> <span class="kw">=</span> <span class="st">'client'</span>)</pre></body></html></div>
</div>
</div>
<div id="pharmacogenomics" class="section level2">
<h2 class="hasAnchor">
<a href="#pharmacogenomics" class="anchor"></a>Pharmacogenomics</h2>
<p>Pharmacogenomics holds great promise for the development of biomarkers of drug response and the design of new therapeutic options, which are key challenges in precision medicine. However, such data are scattered and lack standards for efficient access and analysis, consequently preventing the realization of the full potential of pharmacogenomics. To address these issues, we implemented <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em>, an easy-to-use, open source package for integrative analysis of multiple pharmacogenomic datasets. <code>PharmacoGx</code> provides a unified framework for downloading and analyzing large pharmacogenomic datasets which are extensively curated to ensure maximum overlap and consistency.</p>
<p>Examples of <code>PharmacoGx</code> usage in biomedical research can be found in the following publications:</p>
<ul>
<li>
<a href="https://academic.oup.com/bioinformatics/article/32/8/1244/1744214/PharmacoGx-an-R-package-for-analysis-of-large">Smirnov et al. PharmacoGx: an R package for analysis of large pharmacogenomic datasets." Bioinformatics (2015): 1244-1246</a>.</li>
<li><a href="https://doi.org/10.1093/jamia/ocx062">Yao et al., Tissue specificity of in vitro drug sensitivity, JAMIA (2017)</a></li>
<li><a href="https://www.nature.com/articles/s41467-017-01153-8">Safikhani et al., Gene isoforms as expression-based biomarkers predictive of drug response in vitro, Nature Communications (2017)</a></li>
<li><a href="http://cancerres.aacrjournals.org/content/77/11/3057.long">El-Hachem et al., Integrative Cancer Pharmacogenomics to Infer Large-Scale Drug Taxonomy, Cancer Research (2017)</a></li>
<li><a href="https://cancerres.aacrjournals.org/content/78/17/5124">Singh, M. et al. Therapeutic Targeting of the Premetastatic Stage in Human Lung-to-Brain Metastasis. Cancer Research (2018).</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1006743">Knowles, D. A., Bouchard, G. &amp; Plevritis, S. Sparse discriminative latent characteristics for predicting cancer drug sensitivity from genomic features. PLOS Computational Biology (2019)</a></li>
<li><a href="doi:10.1038/s43018-019-0018-6">Corsello, S. M. et al. Discovering the anticancer potential of non-oncology drugs by systematic viability profiling, Nat Cancer (2020)</a></li>
</ul>
<div id="getting-started" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h3>
<p>Let us first load the <code>PharmacoGx</code> library.</p>
<div class="sourceCode" id="cb67"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">PharmacoGx</span>)</pre></body></html></div>
<pre><code>## Warning: no DISPLAY variable so Tk is not available</code></pre>
<p>We can now access large-scale preclinical pharmacogenomic datasets that have been fully curated for ease of use.</p>
</div>
<div id="overview-of-pharmacogx-datasets-pharmacosets" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-of-pharmacogx-datasets-pharmacosets" class="anchor"></a>Overview of PharmacoGx datasets (PharmacoSets)</h3>
<p>To efficiently store and analyze large pharmacogenomic datasets, we developed the <code>PharmacoSet</code> class (also referred to as <code>PSet</code>), which acts as a data container storing pharmacological and molecular data along with experimental metadata (detailed structure provided in Supplementary materials). This class enables efficient implementation of curated annotations for cell lines, drug compounds and molecular features, which facilitates comparisons between different datasets stored as <code>PharmacoSet</code> objects.</p>
<p>We have made the PharmacoSet objects of the curated datasets available for download using functions provided in the package. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function. Any of the PharmacoSets in the table can then be downloaded by calling <em>downloadPSet</em>, which saves the datasets into a directory of the users choice, and returns the data into the R session.</p>
<div class="figure">
<img src="figures/PharmacoSet_structure.png" alt=""><p class="caption">Overview of PharmacoGx</p>
</div>
<p>To get a list of all the available <code>PharmacoSets</code> in PharmacoGx, we can use the availablePSets` function, which returns a table providing key information for each dataset.</p>
<div class="sourceCode" id="cb69"><html><body><pre class="r">(<span class="no">psets</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html">availablePSets</a></span>()[,-<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">6</span>,<span class="fl">7</span>)])</pre></body></html></div>
<pre><code>##    Dataset Name             Date Created         PSet Name      version
## 1          GDSC 2020-06-24T14:39:26.588Z GDSC_2019(v2-8.0) 2019(v2-8.0)
## 2          FIMM 2020-06-24T14:39:26.588Z         FIMM_2016         2016
## 3        CTRPv2 2020-06-24T14:39:26.588Z       CTRPv2_2015         2015
## 4          CCLE 2020-06-24T14:39:26.588Z         CCLE_2015         2015
## 5          GDSC 2020-06-24T14:39:26.588Z GDSC_2019(v1-8.0) 2019(v1-8.0)
## 6          gCSI 2020-06-24T14:39:26.588Z         gCSI_2017         2017
## 7          GRAY 2020-06-24T14:39:26.587Z         GRAY_2013         2013
## 8     UHNBreast 2020-06-24T14:39:26.588Z    UHNBreast_2019         2019
## 9          GDSC 2020-06-24T14:39:26.588Z GDSC_2020(v2-8.2) 2020(v2-8.2)
## 10         GRAY 2020-06-24T14:39:26.588Z         GRAY_2017         2017
## 11         GDSC 2020-06-24T14:39:26.588Z GDSC_2020(v1-8.2) 2020(v1-8.2)
## 12         CMAP 2020-07-15T16:33:03.834Z         CMAP_2016         2016
##            type                    DOI
## 1   sensitivity 10.5281/zenodo.3905496
## 2   sensitivity 10.5281/zenodo.3905448
## 3   sensitivity 10.5281/zenodo.3905470
## 4   sensitivity 10.5281/zenodo.3905462
## 5   sensitivity 10.5281/zenodo.3905505
## 6   sensitivity 10.5281/zenodo.3905452
## 7   sensitivity 10.5281/zenodo.3905454
## 8          both 10.5281/zenodo.3905460
## 9   sensitivity 10.5281/zenodo.3905481
## 10  sensitivity 10.5281/zenodo.3905458
## 11  sensitivity 10.5281/zenodo.3905485
## 12 perturbation 10.5281/zenodo.3946534
##                                                            Download
## 1      https://zenodo.org/record/3905496/files/GDSC2.rds?download=1
## 2       https://zenodo.org/record/3905448/files/FIMM.rds?download=1
## 3     https://zenodo.org/record/3905470/files/CTRPv2.rds?download=1
## 4       https://zenodo.org/record/3905462/files/CCLE.rds?download=1
## 5      https://zenodo.org/record/3905505/files/GDSC1.rds?download=1
## 6       https://zenodo.org/record/3905452/files/gCSI.rds?download=1
## 7       https://zenodo.org/record/3905454/files/GRAY.rds?download=1
## 8  https://zenodo.org/record/3905460/files/UHNBreast.rds?download=1
## 9      https://zenodo.org/record/3905481/files/GDSC2.rds?download=1
## 10      https://zenodo.org/record/3905458/files/GRAY.rds?download=1
## 11     https://zenodo.org/record/3905485/files/GDSC1.rds?download=1
## 12      https://zenodo.org/record/3946534/files/CMAP.rds?download=1</code></pre>
<p>There are currently 12 datasets available in PharmacoGx, including sensitivity datasets and perturbation datasets (see below).</p>
<div id="drug-sensitivity-datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#drug-sensitivity-datasets" class="anchor"></a>Drug Sensitivity Datasets</h4>
<p>Drug sensitivity datasets refer to pharmacogenomic data where cancer cells are molecularly profiled at baseline (before drug treatment), and the effect of drug treatment on cell viability is measured using a pharmacological assay (e.g., Cell Titer-Glo). These datasets can be used for biomarker discovery by correlating the molecular features of cancer cells to their response to drugs of interest.</p>
<div class="figure">
<img src="figures/Drug_sensitivity.png" alt=""><p class="caption">Schematic view of the drug sensitivity datasets.</p>
</div>
<p>Notably, the Genomics of Drug Sensitivity in Cancer <a href="https://www.cancerrxgene.org/">GDSC</a> and the Cancer Cell Line Encyclopedia <a href="https://portals.broadinstitute.org/ccle">CCLE</a> are large drug sensitivity datasets published in seminal studies in Nature, <a href="https://www.nature.com/articles/nature11005">Garnett et al., https://www.nature.com/articles/nature11005, Nature (2012)</a> and <a href="https://www.nature.com/articles/nature11003">Barretina et al., The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity, Nature (2012)</a>, respectively.</p>
</div>
<div id="drug-perturbation-datasets" class="section level4">
<h4 class="hasAnchor">
<a href="#drug-perturbation-datasets" class="anchor"></a>Drug Perturbation Datasets</h4>
<p>Drug perturbation datasets refer to pharmacogenomic data where gene expression profiles are measured before and after short-term or medium term (e.g., 6h, 24h) drug treatment to identify genes that are up- and down-regulated due to the drug treatment. These datasets can be to classify drug (drug taxonomy), infer their mechanism of action, or find drugs with similar effects (drug repurposing).</p>
<div class="figure">
<img src="figures/Drug_perturbation.png" alt=""><p class="caption">Schematic view of drug perturbation datasets</p>
</div>
<p>Large drug perturbation data have been generated within the Connectivity Map Project <a href="https://clue.io/cmap">CAMP</a>, with CMAPv2 and CMAPv3 available from <code>PharmacoGx</code>, published in <a href="http://science.sciencemag.org/content/313/5795/1929">Lamb et al., The Connectivity Map: Using Gene-Expression Signatures to Connect Small Molecules, Genes, and Disease, Science (2006)</a> and <a href="https://www.cell.com/cell/abstract/S0092-8674(17)31309-0">Subramanian et al., A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles, Cell (2017)</a>, respectively.</p>
</div>
</div>
<div id="exploring-other-treatment-types" class="section level3">
<h3 class="hasAnchor">
<a href="#exploring-other-treatment-types" class="anchor"></a>Exploring Other Treatment Types</h3>
<p>In addition to PharmacoGx, there is a suite of packages in Bioconductor for exploring public high throughput screening data. For Sensitivity datasets, <em><a href="https://bioconductor.org/packages/3.12/Xeva">Xeva</a></em> provides access to public drug screening datasets in Patient Derived Xenograft models, including providing access to the Novartis PDX Encyclopedia dataset, published in <a href="https://www.nature.com/articles/nm.3954">Gao, H. et al. High-throughput screening using patient-derived tumor xenografts to predict clinical trial drug response, Nature Medicine.</a>. Additionally, <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> is currently available in the development branch of Bioconductor, providing access to cell line screening data for response to radiation treatment, featuring data from <a href="https://www.nature.com/articles/ncomms11428">Yard, B. D. et al. A genetic basis for the variation in the vulnerability of cancer to DNA damage. Nature Communications 7, 11428 (2016).</a>.</p>
<p>The <a href="http://bhklab.ca/bioc2020workshop/articles/GxWorkshop.html">Biomarker Discovery from High Throughput Screening Datasets</a> workshop at Bioc2020 goes into depth about using these packages for accessing and exploring public sensitivity datasets.</p>
<p>In addition, Bioconductor also includes the <em><a href="https://bioconductor.org/packages/3.12/ToxicoGx">ToxicoGx</a></em> package, which provides access to Sensitivity and Perturbation datasets characterizing <em>in vitro</em> response of human tissue to toxicant exposure, currently providing access to data from the <a href="https://pubmed.ncbi.nlm.nih.gov/25313160/">TGGates</a> and <a href="https://ntp.niehs.nih.gov/data/drugmatrix/">DrugMatrix</a> datasets. More information about the ToxicoGx package can be found at the <a href="https://doi.org/10.7490/f1000research.1118037.1">Bioc2020 poster presentation on the package</a>.</p>
</div>
</div>
<div id="accessing-the-cancer-genome-atlas-tcga" class="section level2">
<h2 class="hasAnchor">
<a href="#accessing-the-cancer-genome-atlas-tcga" class="anchor"></a>Accessing The Cancer Genome Atlas (TCGA)</h2>
<p>We summarize two approaches to accessing TCGA data:</p>
<ol style="list-style-type: decimal">
<li>
<em><a href="https://bioconductor.org/packages/3.12/TCGAbiolinks">TCGAbiolinks</a></em>:
<ol style="list-style-type: lower-alpha">
<li>data access through <em><a href="https://bioconductor.org/packages/3.12/GenomicDataCommons">GenomicDataCommons</a></em>
</li>
<li>provides data both from the legacy Firehose pipeline used by the TCGA publications (alignments based on hg18 and hg19 builds<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>), and the GDC harmonized GRCh38 pipeline<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</li>
<li>downloads files from the Genomic Data Commons, and provides conversion to <code>(Ranged)SummarizedExperiment</code> where possible</li>
</ol>
</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedTCGAData">curatedTCGAData</a></em>:
<ol style="list-style-type: lower-alpha">
<li>data access through <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>
</li>
<li>provides data from the legacy Firehose pipeline<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>
</li>
<li>provides individual assays as <code>(Ranged)SummarizedExperiment</code> and <code>RaggedExperiment</code>, integrates multiple assays within and across cancer types using <code>MultiAssayExperiment</code>
</li>
</ol>
</li>
</ol>
<div id="tcgabiolinks" class="section level3">
<h3 class="hasAnchor">
<a href="#tcgabiolinks" class="anchor"></a>TCGAbiolinks</h3>
<p>We demonstrate here generating a <code>RangedSummarizedExperiment</code> for RNA-seq data from adrenocortical carcinoma (ACC). For additional information and options, see the TCGAbiolinks vignettes<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<p>Load packages:</p>
<p>Search for matching data:</p>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TCGAbiolinks</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SummarizedExperiment</span>)
<span class="no">query</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCquery.html">GDCquery</a></span>(<span class="kw">project</span> <span class="kw">=</span> <span class="st">"TCGA-ACC"</span>,
                           <span class="kw">data.category</span> <span class="kw">=</span> <span class="st">"Gene expression"</span>,
                           <span class="kw">data.type</span> <span class="kw">=</span> <span class="st">"Gene expression quantification"</span>,
                           <span class="kw">platform</span> <span class="kw">=</span> <span class="st">"Illumina HiSeq"</span>,
                           <span class="kw">file.type</span>  <span class="kw">=</span> <span class="st">"normalized_results"</span>,
                           <span class="kw">experimental.strategy</span> <span class="kw">=</span> <span class="st">"RNA-Seq"</span>,
                           <span class="kw">legacy</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>Download data and convert it to <code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb72"><html><body><pre class="r">#gdcdir &lt;- file.path("Waldron_PublicData", "GDCdata")
#GDCdownload(query, method = "api", files.per.chunk = 10,
#            directory = gdcdir)
#ACCse &lt;- GDCprepare(query, directory = gdcdir)
#ACCse</pre></body></html></div>
</div>
<div id="curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects" class="anchor"></a>curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</h3>
<p><em><a href="https://bioconductor.org/packages/3.12/curatedTCGAData">curatedTCGAData</a></em> does not interface with the Genomic Data Commons, but downloads data from Bioconductor’s <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>.</p>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">curatedTCGAData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">MultiAssayExperiment</span>)</pre></body></html></div>
<p>By default, the <code><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData()</a></code> function will only show available datasets, and not download anything. The arguments are shown here only for demonstration, the same result is obtained with no arguments:</p>
<div class="sourceCode" id="cb74"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span>(<span class="kw">diseaseCode</span> <span class="kw">=</span> <span class="st">"*"</span>, <span class="kw">assays</span> <span class="kw">=</span> <span class="st">"*"</span>)</pre></body></html></div>
<pre><code>## Please see the list below for available cohorts and assays</code></pre>
<pre><code>## Available Cancer codes:
##  ACC BLCA BRCA CESC CHOL COAD DLBC ESCA GBM HNSC KICH
##  KIRC KIRP LAML LGG LIHC LUAD LUSC MESO OV PAAD PCPG
##  PRAD READ SARC SKCM STAD TGCT THCA THYM UCEC UCS UVM 
## Available Data Types:
##  CNACGH CNACGH_CGH_hg_244a
##  CNACGH_CGH_hg_415k_g4124a CNASeq CNASNP
##  CNVSNP GISTIC_AllByGene GISTIC_Peaks
##  GISTIC_ThresholdedByGene Methylation
##  Methylation_methyl27 Methylation_methyl450
##  miRNAArray miRNASeqGene mRNAArray
##  mRNAArray_huex mRNAArray_TX_g4502a
##  mRNAArray_TX_g4502a_1
##  mRNAArray_TX_ht_hg_u133a Mutation
##  RNASeq2GeneNorm RNASeqGene RPPAArray</code></pre>
<p>Check potential files to be downloaded for adrenocortical carcinoma (ACC):</p>
<div class="sourceCode" id="cb77"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span>(<span class="kw">diseaseCode</span> <span class="kw">=</span> <span class="st">"ACC"</span>)</pre></body></html></div>
<pre><code>##                                    Title DispatchClass
## 1                    ACC_CNASNP-20160128           Rda
## 2                    ACC_CNVSNP-20160128           Rda
## 4          ACC_GISTIC_AllByGene-20160128           Rda
## 5              ACC_GISTIC_Peaks-20160128           Rda
## 6  ACC_GISTIC_ThresholdedByGene-20160128           Rda
## 8        ACC_Methylation-20160128_assays        H5File
## 9            ACC_Methylation-20160128_se           Rds
## 10             ACC_miRNASeqGene-20160128           Rda
## 11                 ACC_Mutation-20160128           Rda
## 12          ACC_RNASeq2GeneNorm-20160128           Rda
## 13                ACC_RPPAArray-20160128           Rda</code></pre>
<p>Actually download the reverse phase protein array (RPPA) and RNA-seq data for ACC</p>
<div class="sourceCode" id="cb79"><html><body><pre class="r"><span class="no">ACCmae</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span>(<span class="st">"ACC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"RPPAArray"</span>, <span class="st">"RNASeq2GeneNorm"</span>),
                          <span class="kw">dry.run</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">ACCmae</span></pre></body></html></div>
<pre><code>## A MultiAssayExperiment object of 2 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 2:
##  [1] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns
##  [2] ACC_RPPAArray-20160128: SummarizedExperiment with 192 rows and 46 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
<p><strong>Note</strong>. Data will be downloaded the first time the above command is run; subsequent times it will be loaded from local cache.</p>
<p>This object contains 822 columns of clinical, pathological, specimen, and subtypes data in its <code>colData</code>, merged from all available data levels (1-4) of the Firehose pipeline:</p>
<div class="sourceCode" id="cb81"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">ACCmae</span>))</pre></body></html></div>
<pre><code>## [1]  79 822</code></pre>
<div class="sourceCode" id="cb83"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">ACCmae</span>)))</pre></body></html></div>
<pre><code>## [1] "patientID"             "years_to_birth"        "vital_status"         
## [4] "days_to_death"         "days_to_last_followup" "tumor_tissue_site"</code></pre>
<p>See the <em><a href="https://bioconductor.org/packages/3.12/MultiAssayExperiment">MultiAssayExperiment</a></em> vignette <span class="citation">(Ramos et al. 2017)</span> and the <em>Workflow for Multi-omics Analysis with MultiAssayExperiment</em> workshop for details on using this object.</p>
<div id="subtype-information" class="section level4">
<h4 class="hasAnchor">
<a href="#subtype-information" class="anchor"></a>Subtype information</h4>
<p>Some cancer datasets contain associated subtype information within the clinical datasets provided. This subtype information is included in the metadata of <code>colData</code> of the <code>MultiAssayExperiment</code> object. To obtain these variable names, run the <code>metadata</code> function on the <code>colData</code> of the object such as:</p>
<div class="sourceCode" id="cb85"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">metadata</span>(<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(<span class="no">ACCmae</span>))<span class="kw">[[</span><span class="st">"subtypes"</span>]])</pre></body></html></div>
<pre><code>##         ACC_annotations   ACC_subtype
## 1            Patient_ID        SAMPLE
## 2 histological_subtypes     Histology
## 3         mrna_subtypes       C1A/C1B
## 4         mrna_subtypes       mRNA_K4
## 5                  cimp    MethyLevel
## 6     microrna_subtypes miRNA cluster</code></pre>
</div>
</div>
</div>
<div id="recount-reproducible-rna-seq-analysis-using-recount2" class="section level2">
<h2 class="hasAnchor">
<a href="#recount-reproducible-rna-seq-analysis-using-recount2" class="anchor"></a>recount: Reproducible RNA-seq Analysis Using recount2</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em><span class="citation">(Collado-Torres et al. 2017)</span> package provides uniformly processed <code>RangedSummarizedExperiment</code> objects at the gene, exon, or exon-exon junctions level, the raw counts, the phenotype metadata used, the urls to sample coverage bigWig files and mean coverage bigWig file, for every study available. The <code>RangedSummarizedExperiment</code> objects can be used for differential expression analysis. These are also accessible through a web interface.<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p>
<pre><code>## No methods found in package 'IRanges' for request: 'values&lt;-' when loading 'derfinder'</code></pre>
<pre><code>## No methods found in package 'IRanges' for request: 'values' when loading 'derfinder'</code></pre>
<p><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> provides a search function:</p>
<div class="sourceCode" id="cb89"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">recount</span>)
<span class="no">project_info</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search</a></span>(<span class="st">'GSE32465'</span>)</pre></body></html></div>
<p>It is not an ExperimentHub package, so downloading and serializing is slightly more involved in involves two steps: first, download the gene-level RangedSummarizedExperiment data:</p>
<div class="sourceCode" id="cb90"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span>(<span class="no">project_info</span>$<span class="no">project</span>)</pre></body></html></div>
<pre><code>## 2020-07-28 08:46:29 downloading file rse_gene.Rdata to SRP009615</code></pre>
<p>followed by loading the data</p>
<div class="sourceCode" id="cb92"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">project_info</span>$<span class="no">project</span>, <span class="st">'rse_gene.Rdata'</span>))</pre></body></html></div>
</div>
<div id="curateddata-packages-for-standardized-cancer-transcriptomes" class="section level2">
<h2 class="hasAnchor">
<a href="#curateddata-packages-for-standardized-cancer-transcriptomes" class="anchor"></a>curated*Data packages for standardized cancer transcriptomes</h2>
<p>There are focused databases of cancer microarray data for several cancer types, which can be useful for researchers of those cancer types or for methodological development:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedOvarianData">curatedOvarianData</a></em><span class="citation">(Ganzfried et al. 2013)</span>: Clinically Annotated Data for the Ovarian Cancer Transcriptome (data available with additional options through the <em><a href="https://bioconductor.org/packages/3.12/MetaGxOvarian">MetaGxOvarian</a></em> package).</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedBladderData">curatedBladderData</a></em>: Clinically Annotated Data for the Bladder Cancer Transcriptome</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/curatedCRCData">curatedCRCData</a></em>: Clinically Annotated Data for the Colorectal Cancer Transcriptome</li>
</ul>
<p>These provide data from the Gene Expression Omnibus and other sources, but use a formally vocabulary for clinicopathological data and use a common pipeline for preprocessing of microarray data (for Affymetrix, other for other platforms the processed data are provided as processed by original study authors), merging probesets, and mapping to gene symbols. The pipeline is described by <span class="citation">Ganzfried et al. (2013)</span>.</p>
</div>
<div id="microbiome-data" class="section level2">
<h2 class="hasAnchor">
<a href="#microbiome-data" class="anchor"></a>Microbiome data</h2>
<p>Bioconductor provides curated resources of microbiome data. Most microbiome data are generated either by targeted amplicon sequencing (usually of variable regions of the 16S ribosomal RNA gene) or by metagenomic shotgun sequencing (MGX). These two approaches are analyzed by different sequence analysis tools, but downstream statistical and ecological analysis can involve any of the following types of data:</p>
<ul>
<li>taxonomic abundance at different levels of the taxonomic hierarchy</li>
<li>phylogenetic distances and the phylogenetic tree of life</li>
<li>metabolic potential of the microbiome</li>
<li>abundance of microbial genes and gene families</li>
</ul>
<p>A review of types and properties of microbiome data is provided by <span class="citation">(Morgan and Huttenhower 2012)</span>.</p>
<div id="curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub" class="section level3">
<h3 class="hasAnchor">
<a href="#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub" class="anchor"></a>curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</h3>
<p><code>curatedMetagenomicData</code><span class="citation">(Pasolli et al. 2017)</span> provides 6 types of processed data for &gt;30 publicly available whole-metagenome shotgun sequencing datasets (obtained from the Sequence Read Archive):</p>
<ol style="list-style-type: decimal">
<li>Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level</li>
<li>Presence of unique, clade-specific markers</li>
<li>Abundance of unique, clade-specific markers</li>
<li>Abundance of gene families</li>
<li>Metabolic pathway coverage</li>
<li>Metabolic pathway abundance</li>
</ol>
<p>Types 1-3 are generated by <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a>; 4-6 are generated by <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a>.</p>
<p>Currently, <code>curatedMetagenomicData</code> provides:</p>
<ul>
<li>10199 samples from 57 datasets, primarily of the human gut but including body sites profiled in the Human Microbiome Project</li>
<li>Processed data from whole-metagenome shotgun metagenomics, with manually-curated metadata, as integrated and documented Bioconductor ExpressionSet objects</li>
<li>~80 fields of specimen metadata from original papers, supplementary files, and websites, with manual curation to standardize annotations</li>
<li>Processing of data through the <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a> pipeline for taxonomic abundance, and <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a> pipeline for metabolic analysis</li>
<li>These represent ~100TB of raw sequencing data, but the processed data provided are much smaller.</li>
</ul>
<p>These datasets are documented in the <a href="https://bioconductor.org/packages/devel/data/experiment/manuals/curatedMetagenomicData/man/curatedMetagenomicData.pdf">reference manual</a>.</p>
<p>This is an <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em> package, and its main workhorse function is <code><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData()</a></code>:</p>
<p>The manually curated metadata for all available samples are provided in a single table <code>combined_metadata</code>:</p>
<div class="sourceCode" id="cb93"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">curatedMetagenomicData</span>)
?<span class="no">combined_metadata</span>
<span class="fu"><a href="https://rdrr.io/r/utils/View.html">View</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="no">combined_metadata</span>))</pre></body></html></div>
<p>The main function provides a <code>list</code> of <code>ExpressionSet</code> objects:</p>
<div class="sourceCode" id="cb94"><html><body><pre class="r"><span class="no">oral</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"BritoIL_2016.metaphlan_bugs_list.oralcavity"</span>,
          <span class="st">"Castro-NallarE_2015.metaphlan_bugs_list.oralcavity"</span>)
<span class="no">esl</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData</a></span>(<span class="no">oral</span>, <span class="kw">dryrun</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb95"><html><body><pre class="r"><span class="no">esl</span></pre></body></html></div>
<pre><code>## List of length 2
## names(2): BritoIL_2016.metaphlan_bugs_list.oralcavity ...</code></pre>
<p>These <code>ExpressionSet</code> objects can also be converted to <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> object for ecological analysis and differential abundance analysis using the <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> package, using the <code><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq()</a></code> function:</p>
<div class="sourceCode" id="cb97"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq</a></span>( <span class="no">esl</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="kw">phylogenetictree</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Loading required namespace: phyloseq</code></pre>
<pre><code>## Warning: `data_frame()` is deprecated as of tibble 1.1.0.
## Please use `tibble()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 535 taxa and 140 samples ]
## sample_data() Sample Data:       [ 140 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 535 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 535 tips and 534 internal nodes ]</code></pre>
<p>See the documentation of <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> for more on ecological and differential abundance analysis of the microbiome.</p>
</div>
<div id="hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project" class="section level3">
<h3 class="hasAnchor">
<a href="#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project" class="anchor"></a>HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</h3>
<div class="sourceCode" id="cb101"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">HMP16SData</span>))</pre></body></html></div>
<pre><code>## snapshotDate(): 2020-07-10</code></pre>
<p><em><a href="https://bioconductor.org/packages/3.12/HMP16SData">HMP16SData</a></em><span class="citation">(Schiffer et al. 2018)</span> is a Bioconductor ExperimentData package of the Human Microbiome Project (HMP) 16S rRNA sequencing data. Taxonomic count data files are provided as downloaded from the HMP Data Analysis and Coordination Center from its QIIME pipeline. Processed data is provided as <code>SummarizedExperiment</code> class objects via <em><a href="https://bioconductor.org/packages/3.12/ExperimentHub">ExperimentHub</a></em>. Like other ExperimentHub-based packages, a convenience function does downloading, automatic local caching, and serializing of a Bioconductor data class. This returns taxonomic counts from the V1-3 variable region of the 16S rRNA gene, along with the unrestricted participant data and phylogenetic tree.</p>
<div class="sourceCode" id="cb103"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/HMP16SData/man/V13.html">V13</a></span>()</pre></body></html></div>
<pre><code>## class: SummarizedExperiment 
## dim: 43140 2898 
## metadata(2): experimentData phylogeneticTree
## assays(1): 16SrRNA
## rownames(43140): OTU_97.1 OTU_97.10 ... OTU_97.9997 OTU_97.9999
## rowData names(7): CONSENSUS_LINEAGE SUPERKINGDOM ... FAMILY GENUS
## colnames(2898): 700013549 700014386 ... 700114963 700114965
## colData names(7): RSID VISITNO ... HMP_BODY_SUBSITE SRS_SAMPLE_ID</code></pre>
<p>This can also be converted to <em><a href="https://bioconductor.org/packages/3.12/phyloseq">phyloseq</a></em> for ecological and differential abundance analysis; see the <em><a href="https://bioconductor.org/packages/3.12/HMP16SData">HMP16SData</a></em> vignette for details.</p>
</div>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h2>
<div id="refs" class="references">
<div id="ref-Collado-Torres2017-ww">
<p>Collado-Torres, Leonardo, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, and Jeffrey T Leek. 2017. “Reproducible RNA-seq Analysis Using Recount2.” <em>Nature Biotechnology</em> 35 (4): 319–21. <a href="https://doi.org/10.1038/nbt.3838">https://doi.org/10.1038/nbt.3838</a>.</p>
</div>
<div id="ref-Sean2007-cv">
<p>Davis, Sean R., and Paul S Meltzer. 2007. “GEOquery: A Bridge Between the Gene Expression Omnibus (GEO) and BioConductor.” <em>Bioinformatics</em> 23 (14): 1846–7. <a href="https://doi.org/10.1093/bioinformatics/btm254">https://doi.org/10.1093/bioinformatics/btm254</a>.</p>
</div>
<div id="ref-Ganzfried2013-mc">
<p>Ganzfried, Benjamin Frederick, Markus Riester, Benjamin Haibe-Kains, Thomas Risch, Svitlana Tyekucheva, Ina Jazic, Xin Victoria Wang, et al. 2013. “curatedOvarianData: Clinically Annotated Data for the Ovarian Cancer Transcriptome.” <em>Database: The Journal of Biological Databases and Curation</em> 2013 (April): bat013. <a href="https://doi.org/10.1093/database/bat013">https://doi.org/10.1093/database/bat013</a>.</p>
</div>
<div id="ref-Morgan2012-mq">
<p>Morgan, Xochitl C, and Curtis Huttenhower. 2012. “Chapter 12: Human Microbiome Analysis.” <em>PLoS Computational Biology</em> 8 (12): e1002808. <a href="https://doi.org/10.1371/journal.pcbi.1002808">https://doi.org/10.1371/journal.pcbi.1002808</a>.</p>
</div>
<div id="ref-Pasolli2017-gf">
<p>Pasolli, Edoardo, Lucas Schiffer, Paolo Manghi, Audrey Renson, Valerie Obenchain, Duy Tin Truong, Francesco Beghini, et al. 2017. “Accessible, Curated Metagenomic Data Through ExperimentHub.” <em>Nature Methods</em> 14 (11): 1023–4. <a href="https://doi.org/10.1038/nmeth.4468">https://doi.org/10.1038/nmeth.4468</a>.</p>
</div>
<div id="ref-Ramos2017-nq">
<p>Ramos, Marcel, Lucas Schiffer, Angela Re, Rimsha Azhar, Azfar Basunia, Carmen Rodriguez, Tiffany Chan, et al. 2017. “Software for the Integration of Multiomics Experiments in Bioconductor.” <em>Cancer Research</em> 77 (21): e39–e42. <a href="https://doi.org/10.1158/0008-5472.CAN-17-0344">https://doi.org/10.1158/0008-5472.CAN-17-0344</a>.</p>
</div>
<div id="ref-Schiffer2018-qy">
<p>Schiffer, Lucas, Rimsha Azhar, Lori Shepherd, Marcel Ramos, Ludwig Geistlinger, Curtis Huttenhower, Jennifer B Dowd, Nicola Segata, and Levi Waldron. 2018. “HMP16SData: Efficient Access to the Human Microbiome Project Through Bioconductor.” <em>bioRxiv</em>. <a href="https://doi.org/10.1101/299115">https://doi.org/10.1101/299115</a>.</p>
</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>See individual function and methods documentation for specific details.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing" class="uri">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0" class="uri">https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing" class="uri">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html</a><a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p><a href="https://jhubiostatistics.shinyapps.io/recount/" class="uri">https://jhubiostatistics.shinyapps.io/recount/</a><a href="#fnref6" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sean Davis, Benjamin Haibe-Kains, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
